{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### In a nutshell:\n",
    "- Regulatory Build elements will be intersected with (A) variants and (B) TSS to obtain genomic ranges.\n",
    "- A and B ranges will be intersected with the different datasets.\n",
    "- A-B contact will be calculated from HiC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import pybedtools\n",
    "import gffpandas.gffpandas as gffpd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Reg build elements"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "regbuild_file='/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/original-data/RegBuild/homo_sapiens.GRCh38.CD14_monocyte_1.Regulatory_Build.regulatory_activity.20190329.gff'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/site-packages/gffpandas/gffpandas.py:6: DtypeWarning: Columns (0) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  return Gff3DataFrame(input_file)\n"
     ]
    }
   ],
   "source": [
    "regbuild=gffpd.read_gff3(regbuild_file).attributes_to_columns()[['seq_id','bound_start','bound_end','regulatory_feature_stable_id','type','activity']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "regbuild['seq_id']='chr'+regbuild['seq_id'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "regbuild=regbuild[regbuild['seq_id'] == 'chr1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "regbuild=regbuild.iloc[:,0:4].merge(regbuild[['regulatory_feature_stable_id','type','activity']].replace({'REPRESSED':1,'INACTIVE':2,'POISED':3,'ACTIVE':4,'.':0}).pivot_table(values=['activity'], index=['regulatory_feature_stable_id'],columns='type').droplevel(0,axis=1),how='left',left_on='regulatory_feature_stable_id',right_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>bound_start</th>\n",
       "      <th>bound_end</th>\n",
       "      <th>regulatory_feature_stable_id</th>\n",
       "      <th>CTCF_binding_site</th>\n",
       "      <th>TF_binding_site</th>\n",
       "      <th>enhancer</th>\n",
       "      <th>open_chromatin_region</th>\n",
       "      <th>promoter</th>\n",
       "      <th>promoter_flanking_region</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>20264</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100034201</td>\n",
       "      <td>100035200</td>\n",
       "      <td>ENSR00000253249</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20265</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100127601</td>\n",
       "      <td>100127800</td>\n",
       "      <td>ENSR00000366775</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20266</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100130401</td>\n",
       "      <td>100130600</td>\n",
       "      <td>ENSR00000928323</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20267</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100135001</td>\n",
       "      <td>100135200</td>\n",
       "      <td>ENSR00000366777</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20268</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100147801</td>\n",
       "      <td>100148400</td>\n",
       "      <td>ENSR00000928324</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74013</th>\n",
       "      <td>chr1</td>\n",
       "      <td>99738879</td>\n",
       "      <td>99739345</td>\n",
       "      <td>ENSR00000010583</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74014</th>\n",
       "      <td>chr1</td>\n",
       "      <td>99745620</td>\n",
       "      <td>99746316</td>\n",
       "      <td>ENSR00000366709</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74015</th>\n",
       "      <td>chr1</td>\n",
       "      <td>99772574</td>\n",
       "      <td>99773024</td>\n",
       "      <td>ENSR00000366717</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74016</th>\n",
       "      <td>chr1</td>\n",
       "      <td>99773776</td>\n",
       "      <td>99774196</td>\n",
       "      <td>ENSR00000366718</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74017</th>\n",
       "      <td>chr1</td>\n",
       "      <td>99872095</td>\n",
       "      <td>99872399</td>\n",
       "      <td>ENSR00000366741</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>53754 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      seq_id bound_start  bound_end regulatory_feature_stable_id  \\\n",
       "20264   chr1   100034201  100035200              ENSR00000253249   \n",
       "20265   chr1   100127601  100127800              ENSR00000366775   \n",
       "20266   chr1   100130401  100130600              ENSR00000928323   \n",
       "20267   chr1   100135001  100135200              ENSR00000366777   \n",
       "20268   chr1   100147801  100148400              ENSR00000928324   \n",
       "...      ...         ...        ...                          ...   \n",
       "74013   chr1    99738879   99739345              ENSR00000010583   \n",
       "74014   chr1    99745620   99746316              ENSR00000366709   \n",
       "74015   chr1    99772574   99773024              ENSR00000366717   \n",
       "74016   chr1    99773776   99774196              ENSR00000366718   \n",
       "74017   chr1    99872095   99872399              ENSR00000366741   \n",
       "\n",
       "       CTCF_binding_site  TF_binding_site  enhancer  open_chromatin_region  \\\n",
       "20264                4.0              NaN       NaN                    NaN   \n",
       "20265                2.0              NaN       NaN                    NaN   \n",
       "20266                2.0              NaN       NaN                    NaN   \n",
       "20267                2.0              NaN       NaN                    NaN   \n",
       "20268                2.0              NaN       NaN                    NaN   \n",
       "...                  ...              ...       ...                    ...   \n",
       "74013                NaN              2.0       NaN                    NaN   \n",
       "74014                NaN              2.0       NaN                    NaN   \n",
       "74015                NaN              2.0       NaN                    NaN   \n",
       "74016                NaN              2.0       NaN                    NaN   \n",
       "74017                NaN              2.0       NaN                    NaN   \n",
       "\n",
       "       promoter  promoter_flanking_region  \n",
       "20264       NaN                       NaN  \n",
       "20265       NaN                       NaN  \n",
       "20266       NaN                       NaN  \n",
       "20267       NaN                       NaN  \n",
       "20268       NaN                       NaN  \n",
       "...         ...                       ...  \n",
       "74013       NaN                       NaN  \n",
       "74014       NaN                       NaN  \n",
       "74015       NaN                       NaN  \n",
       "74016       NaN                       NaN  \n",
       "74017       NaN                       NaN  \n",
       "\n",
       "[53754 rows x 10 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regbuild"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "regbuild_bed=pybedtools.BedTool.from_dataframe(regbuild)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## eQTL data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "header=pd.read_csv('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/original-data/eQTL_catalogue/lead_pairs/split/header.tsv',sep='\\t').columns.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "chr1=pd.read_csv('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/original-data/eQTL_catalogue/lead_pairs/split/chr1.tsv',sep='\\t',header=None,names=header)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>eqtl_id</th>\n",
       "      <th>Alasoo_2018.macrophage_IFNg.beta</th>\n",
       "      <th>Alasoo_2018.macrophage_IFNg+Salmonella.beta</th>\n",
       "      <th>Alasoo_2018.macrophage_naive.beta</th>\n",
       "      <th>Alasoo_2018.macrophage_Salmonella.beta</th>\n",
       "      <th>BLUEPRINT_PE.T-cell.beta</th>\n",
       "      <th>BLUEPRINT_SE.monocyte.beta</th>\n",
       "      <th>BLUEPRINT_SE.neutrophil.beta</th>\n",
       "      <th>BrainSeq.brain.beta</th>\n",
       "      <th>FUSION.adipose_naive.beta</th>\n",
       "      <th>...</th>\n",
       "      <th>Schwartzentruber_2018.sensory_neuron.se</th>\n",
       "      <th>TwinsUK.blood.se</th>\n",
       "      <th>TwinsUK.fat.se</th>\n",
       "      <th>TwinsUK.LCL.se</th>\n",
       "      <th>TwinsUK.skin.se</th>\n",
       "      <th>van_de_Bunt_2015.pancreatic_islet.se</th>\n",
       "      <th>molecular_trait_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4524</th>\n",
       "      <td>chr1_54490_G_A.ENSG00000238009</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.092465</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>1</td>\n",
       "      <td>54490</td>\n",
       "      <td>chr1_54490_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4523</th>\n",
       "      <td>chr1_63697_T_C.ENSG00000238009</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>1</td>\n",
       "      <td>63697</td>\n",
       "      <td>chr1_63697_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4526</th>\n",
       "      <td>chr1_64649_A_C.ENSG00000238009</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.275543</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>1</td>\n",
       "      <td>64649</td>\n",
       "      <td>chr1_64649_A_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4525</th>\n",
       "      <td>chr1_115746_C_T.ENSG00000238009</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.875146</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>1</td>\n",
       "      <td>115746</td>\n",
       "      <td>chr1_115746_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4596</th>\n",
       "      <td>chr1_115746_C_T.ENSG00000241860</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.430846</td>\n",
       "      <td>0.181977</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000241860</td>\n",
       "      <td>1</td>\n",
       "      <td>115746</td>\n",
       "      <td>chr1_115746_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4670</th>\n",
       "      <td>chr1_248733454_T_C.ENSG00000259823</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.601130</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000259823</td>\n",
       "      <td>1</td>\n",
       "      <td>248733454</td>\n",
       "      <td>chr1_248733454_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4669</th>\n",
       "      <td>chr1_248739879_C_T.ENSG00000259823</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.499980</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000259823</td>\n",
       "      <td>1</td>\n",
       "      <td>248739879</td>\n",
       "      <td>chr1_248739879_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5123</th>\n",
       "      <td>chr1_248739879_C_T.ENSG00000286015</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.617971</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000286015</td>\n",
       "      <td>1</td>\n",
       "      <td>248739879</td>\n",
       "      <td>chr1_248739879_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4671</th>\n",
       "      <td>chr1_248766065_C_T.ENSG00000259823</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.206758</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.122685</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ENSG00000259823</td>\n",
       "      <td>1</td>\n",
       "      <td>248766065</td>\n",
       "      <td>chr1_248766065_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3079</th>\n",
       "      <td>chr1_248809352_A_G.ENSG00000175137</td>\n",
       "      <td>-0.04459</td>\n",
       "      <td>-0.000477</td>\n",
       "      <td>0.02986</td>\n",
       "      <td>0.025294</td>\n",
       "      <td>0.043894</td>\n",
       "      <td>-0.053092</td>\n",
       "      <td>-0.024233</td>\n",
       "      <td>-0.037208</td>\n",
       "      <td>-0.108717</td>\n",
       "      <td>...</td>\n",
       "      <td>0.066296</td>\n",
       "      <td>0.022230</td>\n",
       "      <td>0.017412</td>\n",
       "      <td>0.014351</td>\n",
       "      <td>0.023962</td>\n",
       "      <td>0.051567</td>\n",
       "      <td>ENSG00000175137</td>\n",
       "      <td>1</td>\n",
       "      <td>248809352</td>\n",
       "      <td>chr1_248809352_A_G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5146 rows Ã— 287 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 eqtl_id  Alasoo_2018.macrophage_IFNg.beta  \\\n",
       "4524      chr1_54490_G_A.ENSG00000238009                               NaN   \n",
       "4523      chr1_63697_T_C.ENSG00000238009                               NaN   \n",
       "4526      chr1_64649_A_C.ENSG00000238009                               NaN   \n",
       "4525     chr1_115746_C_T.ENSG00000238009                               NaN   \n",
       "4596     chr1_115746_C_T.ENSG00000241860                               NaN   \n",
       "...                                  ...                               ...   \n",
       "4670  chr1_248733454_T_C.ENSG00000259823                               NaN   \n",
       "4669  chr1_248739879_C_T.ENSG00000259823                               NaN   \n",
       "5123  chr1_248739879_C_T.ENSG00000286015                               NaN   \n",
       "4671  chr1_248766065_C_T.ENSG00000259823                               NaN   \n",
       "3079  chr1_248809352_A_G.ENSG00000175137                          -0.04459   \n",
       "\n",
       "      Alasoo_2018.macrophage_IFNg+Salmonella.beta  \\\n",
       "4524                                          NaN   \n",
       "4523                                          NaN   \n",
       "4526                                          NaN   \n",
       "4525                                          NaN   \n",
       "4596                                          NaN   \n",
       "...                                           ...   \n",
       "4670                                          NaN   \n",
       "4669                                          NaN   \n",
       "5123                                          NaN   \n",
       "4671                                          NaN   \n",
       "3079                                    -0.000477   \n",
       "\n",
       "      Alasoo_2018.macrophage_naive.beta  \\\n",
       "4524                                NaN   \n",
       "4523                                NaN   \n",
       "4526                                NaN   \n",
       "4525                                NaN   \n",
       "4596                                NaN   \n",
       "...                                 ...   \n",
       "4670                                NaN   \n",
       "4669                                NaN   \n",
       "5123                                NaN   \n",
       "4671                                NaN   \n",
       "3079                            0.02986   \n",
       "\n",
       "      Alasoo_2018.macrophage_Salmonella.beta  BLUEPRINT_PE.T-cell.beta  \\\n",
       "4524                                     NaN                       NaN   \n",
       "4523                                     NaN                       NaN   \n",
       "4526                                     NaN                       NaN   \n",
       "4525                                     NaN                       NaN   \n",
       "4596                                     NaN                       NaN   \n",
       "...                                      ...                       ...   \n",
       "4670                                     NaN                       NaN   \n",
       "4669                                     NaN                       NaN   \n",
       "5123                                     NaN                       NaN   \n",
       "4671                                     NaN                       NaN   \n",
       "3079                                0.025294                  0.043894   \n",
       "\n",
       "      BLUEPRINT_SE.monocyte.beta  BLUEPRINT_SE.neutrophil.beta  \\\n",
       "4524                         NaN                      0.092465   \n",
       "4523                         NaN                           NaN   \n",
       "4526                         NaN                      0.275543   \n",
       "4525                         NaN                      0.875146   \n",
       "4596                    0.430846                      0.181977   \n",
       "...                          ...                           ...   \n",
       "4670                         NaN                           NaN   \n",
       "4669                         NaN                           NaN   \n",
       "5123                         NaN                           NaN   \n",
       "4671                         NaN                           NaN   \n",
       "3079                   -0.053092                     -0.024233   \n",
       "\n",
       "      BrainSeq.brain.beta  FUSION.adipose_naive.beta  ...  \\\n",
       "4524                  NaN                        NaN  ...   \n",
       "4523                  NaN                        NaN  ...   \n",
       "4526                  NaN                        NaN  ...   \n",
       "4525                  NaN                        NaN  ...   \n",
       "4596                  NaN                        NaN  ...   \n",
       "...                   ...                        ...  ...   \n",
       "4670             1.601130                        NaN  ...   \n",
       "4669             1.499980                        NaN  ...   \n",
       "5123            -0.617971                        NaN  ...   \n",
       "4671            -0.206758                        NaN  ...   \n",
       "3079            -0.037208                  -0.108717  ...   \n",
       "\n",
       "      Schwartzentruber_2018.sensory_neuron.se  TwinsUK.blood.se  \\\n",
       "4524                                      NaN               NaN   \n",
       "4523                                      NaN               NaN   \n",
       "4526                                      NaN               NaN   \n",
       "4525                                      NaN               NaN   \n",
       "4596                                      NaN               NaN   \n",
       "...                                       ...               ...   \n",
       "4670                                      NaN               NaN   \n",
       "4669                                      NaN               NaN   \n",
       "5123                                      NaN               NaN   \n",
       "4671                                      NaN          0.122685   \n",
       "3079                                 0.066296          0.022230   \n",
       "\n",
       "      TwinsUK.fat.se  TwinsUK.LCL.se  TwinsUK.skin.se  \\\n",
       "4524             NaN             NaN              NaN   \n",
       "4523             NaN             NaN              NaN   \n",
       "4526             NaN             NaN              NaN   \n",
       "4525             NaN             NaN              NaN   \n",
       "4596             NaN             NaN              NaN   \n",
       "...              ...             ...              ...   \n",
       "4670             NaN             NaN              NaN   \n",
       "4669             NaN             NaN              NaN   \n",
       "5123             NaN             NaN              NaN   \n",
       "4671             NaN             NaN              NaN   \n",
       "3079        0.017412        0.014351         0.023962   \n",
       "\n",
       "      van_de_Bunt_2015.pancreatic_islet.se  molecular_trait_id  chromosome  \\\n",
       "4524                                   NaN     ENSG00000238009           1   \n",
       "4523                                   NaN     ENSG00000238009           1   \n",
       "4526                                   NaN     ENSG00000238009           1   \n",
       "4525                                   NaN     ENSG00000238009           1   \n",
       "4596                                   NaN     ENSG00000241860           1   \n",
       "...                                    ...                 ...         ...   \n",
       "4670                                   NaN     ENSG00000259823           1   \n",
       "4669                                   NaN     ENSG00000259823           1   \n",
       "5123                                   NaN     ENSG00000286015           1   \n",
       "4671                                   NaN     ENSG00000259823           1   \n",
       "3079                              0.051567     ENSG00000175137           1   \n",
       "\n",
       "       position             variant  \n",
       "4524      54490      chr1_54490_G_A  \n",
       "4523      63697      chr1_63697_T_C  \n",
       "4526      64649      chr1_64649_A_C  \n",
       "4525     115746     chr1_115746_C_T  \n",
       "4596     115746     chr1_115746_C_T  \n",
       "...         ...                 ...  \n",
       "4670  248733454  chr1_248733454_T_C  \n",
       "4669  248739879  chr1_248739879_C_T  \n",
       "5123  248739879  chr1_248739879_C_T  \n",
       "4671  248766065  chr1_248766065_C_T  \n",
       "3079  248809352  chr1_248809352_A_G  \n",
       "\n",
       "[5146 rows x 287 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chr1.sort_values('position')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "chr1['chromosome']='chr'+chr1['chromosome'].astype(str)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Â A (Variants)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 287,
   "metadata": {},
   "outputs": [],
   "source": [
    "variants=pd.DataFrame({'chromosome':chr1['chromosome'],'position_start':chr1['position'],'position_end':chr1['position'],'variant':chr1['variant']}).drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 288,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position_start</th>\n",
       "      <th>position_end</th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169796199</td>\n",
       "      <td>169796199</td>\n",
       "      <td>chr1_169796199_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169845687</td>\n",
       "      <td>169845687</td>\n",
       "      <td>chr1_169845687_C_CTTCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169821020</td>\n",
       "      <td>169821020</td>\n",
       "      <td>chr1_169821020_T_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169982267</td>\n",
       "      <td>169982267</td>\n",
       "      <td>chr1_169982267_A_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169787407</td>\n",
       "      <td>169787407</td>\n",
       "      <td>chr1_169787407_G_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5140</th>\n",
       "      <td>chr1</td>\n",
       "      <td>233703526</td>\n",
       "      <td>233703526</td>\n",
       "      <td>chr1_233703526_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5141</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208262975</td>\n",
       "      <td>208262975</td>\n",
       "      <td>chr1_208262975_T_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5142</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208270332</td>\n",
       "      <td>208270332</td>\n",
       "      <td>chr1_208270332_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5144</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150320113</td>\n",
       "      <td>150320113</td>\n",
       "      <td>chr1_150320113_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5145</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150259332</td>\n",
       "      <td>150259332</td>\n",
       "      <td>chr1_150259332_C_T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4823 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chromosome  position_start  position_end                  variant\n",
       "0          chr1       169796199     169796199       chr1_169796199_G_A\n",
       "1          chr1       169845687     169845687  chr1_169845687_C_CTTCCA\n",
       "2          chr1       169821020     169821020       chr1_169821020_T_G\n",
       "3          chr1       169982267     169982267       chr1_169982267_A_T\n",
       "4          chr1       169787407     169787407       chr1_169787407_G_C\n",
       "...         ...             ...           ...                      ...\n",
       "5140       chr1       233703526     233703526       chr1_233703526_T_C\n",
       "5141       chr1       208262975     208262975       chr1_208262975_T_G\n",
       "5142       chr1       208270332     208270332       chr1_208270332_C_A\n",
       "5144       chr1       150320113     150320113       chr1_150320113_C_T\n",
       "5145       chr1       150259332     150259332       chr1_150259332_C_T\n",
       "\n",
       "[4823 rows x 4 columns]"
      ]
     },
     "execution_count": 288,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 289,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "variants_bed=pybedtools.BedTool.from_dataframe(variants)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_var_data=variants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## eQTL - RegBuild intersection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "eqtl_regbuild=variants_bed.intersect(regbuild_bed,wa=True,wb=True,loj=True).to_dataframe(names=['var_chr','var_start','var_end','variant','chr','start','end','reg_id']+regbuild.columns[-6:].to_list()).replace('.',np.nan)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "metadata": {},
   "outputs": [],
   "source": [
    "eqtl_regbuild[eqtl_regbuild.columns[-6:]]=eqtl_regbuild[eqtl_regbuild.columns[-6:]].astype('float')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 293,
   "metadata": {},
   "outputs": [],
   "source": [
    "eqtl_regbuild=eqtl_regbuild[eqtl_regbuild.columns[:4]].merge(eqtl_regbuild.groupby('variant',as_index=False).mean()).drop_duplicates(ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 294,
   "metadata": {},
   "outputs": [],
   "source": [
    "eqtl_regbuild=eqtl_regbuild[eqtl_regbuild.columns[3:]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 295,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_names = [(i,'var_'+i) for i in eqtl_regbuild.columns[-8:].values]\n",
    "eqtl_regbuild.rename(columns = dict(new_names), inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 296,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>variant</th>\n",
       "      <th>var_start</th>\n",
       "      <th>var_end</th>\n",
       "      <th>var_CTCF_binding_site</th>\n",
       "      <th>var_TF_binding_site</th>\n",
       "      <th>var_enhancer</th>\n",
       "      <th>var_open_chromatin_region</th>\n",
       "      <th>var_promoter</th>\n",
       "      <th>var_promoter_flanking_region</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_169796199_G_A</td>\n",
       "      <td>169792002.0</td>\n",
       "      <td>169796999.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_169845687_C_CTTCCA</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1_169821020_T_G</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1_169982267_A_T</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_169787407_G_C</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4818</th>\n",
       "      <td>chr1_233703526_T_C</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4819</th>\n",
       "      <td>chr1_208262975_T_G</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4820</th>\n",
       "      <td>chr1_208270332_C_A</td>\n",
       "      <td>208267601.0</td>\n",
       "      <td>208273000.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4821</th>\n",
       "      <td>chr1_150320113_C_T</td>\n",
       "      <td>150320001.0</td>\n",
       "      <td>150320200.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4822</th>\n",
       "      <td>chr1_150259332_C_T</td>\n",
       "      <td>150255202.0</td>\n",
       "      <td>150261999.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4823 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                      variant    var_start      var_end  \\\n",
       "0          chr1_169796199_G_A  169792002.0  169796999.0   \n",
       "1     chr1_169845687_C_CTTCCA         -1.0         -1.0   \n",
       "2          chr1_169821020_T_G         -1.0         -1.0   \n",
       "3          chr1_169982267_A_T         -1.0         -1.0   \n",
       "4          chr1_169787407_G_C         -1.0         -1.0   \n",
       "...                       ...          ...          ...   \n",
       "4818       chr1_233703526_T_C         -1.0         -1.0   \n",
       "4819       chr1_208262975_T_G         -1.0         -1.0   \n",
       "4820       chr1_208270332_C_A  208267601.0  208273000.0   \n",
       "4821       chr1_150320113_C_T  150320001.0  150320200.0   \n",
       "4822       chr1_150259332_C_T  150255202.0  150261999.0   \n",
       "\n",
       "      var_CTCF_binding_site  var_TF_binding_site  var_enhancer  \\\n",
       "0                       NaN                  NaN           NaN   \n",
       "1                       NaN                  NaN           NaN   \n",
       "2                       NaN                  NaN           NaN   \n",
       "3                       NaN                  NaN           NaN   \n",
       "4                       NaN                  NaN           NaN   \n",
       "...                     ...                  ...           ...   \n",
       "4818                    NaN                  NaN           NaN   \n",
       "4819                    NaN                  NaN           NaN   \n",
       "4820                    NaN                  NaN           NaN   \n",
       "4821                    2.0                  NaN           NaN   \n",
       "4822                    NaN                  NaN           NaN   \n",
       "\n",
       "      var_open_chromatin_region  var_promoter  var_promoter_flanking_region  \n",
       "0                           NaN           4.0                           NaN  \n",
       "1                           NaN           NaN                           NaN  \n",
       "2                           NaN           NaN                           NaN  \n",
       "3                           NaN           NaN                           NaN  \n",
       "4                           NaN           NaN                           NaN  \n",
       "...                         ...           ...                           ...  \n",
       "4818                        NaN           NaN                           NaN  \n",
       "4819                        NaN           NaN                           NaN  \n",
       "4820                        NaN           NaN                           1.0  \n",
       "4821                        NaN           NaN                           NaN  \n",
       "4822                        NaN           4.0                           NaN  \n",
       "\n",
       "[4823 rows x 9 columns]"
      ]
     },
     "execution_count": 296,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eqtl_regbuild"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 297,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_var_data=merged_var_data.merge(eqtl_regbuild)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### When not intersecting with an element, var_start - end = variant position"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 298,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-298-cfb1189a6a54>:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  merged_var_data['var_start'][index]=merged_var_data['position_start'][index]\n",
      "<ipython-input-298-cfb1189a6a54>:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  merged_var_data['var_end'][index]=merged_var_data['position_start'][index]\n"
     ]
    }
   ],
   "source": [
    "for index in merged_var_data[merged_var_data['var_start']==-1].index:\n",
    "    merged_var_data['var_start'][index]=merged_var_data['position_start'][index]\n",
    "    merged_var_data['var_end'][index]=merged_var_data['position_start'][index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position_start</th>\n",
       "      <th>position_end</th>\n",
       "      <th>variant</th>\n",
       "      <th>var_start</th>\n",
       "      <th>var_end</th>\n",
       "      <th>var_CTCF_binding_site</th>\n",
       "      <th>var_TF_binding_site</th>\n",
       "      <th>var_enhancer</th>\n",
       "      <th>var_open_chromatin_region</th>\n",
       "      <th>var_promoter</th>\n",
       "      <th>var_promoter_flanking_region</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169796199</td>\n",
       "      <td>169796199</td>\n",
       "      <td>chr1_169796199_G_A</td>\n",
       "      <td>169792002.0</td>\n",
       "      <td>169796999.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169845687</td>\n",
       "      <td>169845687</td>\n",
       "      <td>chr1_169845687_C_CTTCCA</td>\n",
       "      <td>169845687.0</td>\n",
       "      <td>169845687.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169821020</td>\n",
       "      <td>169821020</td>\n",
       "      <td>chr1_169821020_T_G</td>\n",
       "      <td>169821020.0</td>\n",
       "      <td>169821020.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169982267</td>\n",
       "      <td>169982267</td>\n",
       "      <td>chr1_169982267_A_T</td>\n",
       "      <td>169982267.0</td>\n",
       "      <td>169982267.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169787407</td>\n",
       "      <td>169787407</td>\n",
       "      <td>chr1_169787407_G_C</td>\n",
       "      <td>169787407.0</td>\n",
       "      <td>169787407.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4818</th>\n",
       "      <td>chr1</td>\n",
       "      <td>233703526</td>\n",
       "      <td>233703526</td>\n",
       "      <td>chr1_233703526_T_C</td>\n",
       "      <td>233703526.0</td>\n",
       "      <td>233703526.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4819</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208262975</td>\n",
       "      <td>208262975</td>\n",
       "      <td>chr1_208262975_T_G</td>\n",
       "      <td>208262975.0</td>\n",
       "      <td>208262975.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4820</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208270332</td>\n",
       "      <td>208270332</td>\n",
       "      <td>chr1_208270332_C_A</td>\n",
       "      <td>208267601.0</td>\n",
       "      <td>208273000.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4821</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150320113</td>\n",
       "      <td>150320113</td>\n",
       "      <td>chr1_150320113_C_T</td>\n",
       "      <td>150320001.0</td>\n",
       "      <td>150320200.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4822</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150259332</td>\n",
       "      <td>150259332</td>\n",
       "      <td>chr1_150259332_C_T</td>\n",
       "      <td>150255202.0</td>\n",
       "      <td>150261999.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4823 rows Ã— 12 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chromosome  position_start  position_end                  variant  \\\n",
       "0          chr1       169796199     169796199       chr1_169796199_G_A   \n",
       "1          chr1       169845687     169845687  chr1_169845687_C_CTTCCA   \n",
       "2          chr1       169821020     169821020       chr1_169821020_T_G   \n",
       "3          chr1       169982267     169982267       chr1_169982267_A_T   \n",
       "4          chr1       169787407     169787407       chr1_169787407_G_C   \n",
       "...         ...             ...           ...                      ...   \n",
       "4818       chr1       233703526     233703526       chr1_233703526_T_C   \n",
       "4819       chr1       208262975     208262975       chr1_208262975_T_G   \n",
       "4820       chr1       208270332     208270332       chr1_208270332_C_A   \n",
       "4821       chr1       150320113     150320113       chr1_150320113_C_T   \n",
       "4822       chr1       150259332     150259332       chr1_150259332_C_T   \n",
       "\n",
       "        var_start      var_end  var_CTCF_binding_site  var_TF_binding_site  \\\n",
       "0     169792002.0  169796999.0                    NaN                  NaN   \n",
       "1     169845687.0  169845687.0                    NaN                  NaN   \n",
       "2     169821020.0  169821020.0                    NaN                  NaN   \n",
       "3     169982267.0  169982267.0                    NaN                  NaN   \n",
       "4     169787407.0  169787407.0                    NaN                  NaN   \n",
       "...           ...          ...                    ...                  ...   \n",
       "4818  233703526.0  233703526.0                    NaN                  NaN   \n",
       "4819  208262975.0  208262975.0                    NaN                  NaN   \n",
       "4820  208267601.0  208273000.0                    NaN                  NaN   \n",
       "4821  150320001.0  150320200.0                    2.0                  NaN   \n",
       "4822  150255202.0  150261999.0                    NaN                  NaN   \n",
       "\n",
       "      var_enhancer  var_open_chromatin_region  var_promoter  \\\n",
       "0              NaN                        NaN           4.0   \n",
       "1              NaN                        NaN           NaN   \n",
       "2              NaN                        NaN           NaN   \n",
       "3              NaN                        NaN           NaN   \n",
       "4              NaN                        NaN           NaN   \n",
       "...            ...                        ...           ...   \n",
       "4818           NaN                        NaN           NaN   \n",
       "4819           NaN                        NaN           NaN   \n",
       "4820           NaN                        NaN           NaN   \n",
       "4821           NaN                        NaN           NaN   \n",
       "4822           NaN                        NaN           4.0   \n",
       "\n",
       "      var_promoter_flanking_region  \n",
       "0                              NaN  \n",
       "1                              NaN  \n",
       "2                              NaN  \n",
       "3                              NaN  \n",
       "4                              NaN  \n",
       "...                            ...  \n",
       "4818                           NaN  \n",
       "4819                           NaN  \n",
       "4820                           1.0  \n",
       "4821                           NaN  \n",
       "4822                           NaN  \n",
       "\n",
       "[4823 rows x 12 columns]"
      ]
     },
     "execution_count": 299,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_var_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 304,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_var_data['var_start']=merged_var_data['var_start'].astype(int)\n",
    "merged_var_data['var_end']=merged_var_data['var_end'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 305,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_var_bed=pybedtools.BedTool.from_dataframe(merged_var_data[['chromosome','var_start','var_end','variant']])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TODO check effect of mean reg element start and end"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# B (TSS)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TSS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Gencode v35"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "gencode=gffpd.read_gff3('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/original-data/gencode/gencode.v35.annotation.gff3').attributes_to_columns()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "gencode=gencode[gencode['seq_id']=='chr1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "gencode=gencode[gencode['type']=='gene']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "gencode['gene_id']=gencode['gene_id'].str.split('.',expand=True)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 496,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>seq_id</th>\n",
       "      <th>source</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>phase</th>\n",
       "      <th>attributes</th>\n",
       "      <th>ID</th>\n",
       "      <th>...</th>\n",
       "      <th>havana_transcript</th>\n",
       "      <th>hgnc_id</th>\n",
       "      <th>level</th>\n",
       "      <th>ont</th>\n",
       "      <th>protein_id</th>\n",
       "      <th>tag</th>\n",
       "      <th>transcript_id</th>\n",
       "      <th>transcript_name</th>\n",
       "      <th>transcript_support_level</th>\n",
       "      <th>transcript_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ENSG00000223972.5;gene_id=ENSG00000223972.5...</td>\n",
       "      <td>ENSG00000223972.5</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>HGNC:37102</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ENSG00000227232.5;gene_id=ENSG00000227232.5...</td>\n",
       "      <td>ENSG00000227232.5</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>HGNC:38034</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>gene</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ENSG00000278267.1;gene_id=ENSG00000278267.1...</td>\n",
       "      <td>ENSG00000278267.1</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>HGNC:50039</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>29554</td>\n",
       "      <td>31109</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ENSG00000243485.5;gene_id=ENSG00000243485.5...</td>\n",
       "      <td>ENSG00000243485.5</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>HGNC:52482</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>ncRNA_host</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>gene</td>\n",
       "      <td>30366</td>\n",
       "      <td>30503</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ENSG00000284332.1;gene_id=ENSG00000284332.1...</td>\n",
       "      <td>ENSG00000284332.1</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>HGNC:35294</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271064</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>248850006</td>\n",
       "      <td>248859144</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ENSG00000171163.16;gene_id=ENSG00000171163....</td>\n",
       "      <td>ENSG00000171163.16</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>HGNC:26049</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271410</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>248859164</td>\n",
       "      <td>248864796</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ENSG00000227237.1;gene_id=ENSG00000227237.1...</td>\n",
       "      <td>ENSG00000227237.1</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271415</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>248906196</td>\n",
       "      <td>248919946</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ENSG00000185220.12;gene_id=ENSG00000185220....</td>\n",
       "      <td>ENSG00000185220.12</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>HGNC:19399</td>\n",
       "      <td>2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271442</th>\n",
       "      <td>chr1</td>\n",
       "      <td>ENSEMBL</td>\n",
       "      <td>gene</td>\n",
       "      <td>248912690</td>\n",
       "      <td>248912795</td>\n",
       "      <td>.</td>\n",
       "      <td>-</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ENSG00000200495.1;gene_id=ENSG00000200495.1...</td>\n",
       "      <td>ENSG00000200495.1</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>HGNC:48168</td>\n",
       "      <td>3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>271445</th>\n",
       "      <td>chr1</td>\n",
       "      <td>HAVANA</td>\n",
       "      <td>gene</td>\n",
       "      <td>248936581</td>\n",
       "      <td>248937043</td>\n",
       "      <td>.</td>\n",
       "      <td>+</td>\n",
       "      <td>.</td>\n",
       "      <td>ID=ENSG00000233084.2;gene_id=ENSG00000233084.2...</td>\n",
       "      <td>ENSG00000233084.2</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>HGNC:35672</td>\n",
       "      <td>1</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>pseudo_consens</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5476 rows Ã— 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       seq_id   source  type      start        end score strand phase  \\\n",
       "0        chr1   HAVANA  gene      11869      14409     .      +     .   \n",
       "12       chr1   HAVANA  gene      14404      29570     .      -     .   \n",
       "25       chr1  ENSEMBL  gene      17369      17436     .      -     .   \n",
       "28       chr1   HAVANA  gene      29554      31109     .      +     .   \n",
       "36       chr1  ENSEMBL  gene      30366      30503     .      +     .   \n",
       "...       ...      ...   ...        ...        ...   ...    ...   ...   \n",
       "271064   chr1   HAVANA  gene  248850006  248859144     .      -     .   \n",
       "271410   chr1   HAVANA  gene  248859164  248864796     .      +     .   \n",
       "271415   chr1   HAVANA  gene  248906196  248919946     .      +     .   \n",
       "271442   chr1  ENSEMBL  gene  248912690  248912795     .      -     .   \n",
       "271445   chr1   HAVANA  gene  248936581  248937043     .      +     .   \n",
       "\n",
       "                                               attributes                  ID  \\\n",
       "0       ID=ENSG00000223972.5;gene_id=ENSG00000223972.5...   ENSG00000223972.5   \n",
       "12      ID=ENSG00000227232.5;gene_id=ENSG00000227232.5...   ENSG00000227232.5   \n",
       "25      ID=ENSG00000278267.1;gene_id=ENSG00000278267.1...   ENSG00000278267.1   \n",
       "28      ID=ENSG00000243485.5;gene_id=ENSG00000243485.5...   ENSG00000243485.5   \n",
       "36      ID=ENSG00000284332.1;gene_id=ENSG00000284332.1...   ENSG00000284332.1   \n",
       "...                                                   ...                 ...   \n",
       "271064  ID=ENSG00000171163.16;gene_id=ENSG00000171163....  ENSG00000171163.16   \n",
       "271410  ID=ENSG00000227237.1;gene_id=ENSG00000227237.1...   ENSG00000227237.1   \n",
       "271415  ID=ENSG00000185220.12;gene_id=ENSG00000185220....  ENSG00000185220.12   \n",
       "271442  ID=ENSG00000200495.1;gene_id=ENSG00000200495.1...   ENSG00000200495.1   \n",
       "271445  ID=ENSG00000233084.2;gene_id=ENSG00000233084.2...   ENSG00000233084.2   \n",
       "\n",
       "        ... havana_transcript     hgnc_id level   ont protein_id  \\\n",
       "0       ...              None  HGNC:37102     2  None       None   \n",
       "12      ...              None  HGNC:38034     2  None       None   \n",
       "25      ...              None  HGNC:50039     3  None       None   \n",
       "28      ...              None  HGNC:52482     2  None       None   \n",
       "36      ...              None  HGNC:35294     3  None       None   \n",
       "...     ...               ...         ...   ...   ...        ...   \n",
       "271064  ...              None  HGNC:26049     1  None       None   \n",
       "271410  ...              None        None     1  None       None   \n",
       "271415  ...              None  HGNC:19399     2  None       None   \n",
       "271442  ...              None  HGNC:48168     3  None       None   \n",
       "271445  ...              None  HGNC:35672     1  None       None   \n",
       "\n",
       "                   tag transcript_id transcript_name transcript_support_level  \\\n",
       "0                 None          None            None                     None   \n",
       "12                None          None            None                     None   \n",
       "25                None          None            None                     None   \n",
       "28          ncRNA_host          None            None                     None   \n",
       "36                None          None            None                     None   \n",
       "...                ...           ...             ...                      ...   \n",
       "271064            None          None            None                     None   \n",
       "271410            None          None            None                     None   \n",
       "271415            None          None            None                     None   \n",
       "271442            None          None            None                     None   \n",
       "271445  pseudo_consens          None            None                     None   \n",
       "\n",
       "       transcript_type  \n",
       "0                 None  \n",
       "12                None  \n",
       "25                None  \n",
       "28                None  \n",
       "36                None  \n",
       "...                ...  \n",
       "271064            None  \n",
       "271410            None  \n",
       "271415            None  \n",
       "271442            None  \n",
       "271445            None  \n",
       "\n",
       "[5476 rows x 28 columns]"
      ]
     },
     "execution_count": 496,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gencode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [],
   "source": [
    "gencode_simp=gencode[['gene_id','start','end','strand']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [],
   "source": [
    "gencode_simp=gencode_simp.set_index('gene_id')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Simple approach to get tss from Gencode"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "TSS=pd.concat([gencode_simp[gencode_simp['strand']=='+']['start'],gencode_simp[gencode_simp['strand']=='-']['end']]).sort_values().reset_index(name='TSS')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>TSS</th>\n",
       "      <th>chr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>11869</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>17436</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>29554</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>29570</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000284332</td>\n",
       "      <td>30366</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5471</th>\n",
       "      <td>ENSG00000171163</td>\n",
       "      <td>248859144</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5472</th>\n",
       "      <td>ENSG00000227237</td>\n",
       "      <td>248859164</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5473</th>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>248906196</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5474</th>\n",
       "      <td>ENSG00000200495</td>\n",
       "      <td>248912795</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5475</th>\n",
       "      <td>ENSG00000233084</td>\n",
       "      <td>248936581</td>\n",
       "      <td>chr1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5476 rows Ã— 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              gene_id        TSS   chr\n",
       "0     ENSG00000223972      11869  chr1\n",
       "1     ENSG00000278267      17436  chr1\n",
       "2     ENSG00000243485      29554  chr1\n",
       "3     ENSG00000227232      29570  chr1\n",
       "4     ENSG00000284332      30366  chr1\n",
       "...               ...        ...   ...\n",
       "5471  ENSG00000171163  248859144  chr1\n",
       "5472  ENSG00000227237  248859164  chr1\n",
       "5473  ENSG00000185220  248906196  chr1\n",
       "5474  ENSG00000200495  248912795  chr1\n",
       "5475  ENSG00000233084  248936581  chr1\n",
       "\n",
       "[5476 rows x 3 columns]"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TSS"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TSS - eQTLs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "TSS['chr']='chr1'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "TSS_bed=pybedtools.BedTool.from_dataframe(TSS[['chr','TSS','TSS','gene_id']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 361,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "merged_TSS_data=TSS_bed.intersect(regbuild_bed,wa=True,wb=True,loj=True).to_dataframe(names=['TSS_chr','TSS_start','TSS_end','gene_id','chr','start','end','reg_id']+regbuild.columns[-6:].to_list()).replace('.',np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 363,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_TSS_data[merged_TSS_data.columns[-6:]]=merged_TSS_data[merged_TSS_data.columns[-6:]].astype('float')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 365,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_TSS_data=merged_TSS_data[merged_TSS_data.columns[:4]].merge(merged_TSS_data.groupby('TSS_start',as_index=False).mean()).drop_duplicates(ignore_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 369,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_TSS_data=merged_TSS_data.drop(columns=['TSS_start']).rename(columns={'TSS_end':'TSS'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 370,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_names = [(i,'TSS_'+i) for i in merged_TSS_data.columns[-8:].values]\n",
    "merged_TSS_data.rename(columns = dict(new_names), inplace=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### When not intersecting with an element, TSS_start - end = TSS position"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 372,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-372-b691c8e0a114>:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  merged_TSS_data['TSS_start'][index]=merged_TSS_data['TSS'][index]\n",
      "<ipython-input-372-b691c8e0a114>:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  merged_TSS_data['TSS_end'][index]=merged_TSS_data['TSS'][index]\n"
     ]
    }
   ],
   "source": [
    "for index in merged_TSS_data[merged_TSS_data['TSS_start']==-1].index:\n",
    "    merged_TSS_data['TSS_start'][index]=merged_TSS_data['TSS'][index]\n",
    "    merged_TSS_data['TSS_end'][index]=merged_TSS_data['TSS'][index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 373,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TSS_chr</th>\n",
       "      <th>TSS</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>TSS_start</th>\n",
       "      <th>TSS_end</th>\n",
       "      <th>TSS_CTCF_binding_site</th>\n",
       "      <th>TSS_TF_binding_site</th>\n",
       "      <th>TSS_enhancer</th>\n",
       "      <th>TSS_open_chromatin_region</th>\n",
       "      <th>TSS_promoter</th>\n",
       "      <th>TSS_promoter_flanking_region</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>11869.0</td>\n",
       "      <td>11869.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17436</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>17436.0</td>\n",
       "      <td>17436.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>20602.0</td>\n",
       "      <td>33399.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29570</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>20602.0</td>\n",
       "      <td>33399.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30366</td>\n",
       "      <td>ENSG00000284332</td>\n",
       "      <td>20602.0</td>\n",
       "      <td>33399.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5471</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248859144</td>\n",
       "      <td>ENSG00000171163</td>\n",
       "      <td>248856002.0</td>\n",
       "      <td>248862599.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5472</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248859164</td>\n",
       "      <td>ENSG00000227237</td>\n",
       "      <td>248856002.0</td>\n",
       "      <td>248862599.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5473</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248906196</td>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>248904602.0</td>\n",
       "      <td>248908999.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5474</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248912795</td>\n",
       "      <td>ENSG00000200495</td>\n",
       "      <td>248912795.0</td>\n",
       "      <td>248912795.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5475</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248936581</td>\n",
       "      <td>ENSG00000233084</td>\n",
       "      <td>248936577.0</td>\n",
       "      <td>248937136.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5476 rows Ã— 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     TSS_chr        TSS          gene_id    TSS_start      TSS_end  \\\n",
       "0       chr1      11869  ENSG00000223972      11869.0      11869.0   \n",
       "1       chr1      17436  ENSG00000278267      17436.0      17436.0   \n",
       "2       chr1      29554  ENSG00000243485      20602.0      33399.0   \n",
       "3       chr1      29570  ENSG00000227232      20602.0      33399.0   \n",
       "4       chr1      30366  ENSG00000284332      20602.0      33399.0   \n",
       "...      ...        ...              ...          ...          ...   \n",
       "5471    chr1  248859144  ENSG00000171163  248856002.0  248862599.0   \n",
       "5472    chr1  248859164  ENSG00000227237  248856002.0  248862599.0   \n",
       "5473    chr1  248906196  ENSG00000185220  248904602.0  248908999.0   \n",
       "5474    chr1  248912795  ENSG00000200495  248912795.0  248912795.0   \n",
       "5475    chr1  248936581  ENSG00000233084  248936577.0  248937136.0   \n",
       "\n",
       "      TSS_CTCF_binding_site  TSS_TF_binding_site  TSS_enhancer  \\\n",
       "0                       NaN                  NaN           NaN   \n",
       "1                       NaN                  NaN           NaN   \n",
       "2                       NaN                  NaN           NaN   \n",
       "3                       NaN                  NaN           NaN   \n",
       "4                       NaN                  NaN           NaN   \n",
       "...                     ...                  ...           ...   \n",
       "5471                    NaN                  NaN           NaN   \n",
       "5472                    NaN                  NaN           NaN   \n",
       "5473                    NaN                  NaN           NaN   \n",
       "5474                    NaN                  NaN           NaN   \n",
       "5475                    NaN                  2.0           NaN   \n",
       "\n",
       "      TSS_open_chromatin_region  TSS_promoter  TSS_promoter_flanking_region  \n",
       "0                           NaN           NaN                           NaN  \n",
       "1                           NaN           NaN                           NaN  \n",
       "2                           NaN           4.0                           NaN  \n",
       "3                           NaN           4.0                           NaN  \n",
       "4                           NaN           4.0                           NaN  \n",
       "...                         ...           ...                           ...  \n",
       "5471                        NaN           3.0                           NaN  \n",
       "5472                        NaN           3.0                           NaN  \n",
       "5473                        NaN           4.0                           NaN  \n",
       "5474                        NaN           NaN                           NaN  \n",
       "5475                        NaN           NaN                           NaN  \n",
       "\n",
       "[5476 rows x 11 columns]"
      ]
     },
     "execution_count": 373,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_TSS_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_TSS_data['TSS_start']=merged_TSS_data['TSS_start'].astype(int)\n",
    "merged_TSS_data['TSS_end']=merged_TSS_data['TSS_end'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 375,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_TSS_bed=pybedtools.BedTool.from_dataframe(merged_TSS_data[['TSS_chr','TSS_start','TSS_end','gene_id']])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#  OLD SECTION"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### RegBuild intersection with eQTL TSS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "TSS                             5476\n",
       "gene_id                         5476\n",
       "TSS_start                       5476\n",
       "TSS_end                         5476\n",
       "TSS_CTCF_binding_site            365\n",
       "TSS_TF_binding_site               57\n",
       "TSS_enhancer                     144\n",
       "TSS_open_chromatin_region         51\n",
       "TSS_promoter                    2623\n",
       "TSS_promoter_flanking_region     506\n",
       "dtype: int64"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_TSS_data.count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [],
   "source": [
    "# eQTL_TSS_bed=pybedtools.BedTool.from_dataframe(pd.DataFrame({'chr':'chr1','start':merged_var_data['TSS'].fillna(0).astype(int),'end':merged_var_data['TSS'].fillna(0).astype(int),'eQTL':merged_var_data['variant']}))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# eQTL_TSS_bed.intersect(regbuild_bed,wa=True,wb=True,loj=True).to_dataframe().groupby('itemRgb').count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [],
   "source": [
    "# names='chrom\tstart\tend\tvariant\tscore\tTSS_element_start\tTSS_element_end\tTSS_element_ID\tTSS_element_type\tTSS_element_Activity'.split()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# eQTL_TSS_regbuild=eQTL_TSS_bed.intersect(regbuild_bed,wa=True,wb=True).to_dataframe(names=names)[['variant','TSS_element_start','TSS_element_end','TSS_element_ID','TSS_element_type','TSS_element_Activity']]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merged_var_data.merge(eQTL_TSS_regbuild,how='left')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# DNASE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "metadata": {},
   "outputs": [],
   "source": [
    "dnase=pd.read_csv('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/processed_data/monocyte-ML-input/dnase_seq.bed',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [],
   "source": [
    "dnase=dnase[dnase['chr']=='chr1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chr</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>128619</td>\n",
       "      <td>128769</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>180756</td>\n",
       "      <td>180906</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181395</td>\n",
       "      <td>181415</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181415</td>\n",
       "      <td>181545</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>181545</td>\n",
       "      <td>181565</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74294</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248924802</td>\n",
       "      <td>248924945</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74295</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248924945</td>\n",
       "      <td>248924947</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74296</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248924947</td>\n",
       "      <td>248924952</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74297</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248925318</td>\n",
       "      <td>248925468</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74298</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248945546</td>\n",
       "      <td>248945696</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74299 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        chr      start        end  count\n",
       "0      chr1     128619     128769      1\n",
       "1      chr1     180756     180906      1\n",
       "2      chr1     181395     181415      3\n",
       "3      chr1     181415     181545      5\n",
       "4      chr1     181545     181565      2\n",
       "...     ...        ...        ...    ...\n",
       "74294  chr1  248924802  248924945      4\n",
       "74295  chr1  248924945  248924947      2\n",
       "74296  chr1  248924947  248924952      1\n",
       "74297  chr1  248925318  248925468      1\n",
       "74298  chr1  248945546  248945696      1\n",
       "\n",
       "[74299 rows x 4 columns]"
      ]
     },
     "execution_count": 153,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dnase"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [],
   "source": [
    "dnase_bed=pybedtools.BedTool.from_dataframe(dnase)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## DNAse-seq intersections"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " - A (DNAse not intersected with TSS for Bio reasons)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 309,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "dnase_variants=merged_var_bed.intersect(dnase_bed,wa=True,wb=True).to_dataframe(names=['chromosome','position_start','position_end','variant','var_dnase_chrom','var_dnase_start','var_dnase_end','var_dnase']).groupby('variant',as_index=False).mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_var_data=merged_var_data.merge(dnase_variants[dnase_variants.columns[[0,5]]],how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 314,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position_start</th>\n",
       "      <th>position_end</th>\n",
       "      <th>variant</th>\n",
       "      <th>var_start</th>\n",
       "      <th>var_end</th>\n",
       "      <th>var_CTCF_binding_site</th>\n",
       "      <th>var_TF_binding_site</th>\n",
       "      <th>var_enhancer</th>\n",
       "      <th>var_open_chromatin_region</th>\n",
       "      <th>var_promoter</th>\n",
       "      <th>var_promoter_flanking_region</th>\n",
       "      <th>var_dnase</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169796199</td>\n",
       "      <td>169796199</td>\n",
       "      <td>chr1_169796199_G_A</td>\n",
       "      <td>169792002</td>\n",
       "      <td>169796999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169845687</td>\n",
       "      <td>169845687</td>\n",
       "      <td>chr1_169845687_C_CTTCCA</td>\n",
       "      <td>169845687</td>\n",
       "      <td>169845687</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169821020</td>\n",
       "      <td>169821020</td>\n",
       "      <td>chr1_169821020_T_G</td>\n",
       "      <td>169821020</td>\n",
       "      <td>169821020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169982267</td>\n",
       "      <td>169982267</td>\n",
       "      <td>chr1_169982267_A_T</td>\n",
       "      <td>169982267</td>\n",
       "      <td>169982267</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169787407</td>\n",
       "      <td>169787407</td>\n",
       "      <td>chr1_169787407_G_C</td>\n",
       "      <td>169787407</td>\n",
       "      <td>169787407</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4818</th>\n",
       "      <td>chr1</td>\n",
       "      <td>233703526</td>\n",
       "      <td>233703526</td>\n",
       "      <td>chr1_233703526_T_C</td>\n",
       "      <td>233703526</td>\n",
       "      <td>233703526</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4819</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208262975</td>\n",
       "      <td>208262975</td>\n",
       "      <td>chr1_208262975_T_G</td>\n",
       "      <td>208262975</td>\n",
       "      <td>208262975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4820</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208270332</td>\n",
       "      <td>208270332</td>\n",
       "      <td>chr1_208270332_C_A</td>\n",
       "      <td>208267601</td>\n",
       "      <td>208273000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4821</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150320113</td>\n",
       "      <td>150320113</td>\n",
       "      <td>chr1_150320113_C_T</td>\n",
       "      <td>150320001</td>\n",
       "      <td>150320200</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4822</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150259332</td>\n",
       "      <td>150259332</td>\n",
       "      <td>chr1_150259332_C_T</td>\n",
       "      <td>150255202</td>\n",
       "      <td>150261999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.142857</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4823 rows Ã— 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chromosome  position_start  position_end                  variant  \\\n",
       "0          chr1       169796199     169796199       chr1_169796199_G_A   \n",
       "1          chr1       169845687     169845687  chr1_169845687_C_CTTCCA   \n",
       "2          chr1       169821020     169821020       chr1_169821020_T_G   \n",
       "3          chr1       169982267     169982267       chr1_169982267_A_T   \n",
       "4          chr1       169787407     169787407       chr1_169787407_G_C   \n",
       "...         ...             ...           ...                      ...   \n",
       "4818       chr1       233703526     233703526       chr1_233703526_T_C   \n",
       "4819       chr1       208262975     208262975       chr1_208262975_T_G   \n",
       "4820       chr1       208270332     208270332       chr1_208270332_C_A   \n",
       "4821       chr1       150320113     150320113       chr1_150320113_C_T   \n",
       "4822       chr1       150259332     150259332       chr1_150259332_C_T   \n",
       "\n",
       "      var_start    var_end  var_CTCF_binding_site  var_TF_binding_site  \\\n",
       "0     169792002  169796999                    NaN                  NaN   \n",
       "1     169845687  169845687                    NaN                  NaN   \n",
       "2     169821020  169821020                    NaN                  NaN   \n",
       "3     169982267  169982267                    NaN                  NaN   \n",
       "4     169787407  169787407                    NaN                  NaN   \n",
       "...         ...        ...                    ...                  ...   \n",
       "4818  233703526  233703526                    NaN                  NaN   \n",
       "4819  208262975  208262975                    NaN                  NaN   \n",
       "4820  208267601  208273000                    NaN                  NaN   \n",
       "4821  150320001  150320200                    2.0                  NaN   \n",
       "4822  150255202  150261999                    NaN                  NaN   \n",
       "\n",
       "      var_enhancer  var_open_chromatin_region  var_promoter  \\\n",
       "0              NaN                        NaN           4.0   \n",
       "1              NaN                        NaN           NaN   \n",
       "2              NaN                        NaN           NaN   \n",
       "3              NaN                        NaN           NaN   \n",
       "4              NaN                        NaN           NaN   \n",
       "...            ...                        ...           ...   \n",
       "4818           NaN                        NaN           NaN   \n",
       "4819           NaN                        NaN           NaN   \n",
       "4820           NaN                        NaN           NaN   \n",
       "4821           NaN                        NaN           NaN   \n",
       "4822           NaN                        NaN           4.0   \n",
       "\n",
       "      var_promoter_flanking_region  var_dnase  \n",
       "0                              NaN   3.333333  \n",
       "1                              NaN        NaN  \n",
       "2                              NaN        NaN  \n",
       "3                              NaN        NaN  \n",
       "4                              NaN        NaN  \n",
       "...                            ...        ...  \n",
       "4818                           NaN        NaN  \n",
       "4819                           NaN        NaN  \n",
       "4820                           1.0        NaN  \n",
       "4821                           NaN        NaN  \n",
       "4822                           NaN   2.142857  \n",
       "\n",
       "[4823 rows x 13 columns]"
      ]
     },
     "execution_count": 314,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_var_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# OLD SECTION"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##Â CAGE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "cage=pd.read_csv('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/processed_data/monocyte-ML-input/cage.bed',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "cage=cage[cage['chrom']=='chr1']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### RegBuild intersection with CAGE: Promoters selected"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "regbuild_cage=regbuild_bed.intersect(cage_top1000_bed,wa=True,wb=True,loj=True).to_dataframe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>name</th>\n",
       "      <th>score</th>\n",
       "      <th>strand</th>\n",
       "      <th>thickStart</th>\n",
       "      <th>thickEnd</th>\n",
       "      <th>itemRgb</th>\n",
       "      <th>blockCount</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100034201</td>\n",
       "      <td>100035200</td>\n",
       "      <td>ENSR00000253249</td>\n",
       "      <td>CTCF_binding_site</td>\n",
       "      <td>ACTIVE</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100127601</td>\n",
       "      <td>100127800</td>\n",
       "      <td>ENSR00000366775</td>\n",
       "      <td>CTCF_binding_site</td>\n",
       "      <td>INACTIVE</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100130401</td>\n",
       "      <td>100130600</td>\n",
       "      <td>ENSR00000928323</td>\n",
       "      <td>CTCF_binding_site</td>\n",
       "      <td>INACTIVE</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100135001</td>\n",
       "      <td>100135200</td>\n",
       "      <td>ENSR00000366777</td>\n",
       "      <td>CTCF_binding_site</td>\n",
       "      <td>INACTIVE</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>100147801</td>\n",
       "      <td>100148400</td>\n",
       "      <td>ENSR00000928324</td>\n",
       "      <td>CTCF_binding_site</td>\n",
       "      <td>INACTIVE</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54528</th>\n",
       "      <td>chr1</td>\n",
       "      <td>99738879</td>\n",
       "      <td>99739345</td>\n",
       "      <td>ENSR00000010583</td>\n",
       "      <td>TF_binding_site</td>\n",
       "      <td>INACTIVE</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54529</th>\n",
       "      <td>chr1</td>\n",
       "      <td>99745620</td>\n",
       "      <td>99746316</td>\n",
       "      <td>ENSR00000366709</td>\n",
       "      <td>TF_binding_site</td>\n",
       "      <td>INACTIVE</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54530</th>\n",
       "      <td>chr1</td>\n",
       "      <td>99772574</td>\n",
       "      <td>99773024</td>\n",
       "      <td>ENSR00000366717</td>\n",
       "      <td>TF_binding_site</td>\n",
       "      <td>INACTIVE</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54531</th>\n",
       "      <td>chr1</td>\n",
       "      <td>99773776</td>\n",
       "      <td>99774196</td>\n",
       "      <td>ENSR00000366718</td>\n",
       "      <td>TF_binding_site</td>\n",
       "      <td>INACTIVE</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54532</th>\n",
       "      <td>chr1</td>\n",
       "      <td>99872095</td>\n",
       "      <td>99872399</td>\n",
       "      <td>ENSR00000366741</td>\n",
       "      <td>TF_binding_site</td>\n",
       "      <td>INACTIVE</td>\n",
       "      <td>.</td>\n",
       "      <td>-1</td>\n",
       "      <td>-1</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>54533 rows Ã— 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      chrom      start        end             name              score  \\\n",
       "0      chr1  100034201  100035200  ENSR00000253249  CTCF_binding_site   \n",
       "1      chr1  100127601  100127800  ENSR00000366775  CTCF_binding_site   \n",
       "2      chr1  100130401  100130600  ENSR00000928323  CTCF_binding_site   \n",
       "3      chr1  100135001  100135200  ENSR00000366777  CTCF_binding_site   \n",
       "4      chr1  100147801  100148400  ENSR00000928324  CTCF_binding_site   \n",
       "...     ...        ...        ...              ...                ...   \n",
       "54528  chr1   99738879   99739345  ENSR00000010583    TF_binding_site   \n",
       "54529  chr1   99745620   99746316  ENSR00000366709    TF_binding_site   \n",
       "54530  chr1   99772574   99773024  ENSR00000366717    TF_binding_site   \n",
       "54531  chr1   99773776   99774196  ENSR00000366718    TF_binding_site   \n",
       "54532  chr1   99872095   99872399  ENSR00000366741    TF_binding_site   \n",
       "\n",
       "         strand thickStart  thickEnd  itemRgb blockCount  \n",
       "0        ACTIVE          .        -1       -1          .  \n",
       "1      INACTIVE          .        -1       -1          .  \n",
       "2      INACTIVE          .        -1       -1          .  \n",
       "3      INACTIVE          .        -1       -1          .  \n",
       "4      INACTIVE          .        -1       -1          .  \n",
       "...         ...        ...       ...      ...        ...  \n",
       "54528  INACTIVE          .        -1       -1          .  \n",
       "54529  INACTIVE          .        -1       -1          .  \n",
       "54530  INACTIVE          .        -1       -1          .  \n",
       "54531  INACTIVE          .        -1       -1          .  \n",
       "54532  INACTIVE          .        -1       -1          .  \n",
       "\n",
       "[54533 rows x 10 columns]"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regbuild_cage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "score\n",
       "CTCF_binding_site           15857\n",
       "TF_binding_site              2502\n",
       "enhancer                    10857\n",
       "open_chromatin_region        7997\n",
       "promoter                     4353\n",
       "promoter_flanking_region    12967\n",
       "Name: chrom, dtype: int64"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regbuild_cage.groupby('score').count()['chrom']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "score\n",
       "CTCF_binding_site             47\n",
       "TF_binding_site                1\n",
       "enhancer                       4\n",
       "open_chromatin_region          2\n",
       "promoter                    1651\n",
       "promoter_flanking_region      31\n",
       "Name: chrom, dtype: int64"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regbuild_cage[regbuild_cage['thickStart'] != '.'].groupby('score').count()['chrom']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## CAGE-Gencode intersection check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "TSS_bed=pybedtools.BedTool.from_dataframe(pd.DataFrame({'chr':'chr1','Start':TSS,'End':TSS}))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "cage_bed=pybedtools.BedTool.from_dataframe(cage)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "cage_tss=cage_bed.intersect(TSS_bed,wa=True,wb=True,loj=True).to_dataframe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### All CAGE peaks values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count     1813.000000\n",
       "mean       832.031164\n",
       "std       3636.205305\n",
       "min          2.500000\n",
       "25%         13.500000\n",
       "50%         64.500000\n",
       "75%        417.500000\n",
       "max      71749.500000\n",
       "Name: name, dtype: float64"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cage_tss['name'].describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### CAGE score when intersecting with Gencode TSS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count      982.000000\n",
       "mean      1348.031568\n",
       "std       4662.877531\n",
       "min          2.500000\n",
       "25%         51.000000\n",
       "50%        244.750000\n",
       "75%        852.500000\n",
       "max      71749.500000\n",
       "Name: name, dtype: float64"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cage_tss[cage_tss['score']!='.']['name'].describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### CAGE score when not intersecting with Gencode TSS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count      831.000000\n",
       "mean       222.268953\n",
       "std       1574.834271\n",
       "min          2.500000\n",
       "25%          5.500000\n",
       "50%         14.500000\n",
       "75%         63.750000\n",
       "max      37175.500000\n",
       "Name: name, dtype: float64"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cage_tss[cage_tss['score']=='.']['name'].describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Top 1000 CAGE peaks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "cage_top1000=cage.loc[cage['cage'].nlargest(1000).index]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "cage_top1000_bed=pybedtools.BedTool.from_dataframe(cage_top1000)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>cage</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>196</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26280046</td>\n",
       "      <td>26280156</td>\n",
       "      <td>71749.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>chr1</td>\n",
       "      <td>26317933</td>\n",
       "      <td>26317974</td>\n",
       "      <td>38776.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>563</th>\n",
       "      <td>chr1</td>\n",
       "      <td>111473890</td>\n",
       "      <td>111473992</td>\n",
       "      <td>37175.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>738</th>\n",
       "      <td>chr1</td>\n",
       "      <td>154183153</td>\n",
       "      <td>154183208</td>\n",
       "      <td>31458.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>860</th>\n",
       "      <td>chr1</td>\n",
       "      <td>161215213</td>\n",
       "      <td>161215300</td>\n",
       "      <td>29923.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>220</th>\n",
       "      <td>chr1</td>\n",
       "      <td>27409921</td>\n",
       "      <td>27409954</td>\n",
       "      <td>5.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>460</th>\n",
       "      <td>chr1</td>\n",
       "      <td>76080114</td>\n",
       "      <td>76080161</td>\n",
       "      <td>5.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>494</th>\n",
       "      <td>chr1</td>\n",
       "      <td>93136710</td>\n",
       "      <td>93136757</td>\n",
       "      <td>5.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>589</th>\n",
       "      <td>chr1</td>\n",
       "      <td>114719580</td>\n",
       "      <td>114719710</td>\n",
       "      <td>5.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>599</th>\n",
       "      <td>chr1</td>\n",
       "      <td>116540272</td>\n",
       "      <td>116540302</td>\n",
       "      <td>5.5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1000 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    chrom      start        end     cage\n",
       "196  chr1   26280046   26280156  71749.5\n",
       "200  chr1   26317933   26317974  38776.0\n",
       "563  chr1  111473890  111473992  37175.5\n",
       "738  chr1  154183153  154183208  31458.0\n",
       "860  chr1  161215213  161215300  29923.0\n",
       "..    ...        ...        ...      ...\n",
       "220  chr1   27409921   27409954      5.5\n",
       "460  chr1   76080114   76080161      5.5\n",
       "494  chr1   93136710   93136757      5.5\n",
       "589  chr1  114719580  114719710      5.5\n",
       "599  chr1  116540272  116540302      5.5\n",
       "\n",
       "[1000 rows x 4 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cage_top1000"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Histone Marks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {},
   "outputs": [],
   "source": [
    "H3K27ac=pd.read_csv('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/processed_data/monocyte-ML-input/H3K27ac-human.bed',sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {},
   "outputs": [],
   "source": [
    "H3K4me3=pd.read_csv('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/processed_data/monocyte-ML-input/H3K4me3-human.bed',sep='\\t')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>H3K27ac-human_freq</th>\n",
       "      <th>H3K27ac-human_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>778337</td>\n",
       "      <td>778364</td>\n",
       "      <td>1</td>\n",
       "      <td>118.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>778364</td>\n",
       "      <td>778778</td>\n",
       "      <td>2</td>\n",
       "      <td>97.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>778778</td>\n",
       "      <td>778801</td>\n",
       "      <td>1</td>\n",
       "      <td>77.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>778893</td>\n",
       "      <td>779077</td>\n",
       "      <td>1</td>\n",
       "      <td>38.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>779077</td>\n",
       "      <td>779297</td>\n",
       "      <td>2</td>\n",
       "      <td>28.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251898</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21177632</td>\n",
       "      <td>21178034</td>\n",
       "      <td>1</td>\n",
       "      <td>35.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251899</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21214315</td>\n",
       "      <td>21214535</td>\n",
       "      <td>1</td>\n",
       "      <td>14.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251900</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21239955</td>\n",
       "      <td>21240494</td>\n",
       "      <td>1</td>\n",
       "      <td>21.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251901</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21240788</td>\n",
       "      <td>21241322</td>\n",
       "      <td>1</td>\n",
       "      <td>17.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>251902</th>\n",
       "      <td>chrY</td>\n",
       "      <td>21421663</td>\n",
       "      <td>21422532</td>\n",
       "      <td>1</td>\n",
       "      <td>51.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>251903 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chrom     start       end  H3K27ac-human_freq  H3K27ac-human_score\n",
       "0       chr1    778337    778364                   1                118.0\n",
       "1       chr1    778364    778778                   2                 97.5\n",
       "2       chr1    778778    778801                   1                 77.0\n",
       "3       chr1    778893    779077                   1                 38.0\n",
       "4       chr1    779077    779297                   2                 28.0\n",
       "...      ...       ...       ...                 ...                  ...\n",
       "251898  chrY  21177632  21178034                   1                 35.0\n",
       "251899  chrY  21214315  21214535                   1                 14.0\n",
       "251900  chrY  21239955  21240494                   1                 21.0\n",
       "251901  chrY  21240788  21241322                   1                 17.0\n",
       "251902  chrY  21421663  21422532                   1                 51.0\n",
       "\n",
       "[251903 rows x 5 columns]"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "H3K27ac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>H3K4me3-human_freq</th>\n",
       "      <th>H3K4me3-human_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>778343</td>\n",
       "      <td>778362</td>\n",
       "      <td>1</td>\n",
       "      <td>54.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>778362</td>\n",
       "      <td>778381</td>\n",
       "      <td>2</td>\n",
       "      <td>87.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>778381</td>\n",
       "      <td>778710</td>\n",
       "      <td>3</td>\n",
       "      <td>89.333333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>778710</td>\n",
       "      <td>778882</td>\n",
       "      <td>2</td>\n",
       "      <td>87.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>778882</td>\n",
       "      <td>778952</td>\n",
       "      <td>3</td>\n",
       "      <td>79.666667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192735</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56836353</td>\n",
       "      <td>56837136</td>\n",
       "      <td>1</td>\n",
       "      <td>34.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192736</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56843141</td>\n",
       "      <td>56843432</td>\n",
       "      <td>1</td>\n",
       "      <td>18.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192737</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56843872</td>\n",
       "      <td>56844132</td>\n",
       "      <td>1</td>\n",
       "      <td>12.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192738</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56850060</td>\n",
       "      <td>56850320</td>\n",
       "      <td>1</td>\n",
       "      <td>12.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>192739</th>\n",
       "      <td>chrY</td>\n",
       "      <td>56851087</td>\n",
       "      <td>56851662</td>\n",
       "      <td>1</td>\n",
       "      <td>23.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>192740 rows Ã— 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       chrom     start       end  H3K4me3-human_freq  H3K4me3-human_score\n",
       "0       chr1    778343    778362                   1            54.000000\n",
       "1       chr1    778362    778381                   2            87.000000\n",
       "2       chr1    778381    778710                   3            89.333333\n",
       "3       chr1    778710    778882                   2            87.000000\n",
       "4       chr1    778882    778952                   3            79.666667\n",
       "...      ...       ...       ...                 ...                  ...\n",
       "192735  chrY  56836353  56837136                   1            34.000000\n",
       "192736  chrY  56843141  56843432                   1            18.000000\n",
       "192737  chrY  56843872  56844132                   1            12.000000\n",
       "192738  chrY  56850060  56850320                   1            12.000000\n",
       "192739  chrY  56851087  56851662                   1            23.000000\n",
       "\n",
       "[192740 rows x 5 columns]"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "H3K4me3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Sanity check with RegBuild"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# OLD SECTION"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 395,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "blockCount\n",
       "CTCF_binding_site            2528\n",
       "TF_binding_site                99\n",
       "enhancer                      410\n",
       "open_chromatin_region         188\n",
       "promoter                    27767\n",
       "promoter_flanking_region     5302\n",
       "Name: chrom, dtype: int64"
      ]
     },
     "execution_count": 395,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pybedtools.BedTool.from_dataframe(H3K27ac).intersect(regbuild_bed,wa=True,wb=True).to_dataframe().groupby('blockCount').count()['chrom']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 396,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "blockCount\n",
       "CTCF_binding_site            2517\n",
       "TF_binding_site                 5\n",
       "enhancer                       21\n",
       "open_chromatin_region          14\n",
       "promoter                    26423\n",
       "promoter_flanking_region     1501\n",
       "Name: chrom, dtype: int64"
      ]
     },
     "execution_count": 396,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pybedtools.BedTool.from_dataframe(H3K4me3).intersect(regbuild_bed,wa=True,wb=True).to_dataframe().groupby('blockCount').count()['chrom']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Histon marks intersection"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 280,
   "metadata": {},
   "outputs": [],
   "source": [
    "H3K27ac_bed=pybedtools.BedTool.from_dataframe(H3K27ac)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 316,
   "metadata": {},
   "outputs": [],
   "source": [
    "H3K4me3_bed=pybedtools.BedTool.from_dataframe(H3K4me3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# H3K27ac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "H3K27ac_var=merged_var_bed.intersect(H3K27ac_bed,wa=True,wb=True,loj=True).to_dataframe(names=['chr','start','end','variant','H_chr','H_start','H_end','var_H3K27ac_freq','var_H3K27ac_score']).replace('.',0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "metadata": {},
   "outputs": [],
   "source": [
    "H3K27ac_var['var_H3K27ac_freq']=H3K27ac_var['var_H3K27ac_freq'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 344,
   "metadata": {},
   "outputs": [],
   "source": [
    "H3K27ac_var=H3K27ac_var.groupby('variant',as_index=False).mean()[['variant','var_H3K27ac_freq','var_H3K27ac_score']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_var_data=merged_var_data.merge(H3K27ac_var,how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# H3K4me3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 349,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "H3K4me3_var=merged_var_bed.intersect(H3K4me3_bed,wa=True,wb=True,loj=True).to_dataframe(names=['chr','start','end','variant','H_chr','H_start','H_end','var_H3K4me3_freq','var_H3K4me3_score']).replace('.',0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 350,
   "metadata": {},
   "outputs": [],
   "source": [
    "H3K4me3_var['var_H3K4me3_freq']=H3K4me3_var['var_H3K4me3_freq'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 351,
   "metadata": {},
   "outputs": [],
   "source": [
    "H3K4me3_var=H3K4me3_var.groupby('variant',as_index=False).mean()[['variant','var_H3K4me3_freq','var_H3K4me3_score']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_var_data=merged_var_data.merge(H3K4me3_var,how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position_start</th>\n",
       "      <th>position_end</th>\n",
       "      <th>variant</th>\n",
       "      <th>var_start</th>\n",
       "      <th>var_end</th>\n",
       "      <th>var_CTCF_binding_site</th>\n",
       "      <th>var_TF_binding_site</th>\n",
       "      <th>var_enhancer</th>\n",
       "      <th>var_open_chromatin_region</th>\n",
       "      <th>var_promoter</th>\n",
       "      <th>var_promoter_flanking_region</th>\n",
       "      <th>var_dnase</th>\n",
       "      <th>var_H3K27ac_freq</th>\n",
       "      <th>var_H3K27ac_score</th>\n",
       "      <th>var_H3K4me3_freq</th>\n",
       "      <th>var_H3K4me3_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169796199</td>\n",
       "      <td>169796199</td>\n",
       "      <td>chr1_169796199_G_A</td>\n",
       "      <td>169792002</td>\n",
       "      <td>169796999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.333333</td>\n",
       "      <td>1.333333</td>\n",
       "      <td>157.5</td>\n",
       "      <td>1.8</td>\n",
       "      <td>463.066667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169845687</td>\n",
       "      <td>169845687</td>\n",
       "      <td>chr1_169845687_C_CTTCCA</td>\n",
       "      <td>169845687</td>\n",
       "      <td>169845687</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169821020</td>\n",
       "      <td>169821020</td>\n",
       "      <td>chr1_169821020_T_G</td>\n",
       "      <td>169821020</td>\n",
       "      <td>169821020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169982267</td>\n",
       "      <td>169982267</td>\n",
       "      <td>chr1_169982267_A_T</td>\n",
       "      <td>169982267</td>\n",
       "      <td>169982267</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169787407</td>\n",
       "      <td>169787407</td>\n",
       "      <td>chr1_169787407_G_C</td>\n",
       "      <td>169787407</td>\n",
       "      <td>169787407</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4818</th>\n",
       "      <td>chr1</td>\n",
       "      <td>233703526</td>\n",
       "      <td>233703526</td>\n",
       "      <td>chr1_233703526_T_C</td>\n",
       "      <td>233703526</td>\n",
       "      <td>233703526</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4819</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208262975</td>\n",
       "      <td>208262975</td>\n",
       "      <td>chr1_208262975_T_G</td>\n",
       "      <td>208262975</td>\n",
       "      <td>208262975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4820</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208270332</td>\n",
       "      <td>208270332</td>\n",
       "      <td>chr1_208270332_C_A</td>\n",
       "      <td>208267601</td>\n",
       "      <td>208273000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4821</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150320113</td>\n",
       "      <td>150320113</td>\n",
       "      <td>chr1_150320113_C_T</td>\n",
       "      <td>150320001</td>\n",
       "      <td>150320200</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4822</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150259332</td>\n",
       "      <td>150259332</td>\n",
       "      <td>chr1_150259332_C_T</td>\n",
       "      <td>150255202</td>\n",
       "      <td>150261999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.142857</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>14.0</td>\n",
       "      <td>1.8</td>\n",
       "      <td>145.466667</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4823 rows Ã— 17 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chromosome  position_start  position_end                  variant  \\\n",
       "0          chr1       169796199     169796199       chr1_169796199_G_A   \n",
       "1          chr1       169845687     169845687  chr1_169845687_C_CTTCCA   \n",
       "2          chr1       169821020     169821020       chr1_169821020_T_G   \n",
       "3          chr1       169982267     169982267       chr1_169982267_A_T   \n",
       "4          chr1       169787407     169787407       chr1_169787407_G_C   \n",
       "...         ...             ...           ...                      ...   \n",
       "4818       chr1       233703526     233703526       chr1_233703526_T_C   \n",
       "4819       chr1       208262975     208262975       chr1_208262975_T_G   \n",
       "4820       chr1       208270332     208270332       chr1_208270332_C_A   \n",
       "4821       chr1       150320113     150320113       chr1_150320113_C_T   \n",
       "4822       chr1       150259332     150259332       chr1_150259332_C_T   \n",
       "\n",
       "      var_start    var_end  var_CTCF_binding_site  var_TF_binding_site  \\\n",
       "0     169792002  169796999                    NaN                  NaN   \n",
       "1     169845687  169845687                    NaN                  NaN   \n",
       "2     169821020  169821020                    NaN                  NaN   \n",
       "3     169982267  169982267                    NaN                  NaN   \n",
       "4     169787407  169787407                    NaN                  NaN   \n",
       "...         ...        ...                    ...                  ...   \n",
       "4818  233703526  233703526                    NaN                  NaN   \n",
       "4819  208262975  208262975                    NaN                  NaN   \n",
       "4820  208267601  208273000                    NaN                  NaN   \n",
       "4821  150320001  150320200                    2.0                  NaN   \n",
       "4822  150255202  150261999                    NaN                  NaN   \n",
       "\n",
       "      var_enhancer  var_open_chromatin_region  var_promoter  \\\n",
       "0              NaN                        NaN           4.0   \n",
       "1              NaN                        NaN           NaN   \n",
       "2              NaN                        NaN           NaN   \n",
       "3              NaN                        NaN           NaN   \n",
       "4              NaN                        NaN           NaN   \n",
       "...            ...                        ...           ...   \n",
       "4818           NaN                        NaN           NaN   \n",
       "4819           NaN                        NaN           NaN   \n",
       "4820           NaN                        NaN           NaN   \n",
       "4821           NaN                        NaN           NaN   \n",
       "4822           NaN                        NaN           4.0   \n",
       "\n",
       "      var_promoter_flanking_region  var_dnase  var_H3K27ac_freq  \\\n",
       "0                              NaN   3.333333          1.333333   \n",
       "1                              NaN        NaN          0.000000   \n",
       "2                              NaN        NaN          0.000000   \n",
       "3                              NaN        NaN          0.000000   \n",
       "4                              NaN        NaN          0.000000   \n",
       "...                            ...        ...               ...   \n",
       "4818                           NaN        NaN          0.000000   \n",
       "4819                           NaN        NaN          0.000000   \n",
       "4820                           1.0        NaN          0.000000   \n",
       "4821                           NaN        NaN          0.000000   \n",
       "4822                           NaN   2.142857          1.000000   \n",
       "\n",
       "      var_H3K27ac_score  var_H3K4me3_freq  var_H3K4me3_score  \n",
       "0                 157.5               1.8         463.066667  \n",
       "1                  -1.0               0.0          -1.000000  \n",
       "2                  -1.0               0.0          -1.000000  \n",
       "3                  -1.0               0.0          -1.000000  \n",
       "4                  -1.0               0.0          -1.000000  \n",
       "...                 ...               ...                ...  \n",
       "4818               -1.0               0.0          -1.000000  \n",
       "4819               -1.0               0.0          -1.000000  \n",
       "4820               -1.0               0.0          -1.000000  \n",
       "4821               -1.0               0.0          -1.000000  \n",
       "4822               14.0               1.8         145.466667  \n",
       "\n",
       "[4823 rows x 17 columns]"
      ]
     },
     "execution_count": 353,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_var_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# HERE"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- B"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# H3K27ac"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 381,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "H3K27ac_TSS=merged_TSS_bed.intersect(H3K27ac_bed,wa=True,wb=True,loj=True).to_dataframe(names=['chr','start','end','gene_id','H_chr','H_start','H_end','TSS_H3K27ac_freq','TSS_H3K27ac_score']).replace('.',0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 382,
   "metadata": {},
   "outputs": [],
   "source": [
    "H3K27ac_TSS['TSS_H3K27ac_freq']=H3K27ac_TSS['TSS_H3K27ac_freq'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 383,
   "metadata": {},
   "outputs": [],
   "source": [
    "H3K27ac_TSS=H3K27ac_TSS.groupby('gene_id',as_index=False).mean()[['gene_id','TSS_H3K27ac_freq','TSS_H3K27ac_score']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 384,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_TSS_data=merged_TSS_data.merge(H3K27ac_TSS,how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# H3K4me3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 385,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "H3K4me3_TSS=merged_TSS_bed.intersect(H3K4me3_bed,wa=True,wb=True,loj=True).to_dataframe(names=['chr','start','end','gene_id','H_chr','H_start','H_end','TSS_H3K4me3_freq','TSS_H3K4me3_score']).replace('.',0)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 386,
   "metadata": {},
   "outputs": [],
   "source": [
    "H3K4me3_TSS['TSS_H3K4me3_freq']=H3K4me3_TSS['TSS_H3K4me3_freq'].astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 387,
   "metadata": {},
   "outputs": [],
   "source": [
    "H3K4me3_TSS=H3K4me3_TSS.groupby('gene_id',as_index=False).mean()[['gene_id','TSS_H3K4me3_freq','TSS_H3K4me3_score']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 388,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_TSS_data=merged_TSS_data.merge(H3K4me3_TSS,how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 389,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TSS_chr</th>\n",
       "      <th>TSS</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>TSS_start</th>\n",
       "      <th>TSS_end</th>\n",
       "      <th>TSS_CTCF_binding_site</th>\n",
       "      <th>TSS_TF_binding_site</th>\n",
       "      <th>TSS_enhancer</th>\n",
       "      <th>TSS_open_chromatin_region</th>\n",
       "      <th>TSS_promoter</th>\n",
       "      <th>TSS_promoter_flanking_region</th>\n",
       "      <th>TSS_H3K27ac_freq</th>\n",
       "      <th>TSS_H3K27ac_score</th>\n",
       "      <th>TSS_H3K4me3_freq</th>\n",
       "      <th>TSS_H3K4me3_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>11869</td>\n",
       "      <td>11869</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17436</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>17436</td>\n",
       "      <td>17436</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>20602</td>\n",
       "      <td>33399</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29570</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>20602</td>\n",
       "      <td>33399</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30366</td>\n",
       "      <td>ENSG00000284332</td>\n",
       "      <td>20602</td>\n",
       "      <td>33399</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5471</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248859144</td>\n",
       "      <td>ENSG00000171163</td>\n",
       "      <td>248856002</td>\n",
       "      <td>248862599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.400000</td>\n",
       "      <td>240.100000</td>\n",
       "      <td>1.916667</td>\n",
       "      <td>138.736111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5472</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248859164</td>\n",
       "      <td>ENSG00000227237</td>\n",
       "      <td>248856002</td>\n",
       "      <td>248862599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.400000</td>\n",
       "      <td>240.100000</td>\n",
       "      <td>1.916667</td>\n",
       "      <td>138.736111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5473</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248906196</td>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>248904602</td>\n",
       "      <td>248908999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.285714</td>\n",
       "      <td>132.142857</td>\n",
       "      <td>1.800000</td>\n",
       "      <td>184.516667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5474</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248912795</td>\n",
       "      <td>ENSG00000200495</td>\n",
       "      <td>248912795</td>\n",
       "      <td>248912795</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5475</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248936581</td>\n",
       "      <td>ENSG00000233084</td>\n",
       "      <td>248936577</td>\n",
       "      <td>248937136</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5476 rows Ã— 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     TSS_chr        TSS          gene_id  TSS_start    TSS_end  \\\n",
       "0       chr1      11869  ENSG00000223972      11869      11869   \n",
       "1       chr1      17436  ENSG00000278267      17436      17436   \n",
       "2       chr1      29554  ENSG00000243485      20602      33399   \n",
       "3       chr1      29570  ENSG00000227232      20602      33399   \n",
       "4       chr1      30366  ENSG00000284332      20602      33399   \n",
       "...      ...        ...              ...        ...        ...   \n",
       "5471    chr1  248859144  ENSG00000171163  248856002  248862599   \n",
       "5472    chr1  248859164  ENSG00000227237  248856002  248862599   \n",
       "5473    chr1  248906196  ENSG00000185220  248904602  248908999   \n",
       "5474    chr1  248912795  ENSG00000200495  248912795  248912795   \n",
       "5475    chr1  248936581  ENSG00000233084  248936577  248937136   \n",
       "\n",
       "      TSS_CTCF_binding_site  TSS_TF_binding_site  TSS_enhancer  \\\n",
       "0                       NaN                  NaN           NaN   \n",
       "1                       NaN                  NaN           NaN   \n",
       "2                       NaN                  NaN           NaN   \n",
       "3                       NaN                  NaN           NaN   \n",
       "4                       NaN                  NaN           NaN   \n",
       "...                     ...                  ...           ...   \n",
       "5471                    NaN                  NaN           NaN   \n",
       "5472                    NaN                  NaN           NaN   \n",
       "5473                    NaN                  NaN           NaN   \n",
       "5474                    NaN                  NaN           NaN   \n",
       "5475                    NaN                  2.0           NaN   \n",
       "\n",
       "      TSS_open_chromatin_region  TSS_promoter  TSS_promoter_flanking_region  \\\n",
       "0                           NaN           NaN                           NaN   \n",
       "1                           NaN           NaN                           NaN   \n",
       "2                           NaN           4.0                           NaN   \n",
       "3                           NaN           4.0                           NaN   \n",
       "4                           NaN           4.0                           NaN   \n",
       "...                         ...           ...                           ...   \n",
       "5471                        NaN           3.0                           NaN   \n",
       "5472                        NaN           3.0                           NaN   \n",
       "5473                        NaN           4.0                           NaN   \n",
       "5474                        NaN           NaN                           NaN   \n",
       "5475                        NaN           NaN                           NaN   \n",
       "\n",
       "      TSS_H3K27ac_freq  TSS_H3K27ac_score  TSS_H3K4me3_freq  TSS_H3K4me3_score  \n",
       "0             0.000000          -1.000000          0.000000          -1.000000  \n",
       "1             0.000000          -1.000000          0.000000          -1.000000  \n",
       "2             0.000000          -1.000000          0.000000          -1.000000  \n",
       "3             0.000000          -1.000000          0.000000          -1.000000  \n",
       "4             0.000000          -1.000000          0.000000          -1.000000  \n",
       "...                ...                ...               ...                ...  \n",
       "5471          1.400000         240.100000          1.916667         138.736111  \n",
       "5472          1.400000         240.100000          1.916667         138.736111  \n",
       "5473          1.285714         132.142857          1.800000         184.516667  \n",
       "5474          0.000000          -1.000000          0.000000          -1.000000  \n",
       "5475          0.000000          -1.000000          0.000000          -1.000000  \n",
       "\n",
       "[5476 rows x 15 columns]"
      ]
     },
     "execution_count": 389,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_TSS_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Methylation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 390,
   "metadata": {},
   "outputs": [],
   "source": [
    "meth_CHG_file='/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/original-data/monocyte_data/ENCODE/WGBS/CHG_values.bed'\n",
    "meth_CHH_file='/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/original-data/monocyte_data/ENCODE/WGBS/CHH_values.bed'\n",
    "meth_CpG_file='/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/original-data/monocyte_data/ENCODE/WGBS/CpG_values.bed'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 391,
   "metadata": {},
   "outputs": [],
   "source": [
    "CpG_bed=pybedtools.BedTool.from_dataframe(pd.read_csv(meth_CpG_file,sep='\\t',header=None))\n",
    "#names=['chr', 'start', 'end', 'M_CpG_freq', 'M_CpG_score']                                          "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 392,
   "metadata": {},
   "outputs": [],
   "source": [
    "CHG_bed=pybedtools.BedTool.from_dataframe(pd.read_csv(meth_CHG_file,sep='\\t',header=None))\n",
    "#names=['chr', 'start', 'end', 'M_CHG_freq', 'M_CHG_score'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 393,
   "metadata": {},
   "outputs": [],
   "source": [
    "CHH_bed=pybedtools.BedTool.from_dataframe(pd.read_csv(meth_CHH_file,sep='\\t',header=None))\n",
    "#names=['chr', 'start', 'end', 'M_CHH_freq', 'M_CHH_score'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# CpG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 431,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "CpG_var=merged_var_bed.intersect(CpG_bed,wa=True,wb=True).to_dataframe(names=['chr', 'start', 'end','variant','M_chr','M_start','M_end', 'var_M_CpG_freq', 'var_M_CpG_score'] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 434,
   "metadata": {},
   "outputs": [],
   "source": [
    "CpG_var_clean=CpG_var.groupby('variant',as_index=False).mean().iloc[:,[0,1,2,5,6]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 439,
   "metadata": {},
   "outputs": [],
   "source": [
    "CpG_var_clean['var_M_CpG_density']=(CpG_var_clean['var_M_CpG_score']/(CpG_var_clean['end']-CpG_var_clean['start'])).replace(np.inf,0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# CHG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 441,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "CHG_var=merged_var_bed.intersect(CHG_bed,wa=True,wb=True).to_dataframe(names=['chr', 'start', 'end','variant','M_chr','M_start','M_end', 'var_M_CHG_freq', 'var_M_CHG_score'] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 442,
   "metadata": {},
   "outputs": [],
   "source": [
    "CHG_var_clean=CHG_var.groupby('variant',as_index=False).mean().iloc[:,[0,1,2,5,6]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 443,
   "metadata": {},
   "outputs": [],
   "source": [
    "CHG_var_clean['var_M_CHG_density']=(CHG_var_clean['var_M_CHG_score']/(CHG_var_clean['end']-CHG_var_clean['start'])).replace(np.inf,0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# CHH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 444,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "CHH_var=merged_var_bed.intersect(CHH_bed,wa=True,wb=True).to_dataframe(names=['chr', 'start', 'end','variant','M_chr','M_start','M_end', 'var_M_CHH_freq', 'var_M_CHH_score'] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 445,
   "metadata": {},
   "outputs": [],
   "source": [
    "CHH_var_clean=CHH_var.groupby('variant',as_index=False).mean().iloc[:,[0,1,2,5,6]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 446,
   "metadata": {},
   "outputs": [],
   "source": [
    "CHH_var_clean['var_M_CHH_density']=(CHH_var_clean['var_M_CHH_score']/(CHH_var_clean['end']-CHH_var_clean['start'])).replace(np.inf,0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# CpG,CHG,CHH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 448,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_var_data=merged_var_data.merge(CpG_var_clean.merge(CHG_var_clean,how='outer').merge(CHH_var_clean,how='outer'),how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 449,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position_start</th>\n",
       "      <th>position_end</th>\n",
       "      <th>variant</th>\n",
       "      <th>var_start</th>\n",
       "      <th>var_end</th>\n",
       "      <th>var_CTCF_binding_site</th>\n",
       "      <th>var_TF_binding_site</th>\n",
       "      <th>var_enhancer</th>\n",
       "      <th>var_open_chromatin_region</th>\n",
       "      <th>...</th>\n",
       "      <th>end</th>\n",
       "      <th>var_M_CpG_freq</th>\n",
       "      <th>var_M_CpG_score</th>\n",
       "      <th>var_M_CpG_density</th>\n",
       "      <th>var_M_CHG_freq</th>\n",
       "      <th>var_M_CHG_score</th>\n",
       "      <th>var_M_CHG_density</th>\n",
       "      <th>var_M_CHH_freq</th>\n",
       "      <th>var_M_CHH_score</th>\n",
       "      <th>var_M_CHH_density</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169796199</td>\n",
       "      <td>169796199</td>\n",
       "      <td>chr1_169796199_G_A</td>\n",
       "      <td>169792002</td>\n",
       "      <td>169796999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>169796999.0</td>\n",
       "      <td>22.093023</td>\n",
       "      <td>50.558140</td>\n",
       "      <td>0.010118</td>\n",
       "      <td>30.250000</td>\n",
       "      <td>4.850000</td>\n",
       "      <td>0.000971</td>\n",
       "      <td>20.774194</td>\n",
       "      <td>11.967742</td>\n",
       "      <td>0.002395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169845687</td>\n",
       "      <td>169845687</td>\n",
       "      <td>chr1_169845687_C_CTTCCA</td>\n",
       "      <td>169845687</td>\n",
       "      <td>169845687</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169821020</td>\n",
       "      <td>169821020</td>\n",
       "      <td>chr1_169821020_T_G</td>\n",
       "      <td>169821020</td>\n",
       "      <td>169821020</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169982267</td>\n",
       "      <td>169982267</td>\n",
       "      <td>chr1_169982267_A_T</td>\n",
       "      <td>169982267</td>\n",
       "      <td>169982267</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>169787407</td>\n",
       "      <td>169787407</td>\n",
       "      <td>chr1_169787407_G_C</td>\n",
       "      <td>169787407</td>\n",
       "      <td>169787407</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4818</th>\n",
       "      <td>chr1</td>\n",
       "      <td>233703526</td>\n",
       "      <td>233703526</td>\n",
       "      <td>chr1_233703526_T_C</td>\n",
       "      <td>233703526</td>\n",
       "      <td>233703526</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4819</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208262975</td>\n",
       "      <td>208262975</td>\n",
       "      <td>chr1_208262975_T_G</td>\n",
       "      <td>208262975</td>\n",
       "      <td>208262975</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4820</th>\n",
       "      <td>chr1</td>\n",
       "      <td>208270332</td>\n",
       "      <td>208270332</td>\n",
       "      <td>chr1_208270332_C_A</td>\n",
       "      <td>208267601</td>\n",
       "      <td>208273000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>208273000.0</td>\n",
       "      <td>21.646465</td>\n",
       "      <td>89.101010</td>\n",
       "      <td>0.016503</td>\n",
       "      <td>33.571429</td>\n",
       "      <td>7.742857</td>\n",
       "      <td>0.001434</td>\n",
       "      <td>26.058824</td>\n",
       "      <td>8.058824</td>\n",
       "      <td>0.001493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4821</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150320113</td>\n",
       "      <td>150320113</td>\n",
       "      <td>chr1_150320113_C_T</td>\n",
       "      <td>150320001</td>\n",
       "      <td>150320200</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>150320200.0</td>\n",
       "      <td>5.000000</td>\n",
       "      <td>80.000000</td>\n",
       "      <td>0.402010</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4822</th>\n",
       "      <td>chr1</td>\n",
       "      <td>150259332</td>\n",
       "      <td>150259332</td>\n",
       "      <td>chr1_150259332_C_T</td>\n",
       "      <td>150255202</td>\n",
       "      <td>150261999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>150261999.0</td>\n",
       "      <td>19.250000</td>\n",
       "      <td>62.566667</td>\n",
       "      <td>0.009205</td>\n",
       "      <td>26.200000</td>\n",
       "      <td>6.900000</td>\n",
       "      <td>0.001015</td>\n",
       "      <td>28.038462</td>\n",
       "      <td>9.397436</td>\n",
       "      <td>0.001383</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4823 rows Ã— 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     chromosome  position_start  position_end                  variant  \\\n",
       "0          chr1       169796199     169796199       chr1_169796199_G_A   \n",
       "1          chr1       169845687     169845687  chr1_169845687_C_CTTCCA   \n",
       "2          chr1       169821020     169821020       chr1_169821020_T_G   \n",
       "3          chr1       169982267     169982267       chr1_169982267_A_T   \n",
       "4          chr1       169787407     169787407       chr1_169787407_G_C   \n",
       "...         ...             ...           ...                      ...   \n",
       "4818       chr1       233703526     233703526       chr1_233703526_T_C   \n",
       "4819       chr1       208262975     208262975       chr1_208262975_T_G   \n",
       "4820       chr1       208270332     208270332       chr1_208270332_C_A   \n",
       "4821       chr1       150320113     150320113       chr1_150320113_C_T   \n",
       "4822       chr1       150259332     150259332       chr1_150259332_C_T   \n",
       "\n",
       "      var_start    var_end  var_CTCF_binding_site  var_TF_binding_site  \\\n",
       "0     169792002  169796999                    NaN                  NaN   \n",
       "1     169845687  169845687                    NaN                  NaN   \n",
       "2     169821020  169821020                    NaN                  NaN   \n",
       "3     169982267  169982267                    NaN                  NaN   \n",
       "4     169787407  169787407                    NaN                  NaN   \n",
       "...         ...        ...                    ...                  ...   \n",
       "4818  233703526  233703526                    NaN                  NaN   \n",
       "4819  208262975  208262975                    NaN                  NaN   \n",
       "4820  208267601  208273000                    NaN                  NaN   \n",
       "4821  150320001  150320200                    2.0                  NaN   \n",
       "4822  150255202  150261999                    NaN                  NaN   \n",
       "\n",
       "      var_enhancer  var_open_chromatin_region  ...          end  \\\n",
       "0              NaN                        NaN  ...  169796999.0   \n",
       "1              NaN                        NaN  ...          NaN   \n",
       "2              NaN                        NaN  ...          NaN   \n",
       "3              NaN                        NaN  ...          NaN   \n",
       "4              NaN                        NaN  ...          NaN   \n",
       "...            ...                        ...  ...          ...   \n",
       "4818           NaN                        NaN  ...          NaN   \n",
       "4819           NaN                        NaN  ...          NaN   \n",
       "4820           NaN                        NaN  ...  208273000.0   \n",
       "4821           NaN                        NaN  ...  150320200.0   \n",
       "4822           NaN                        NaN  ...  150261999.0   \n",
       "\n",
       "      var_M_CpG_freq  var_M_CpG_score  var_M_CpG_density  var_M_CHG_freq  \\\n",
       "0          22.093023        50.558140           0.010118       30.250000   \n",
       "1                NaN              NaN                NaN             NaN   \n",
       "2                NaN              NaN                NaN             NaN   \n",
       "3                NaN              NaN                NaN             NaN   \n",
       "4                NaN              NaN                NaN             NaN   \n",
       "...              ...              ...                ...             ...   \n",
       "4818             NaN              NaN                NaN             NaN   \n",
       "4819             NaN              NaN                NaN             NaN   \n",
       "4820       21.646465        89.101010           0.016503       33.571429   \n",
       "4821        5.000000        80.000000           0.402010             NaN   \n",
       "4822       19.250000        62.566667           0.009205       26.200000   \n",
       "\n",
       "      var_M_CHG_score  var_M_CHG_density  var_M_CHH_freq  var_M_CHH_score  \\\n",
       "0            4.850000           0.000971       20.774194        11.967742   \n",
       "1                 NaN                NaN             NaN              NaN   \n",
       "2                 NaN                NaN             NaN              NaN   \n",
       "3                 NaN                NaN             NaN              NaN   \n",
       "4                 NaN                NaN             NaN              NaN   \n",
       "...               ...                ...             ...              ...   \n",
       "4818              NaN                NaN             NaN              NaN   \n",
       "4819              NaN                NaN             NaN              NaN   \n",
       "4820         7.742857           0.001434       26.058824         8.058824   \n",
       "4821              NaN                NaN             NaN              NaN   \n",
       "4822         6.900000           0.001015       28.038462         9.397436   \n",
       "\n",
       "      var_M_CHH_density  \n",
       "0              0.002395  \n",
       "1                   NaN  \n",
       "2                   NaN  \n",
       "3                   NaN  \n",
       "4                   NaN  \n",
       "...                 ...  \n",
       "4818                NaN  \n",
       "4819                NaN  \n",
       "4820           0.001493  \n",
       "4821                NaN  \n",
       "4822           0.001383  \n",
       "\n",
       "[4823 rows x 28 columns]"
      ]
     },
     "execution_count": 449,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_var_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "- B"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# CpG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 450,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "CpG_TSS=merged_TSS_bed.intersect(CpG_bed,wa=True,wb=True).to_dataframe(names=['chr', 'start', 'end','gene_id','M_chr','M_start','M_end', 'TSS_M_CpG_freq', 'TSS_M_CpG_score'] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 451,
   "metadata": {},
   "outputs": [],
   "source": [
    "CpG_TSS_clean=CpG_TSS.groupby('gene_id',as_index=False).mean().iloc[:,[0,1,2,5,6]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 452,
   "metadata": {},
   "outputs": [],
   "source": [
    "CpG_TSS_clean['TSS_M_CpG_density']=(CpG_TSS_clean['TSS_M_CpG_score']/(CpG_TSS_clean['end']-CpG_TSS_clean['start'])).replace(np.inf,0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# CHG"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 454,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "CHG_TSS=merged_TSS_bed.intersect(CHG_bed,wa=True,wb=True).to_dataframe(names=['chr', 'start', 'end','gene_id','M_chr','M_start','M_end', 'TSS_M_CHG_freq', 'TSS_M_CHG_score'] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 455,
   "metadata": {},
   "outputs": [],
   "source": [
    "CHG_TSS_clean=CHG_TSS.groupby('gene_id',as_index=False).mean().iloc[:,[0,1,2,5,6]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 456,
   "metadata": {},
   "outputs": [],
   "source": [
    "CHG_TSS_clean['TSS_M_CHG_density']=(CHG_TSS_clean['TSS_M_CHG_score']/(CHG_TSS_clean['end']-CHG_TSS_clean['start'])).replace(np.inf,0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# CHH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 457,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/homes/jhidalgo/.local/lib/python3.8/subprocess.py:849: RuntimeWarning: line buffering (buffering=1) isn't supported in binary mode, the default buffer size will be used\n",
      "  self.stderr = io.open(errread, 'rb', bufsize)\n"
     ]
    }
   ],
   "source": [
    "CHH_TSS=merged_TSS_bed.intersect(CHH_bed,wa=True,wb=True).to_dataframe(names=['chr', 'start', 'end','gene_id','M_chr','M_start','M_end', 'TSS_M_CHH_freq', 'TSS_M_CHH_score'] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 458,
   "metadata": {},
   "outputs": [],
   "source": [
    "CHH_TSS_clean=CHH_TSS.groupby('gene_id',as_index=False).mean().iloc[:,[0,1,2,5,6]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 459,
   "metadata": {},
   "outputs": [],
   "source": [
    "CHH_TSS_clean['TSS_M_CHH_density']=(CHH_TSS_clean['TSS_M_CHH_score']/(CHH_TSS_clean['end']-CHH_TSS_clean['start'])).replace(np.inf,0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# CpG,CHG,CHH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 460,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_TSS_data=merged_TSS_data.merge(CpG_TSS_clean.merge(CHG_TSS_clean,how='outer').merge(CHH_TSS_clean,how='outer'),how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 461,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>TSS_chr</th>\n",
       "      <th>TSS</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>TSS_start</th>\n",
       "      <th>TSS_end</th>\n",
       "      <th>TSS_CTCF_binding_site</th>\n",
       "      <th>TSS_TF_binding_site</th>\n",
       "      <th>TSS_enhancer</th>\n",
       "      <th>TSS_open_chromatin_region</th>\n",
       "      <th>TSS_promoter</th>\n",
       "      <th>...</th>\n",
       "      <th>end</th>\n",
       "      <th>TSS_M_CpG_freq</th>\n",
       "      <th>TSS_M_CpG_score</th>\n",
       "      <th>TSS_M_CpG_density</th>\n",
       "      <th>TSS_M_CHG_freq</th>\n",
       "      <th>TSS_M_CHG_score</th>\n",
       "      <th>TSS_M_CHG_density</th>\n",
       "      <th>TSS_M_CHH_freq</th>\n",
       "      <th>TSS_M_CHH_score</th>\n",
       "      <th>TSS_M_CHH_density</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>11869</td>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>11869</td>\n",
       "      <td>11869</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>17436</td>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>17436</td>\n",
       "      <td>17436</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29554</td>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>20602</td>\n",
       "      <td>33399</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>33399.0</td>\n",
       "      <td>7.692308</td>\n",
       "      <td>68.846154</td>\n",
       "      <td>0.005380</td>\n",
       "      <td>26.000000</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>0.000313</td>\n",
       "      <td>5.750000</td>\n",
       "      <td>36.000000</td>\n",
       "      <td>0.002813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>29570</td>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>20602</td>\n",
       "      <td>33399</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>33399.0</td>\n",
       "      <td>7.692308</td>\n",
       "      <td>68.846154</td>\n",
       "      <td>0.005380</td>\n",
       "      <td>26.000000</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>0.000313</td>\n",
       "      <td>5.750000</td>\n",
       "      <td>36.000000</td>\n",
       "      <td>0.002813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>30366</td>\n",
       "      <td>ENSG00000284332</td>\n",
       "      <td>20602</td>\n",
       "      <td>33399</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>33399.0</td>\n",
       "      <td>7.692308</td>\n",
       "      <td>68.846154</td>\n",
       "      <td>0.005380</td>\n",
       "      <td>26.000000</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>0.000313</td>\n",
       "      <td>5.750000</td>\n",
       "      <td>36.000000</td>\n",
       "      <td>0.002813</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5471</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248859144</td>\n",
       "      <td>ENSG00000171163</td>\n",
       "      <td>248856002</td>\n",
       "      <td>248862599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>248862599.0</td>\n",
       "      <td>31.000000</td>\n",
       "      <td>58.647059</td>\n",
       "      <td>0.008890</td>\n",
       "      <td>46.448980</td>\n",
       "      <td>4.224490</td>\n",
       "      <td>0.000640</td>\n",
       "      <td>38.061728</td>\n",
       "      <td>7.012346</td>\n",
       "      <td>0.001063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5472</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248859164</td>\n",
       "      <td>ENSG00000227237</td>\n",
       "      <td>248856002</td>\n",
       "      <td>248862599</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>...</td>\n",
       "      <td>248862599.0</td>\n",
       "      <td>31.000000</td>\n",
       "      <td>58.647059</td>\n",
       "      <td>0.008890</td>\n",
       "      <td>46.448980</td>\n",
       "      <td>4.224490</td>\n",
       "      <td>0.000640</td>\n",
       "      <td>38.061728</td>\n",
       "      <td>7.012346</td>\n",
       "      <td>0.001063</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5473</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248906196</td>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>248904602</td>\n",
       "      <td>248908999</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>...</td>\n",
       "      <td>248908999.0</td>\n",
       "      <td>29.149254</td>\n",
       "      <td>61.223881</td>\n",
       "      <td>0.013924</td>\n",
       "      <td>33.884615</td>\n",
       "      <td>9.961538</td>\n",
       "      <td>0.002266</td>\n",
       "      <td>35.900000</td>\n",
       "      <td>9.700000</td>\n",
       "      <td>0.002206</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5474</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248912795</td>\n",
       "      <td>ENSG00000200495</td>\n",
       "      <td>248912795</td>\n",
       "      <td>248912795</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5475</th>\n",
       "      <td>chr1</td>\n",
       "      <td>248936581</td>\n",
       "      <td>ENSG00000233084</td>\n",
       "      <td>248936577</td>\n",
       "      <td>248937136</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>248937136.0</td>\n",
       "      <td>7.206897</td>\n",
       "      <td>96.379310</td>\n",
       "      <td>0.172414</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.666667</td>\n",
       "      <td>12.000000</td>\n",
       "      <td>0.021467</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5476 rows Ã— 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     TSS_chr        TSS          gene_id  TSS_start    TSS_end  \\\n",
       "0       chr1      11869  ENSG00000223972      11869      11869   \n",
       "1       chr1      17436  ENSG00000278267      17436      17436   \n",
       "2       chr1      29554  ENSG00000243485      20602      33399   \n",
       "3       chr1      29570  ENSG00000227232      20602      33399   \n",
       "4       chr1      30366  ENSG00000284332      20602      33399   \n",
       "...      ...        ...              ...        ...        ...   \n",
       "5471    chr1  248859144  ENSG00000171163  248856002  248862599   \n",
       "5472    chr1  248859164  ENSG00000227237  248856002  248862599   \n",
       "5473    chr1  248906196  ENSG00000185220  248904602  248908999   \n",
       "5474    chr1  248912795  ENSG00000200495  248912795  248912795   \n",
       "5475    chr1  248936581  ENSG00000233084  248936577  248937136   \n",
       "\n",
       "      TSS_CTCF_binding_site  TSS_TF_binding_site  TSS_enhancer  \\\n",
       "0                       NaN                  NaN           NaN   \n",
       "1                       NaN                  NaN           NaN   \n",
       "2                       NaN                  NaN           NaN   \n",
       "3                       NaN                  NaN           NaN   \n",
       "4                       NaN                  NaN           NaN   \n",
       "...                     ...                  ...           ...   \n",
       "5471                    NaN                  NaN           NaN   \n",
       "5472                    NaN                  NaN           NaN   \n",
       "5473                    NaN                  NaN           NaN   \n",
       "5474                    NaN                  NaN           NaN   \n",
       "5475                    NaN                  2.0           NaN   \n",
       "\n",
       "      TSS_open_chromatin_region  TSS_promoter  ...          end  \\\n",
       "0                           NaN           NaN  ...          NaN   \n",
       "1                           NaN           NaN  ...          NaN   \n",
       "2                           NaN           4.0  ...      33399.0   \n",
       "3                           NaN           4.0  ...      33399.0   \n",
       "4                           NaN           4.0  ...      33399.0   \n",
       "...                         ...           ...  ...          ...   \n",
       "5471                        NaN           3.0  ...  248862599.0   \n",
       "5472                        NaN           3.0  ...  248862599.0   \n",
       "5473                        NaN           4.0  ...  248908999.0   \n",
       "5474                        NaN           NaN  ...          NaN   \n",
       "5475                        NaN           NaN  ...  248937136.0   \n",
       "\n",
       "      TSS_M_CpG_freq  TSS_M_CpG_score  TSS_M_CpG_density  TSS_M_CHG_freq  \\\n",
       "0                NaN              NaN                NaN             NaN   \n",
       "1                NaN              NaN                NaN             NaN   \n",
       "2           7.692308        68.846154           0.005380       26.000000   \n",
       "3           7.692308        68.846154           0.005380       26.000000   \n",
       "4           7.692308        68.846154           0.005380       26.000000   \n",
       "...              ...              ...                ...             ...   \n",
       "5471       31.000000        58.647059           0.008890       46.448980   \n",
       "5472       31.000000        58.647059           0.008890       46.448980   \n",
       "5473       29.149254        61.223881           0.013924       33.884615   \n",
       "5474             NaN              NaN                NaN             NaN   \n",
       "5475        7.206897        96.379310           0.172414             NaN   \n",
       "\n",
       "      TSS_M_CHG_score  TSS_M_CHG_density  TSS_M_CHH_freq  TSS_M_CHH_score  \\\n",
       "0                 NaN                NaN             NaN              NaN   \n",
       "1                 NaN                NaN             NaN              NaN   \n",
       "2            4.000000           0.000313        5.750000        36.000000   \n",
       "3            4.000000           0.000313        5.750000        36.000000   \n",
       "4            4.000000           0.000313        5.750000        36.000000   \n",
       "...               ...                ...             ...              ...   \n",
       "5471         4.224490           0.000640       38.061728         7.012346   \n",
       "5472         4.224490           0.000640       38.061728         7.012346   \n",
       "5473         9.961538           0.002266       35.900000         9.700000   \n",
       "5474              NaN                NaN             NaN              NaN   \n",
       "5475              NaN                NaN        8.666667        12.000000   \n",
       "\n",
       "      TSS_M_CHH_density  \n",
       "0                   NaN  \n",
       "1                   NaN  \n",
       "2              0.002813  \n",
       "3              0.002813  \n",
       "4              0.002813  \n",
       "...                 ...  \n",
       "5471           0.001063  \n",
       "5472           0.001063  \n",
       "5473           0.002206  \n",
       "5474                NaN  \n",
       "5475           0.021467  \n",
       "\n",
       "[5476 rows x 26 columns]"
      ]
     },
     "execution_count": 461,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged_TSS_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merging datasets"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 555,
   "metadata": {},
   "outputs": [],
   "source": [
    "chr1_merged=chr1.merge(merged_var_data,how='left').merge(merged_TSS_data,left_on='molecular_trait_id',right_on='gene_id',how='left')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 557,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>eqtl_id</th>\n",
       "      <th>Alasoo_2018.macrophage_IFNg.beta</th>\n",
       "      <th>Alasoo_2018.macrophage_IFNg+Salmonella.beta</th>\n",
       "      <th>Alasoo_2018.macrophage_naive.beta</th>\n",
       "      <th>Alasoo_2018.macrophage_Salmonella.beta</th>\n",
       "      <th>BLUEPRINT_PE.T-cell.beta</th>\n",
       "      <th>BLUEPRINT_SE.monocyte.beta</th>\n",
       "      <th>BLUEPRINT_SE.neutrophil.beta</th>\n",
       "      <th>BrainSeq.brain.beta</th>\n",
       "      <th>FUSION.adipose_naive.beta</th>\n",
       "      <th>...</th>\n",
       "      <th>end_y</th>\n",
       "      <th>TSS_M_CpG_freq</th>\n",
       "      <th>TSS_M_CpG_score</th>\n",
       "      <th>TSS_M_CpG_density</th>\n",
       "      <th>TSS_M_CHG_freq</th>\n",
       "      <th>TSS_M_CHG_score</th>\n",
       "      <th>TSS_M_CHG_density</th>\n",
       "      <th>TSS_M_CHH_freq</th>\n",
       "      <th>TSS_M_CHH_score</th>\n",
       "      <th>TSS_M_CHH_density</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_169796199_G_A.ENSG00000000457</td>\n",
       "      <td>0.014878</td>\n",
       "      <td>0.019929</td>\n",
       "      <td>0.007145</td>\n",
       "      <td>0.008409</td>\n",
       "      <td>-0.009101</td>\n",
       "      <td>-0.003149</td>\n",
       "      <td>0.012835</td>\n",
       "      <td>-0.064450</td>\n",
       "      <td>-0.124903</td>\n",
       "      <td>...</td>\n",
       "      <td>169896999.0</td>\n",
       "      <td>23.849057</td>\n",
       "      <td>66.650943</td>\n",
       "      <td>0.008131</td>\n",
       "      <td>38.194444</td>\n",
       "      <td>4.916667</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>27.355140</td>\n",
       "      <td>10.112150</td>\n",
       "      <td>0.001234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_169845687_C_CTTCCA.ENSG00000000457</td>\n",
       "      <td>0.010908</td>\n",
       "      <td>-0.003541</td>\n",
       "      <td>0.007549</td>\n",
       "      <td>0.010561</td>\n",
       "      <td>-0.016851</td>\n",
       "      <td>-0.032392</td>\n",
       "      <td>0.011638</td>\n",
       "      <td>-0.059007</td>\n",
       "      <td>-0.112610</td>\n",
       "      <td>...</td>\n",
       "      <td>169896999.0</td>\n",
       "      <td>23.849057</td>\n",
       "      <td>66.650943</td>\n",
       "      <td>0.008131</td>\n",
       "      <td>38.194444</td>\n",
       "      <td>4.916667</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>27.355140</td>\n",
       "      <td>10.112150</td>\n",
       "      <td>0.001234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1_169821020_T_G.ENSG00000000457</td>\n",
       "      <td>0.032490</td>\n",
       "      <td>0.035531</td>\n",
       "      <td>-0.010889</td>\n",
       "      <td>0.005276</td>\n",
       "      <td>-0.024435</td>\n",
       "      <td>-0.002244</td>\n",
       "      <td>0.015308</td>\n",
       "      <td>-0.017674</td>\n",
       "      <td>-0.078033</td>\n",
       "      <td>...</td>\n",
       "      <td>169896999.0</td>\n",
       "      <td>23.849057</td>\n",
       "      <td>66.650943</td>\n",
       "      <td>0.008131</td>\n",
       "      <td>38.194444</td>\n",
       "      <td>4.916667</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>27.355140</td>\n",
       "      <td>10.112150</td>\n",
       "      <td>0.001234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1_169982267_A_T.ENSG00000000457</td>\n",
       "      <td>-0.015936</td>\n",
       "      <td>-0.023199</td>\n",
       "      <td>-0.003959</td>\n",
       "      <td>-0.011975</td>\n",
       "      <td>0.006480</td>\n",
       "      <td>0.006495</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0.040951</td>\n",
       "      <td>0.102934</td>\n",
       "      <td>...</td>\n",
       "      <td>169896999.0</td>\n",
       "      <td>23.849057</td>\n",
       "      <td>66.650943</td>\n",
       "      <td>0.008131</td>\n",
       "      <td>38.194444</td>\n",
       "      <td>4.916667</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>27.355140</td>\n",
       "      <td>10.112150</td>\n",
       "      <td>0.001234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_169787407_G_C.ENSG00000000457</td>\n",
       "      <td>0.017091</td>\n",
       "      <td>0.040628</td>\n",
       "      <td>0.002270</td>\n",
       "      <td>-0.012849</td>\n",
       "      <td>-0.046222</td>\n",
       "      <td>-0.001068</td>\n",
       "      <td>0.026307</td>\n",
       "      <td>-0.033465</td>\n",
       "      <td>-0.096427</td>\n",
       "      <td>...</td>\n",
       "      <td>169896999.0</td>\n",
       "      <td>23.849057</td>\n",
       "      <td>66.650943</td>\n",
       "      <td>0.008131</td>\n",
       "      <td>38.194444</td>\n",
       "      <td>4.916667</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>27.355140</td>\n",
       "      <td>10.112150</td>\n",
       "      <td>0.001234</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5141</th>\n",
       "      <td>chr1_208262975_T_G.ENSG00000286198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.169476</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>208273000.0</td>\n",
       "      <td>21.646465</td>\n",
       "      <td>89.101010</td>\n",
       "      <td>0.016503</td>\n",
       "      <td>33.571429</td>\n",
       "      <td>7.742857</td>\n",
       "      <td>0.001434</td>\n",
       "      <td>26.058824</td>\n",
       "      <td>8.058824</td>\n",
       "      <td>0.001493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5142</th>\n",
       "      <td>chr1_208270332_C_A.ENSG00000286198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.481553</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>208273000.0</td>\n",
       "      <td>21.646465</td>\n",
       "      <td>89.101010</td>\n",
       "      <td>0.016503</td>\n",
       "      <td>33.571429</td>\n",
       "      <td>7.742857</td>\n",
       "      <td>0.001434</td>\n",
       "      <td>26.058824</td>\n",
       "      <td>8.058824</td>\n",
       "      <td>0.001493</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5143</th>\n",
       "      <td>chr1_247323434_C_T.ENSG00000286216</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.039183</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>247349999.0</td>\n",
       "      <td>19.274510</td>\n",
       "      <td>53.637255</td>\n",
       "      <td>0.019177</td>\n",
       "      <td>27.526316</td>\n",
       "      <td>5.105263</td>\n",
       "      <td>0.001825</td>\n",
       "      <td>21.777778</td>\n",
       "      <td>10.688889</td>\n",
       "      <td>0.003822</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5144</th>\n",
       "      <td>chr1_150320113_C_T.ENSG00000286219</td>\n",
       "      <td>-0.835799</td>\n",
       "      <td>-0.444700</td>\n",
       "      <td>-0.555345</td>\n",
       "      <td>-1.044910</td>\n",
       "      <td>-0.515010</td>\n",
       "      <td>-1.162060</td>\n",
       "      <td>-0.775167</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>149432199.0</td>\n",
       "      <td>14.787097</td>\n",
       "      <td>91.103226</td>\n",
       "      <td>0.002071</td>\n",
       "      <td>26.704918</td>\n",
       "      <td>10.065574</td>\n",
       "      <td>0.000229</td>\n",
       "      <td>20.393103</td>\n",
       "      <td>10.813793</td>\n",
       "      <td>0.000246</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5145</th>\n",
       "      <td>chr1_150259332_C_T.ENSG00000286219</td>\n",
       "      <td>-0.162653</td>\n",
       "      <td>-0.402694</td>\n",
       "      <td>-0.188404</td>\n",
       "      <td>-1.284650</td>\n",
       "      <td>-0.188205</td>\n",
       "      <td>-0.020301</td>\n",
       "      <td>0.033504</td>\n",
       "      <td>-0.115315</td>\n",
       "      <td>-0.343010</td>\n",
       "      <td>...</td>\n",
       "      <td>149432199.0</td>\n",
       "      <td>14.787097</td>\n",
       "      <td>91.103226</td>\n",
       "      <td>0.002071</td>\n",
       "      <td>26.704918</td>\n",
       "      <td>10.065574</td>\n",
       "      <td>0.000229</td>\n",
       "      <td>20.393103</td>\n",
       "      <td>10.813793</td>\n",
       "      <td>0.000246</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5146 rows Ã— 339 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      eqtl_id  \\\n",
       "0          chr1_169796199_G_A.ENSG00000000457   \n",
       "1     chr1_169845687_C_CTTCCA.ENSG00000000457   \n",
       "2          chr1_169821020_T_G.ENSG00000000457   \n",
       "3          chr1_169982267_A_T.ENSG00000000457   \n",
       "4          chr1_169787407_G_C.ENSG00000000457   \n",
       "...                                       ...   \n",
       "5141       chr1_208262975_T_G.ENSG00000286198   \n",
       "5142       chr1_208270332_C_A.ENSG00000286198   \n",
       "5143       chr1_247323434_C_T.ENSG00000286216   \n",
       "5144       chr1_150320113_C_T.ENSG00000286219   \n",
       "5145       chr1_150259332_C_T.ENSG00000286219   \n",
       "\n",
       "      Alasoo_2018.macrophage_IFNg.beta  \\\n",
       "0                             0.014878   \n",
       "1                             0.010908   \n",
       "2                             0.032490   \n",
       "3                            -0.015936   \n",
       "4                             0.017091   \n",
       "...                                ...   \n",
       "5141                               NaN   \n",
       "5142                               NaN   \n",
       "5143                               NaN   \n",
       "5144                         -0.835799   \n",
       "5145                         -0.162653   \n",
       "\n",
       "      Alasoo_2018.macrophage_IFNg+Salmonella.beta  \\\n",
       "0                                        0.019929   \n",
       "1                                       -0.003541   \n",
       "2                                        0.035531   \n",
       "3                                       -0.023199   \n",
       "4                                        0.040628   \n",
       "...                                           ...   \n",
       "5141                                          NaN   \n",
       "5142                                          NaN   \n",
       "5143                                          NaN   \n",
       "5144                                    -0.444700   \n",
       "5145                                    -0.402694   \n",
       "\n",
       "      Alasoo_2018.macrophage_naive.beta  \\\n",
       "0                              0.007145   \n",
       "1                              0.007549   \n",
       "2                             -0.010889   \n",
       "3                             -0.003959   \n",
       "4                              0.002270   \n",
       "...                                 ...   \n",
       "5141                                NaN   \n",
       "5142                                NaN   \n",
       "5143                                NaN   \n",
       "5144                          -0.555345   \n",
       "5145                          -0.188404   \n",
       "\n",
       "      Alasoo_2018.macrophage_Salmonella.beta  BLUEPRINT_PE.T-cell.beta  \\\n",
       "0                                   0.008409                 -0.009101   \n",
       "1                                   0.010561                 -0.016851   \n",
       "2                                   0.005276                 -0.024435   \n",
       "3                                  -0.011975                  0.006480   \n",
       "4                                  -0.012849                 -0.046222   \n",
       "...                                      ...                       ...   \n",
       "5141                                     NaN                       NaN   \n",
       "5142                                     NaN                       NaN   \n",
       "5143                                     NaN                       NaN   \n",
       "5144                               -1.044910                 -0.515010   \n",
       "5145                               -1.284650                 -0.188205   \n",
       "\n",
       "      BLUEPRINT_SE.monocyte.beta  BLUEPRINT_SE.neutrophil.beta  \\\n",
       "0                      -0.003149                      0.012835   \n",
       "1                      -0.032392                      0.011638   \n",
       "2                      -0.002244                      0.015308   \n",
       "3                       0.006495                      0.001639   \n",
       "4                      -0.001068                      0.026307   \n",
       "...                          ...                           ...   \n",
       "5141                         NaN                           NaN   \n",
       "5142                         NaN                           NaN   \n",
       "5143                         NaN                           NaN   \n",
       "5144                   -1.162060                     -0.775167   \n",
       "5145                   -0.020301                      0.033504   \n",
       "\n",
       "      BrainSeq.brain.beta  FUSION.adipose_naive.beta  ...        end_y  \\\n",
       "0               -0.064450                  -0.124903  ...  169896999.0   \n",
       "1               -0.059007                  -0.112610  ...  169896999.0   \n",
       "2               -0.017674                  -0.078033  ...  169896999.0   \n",
       "3                0.040951                   0.102934  ...  169896999.0   \n",
       "4               -0.033465                  -0.096427  ...  169896999.0   \n",
       "...                   ...                        ...  ...          ...   \n",
       "5141             0.169476                        NaN  ...  208273000.0   \n",
       "5142            -0.481553                        NaN  ...  208273000.0   \n",
       "5143            -0.039183                        NaN  ...  247349999.0   \n",
       "5144                  NaN                        NaN  ...  149432199.0   \n",
       "5145            -0.115315                  -0.343010  ...  149432199.0   \n",
       "\n",
       "      TSS_M_CpG_freq  TSS_M_CpG_score  TSS_M_CpG_density  TSS_M_CHG_freq  \\\n",
       "0          23.849057        66.650943           0.008131       38.194444   \n",
       "1          23.849057        66.650943           0.008131       38.194444   \n",
       "2          23.849057        66.650943           0.008131       38.194444   \n",
       "3          23.849057        66.650943           0.008131       38.194444   \n",
       "4          23.849057        66.650943           0.008131       38.194444   \n",
       "...              ...              ...                ...             ...   \n",
       "5141       21.646465        89.101010           0.016503       33.571429   \n",
       "5142       21.646465        89.101010           0.016503       33.571429   \n",
       "5143       19.274510        53.637255           0.019177       27.526316   \n",
       "5144       14.787097        91.103226           0.002071       26.704918   \n",
       "5145       14.787097        91.103226           0.002071       26.704918   \n",
       "\n",
       "      TSS_M_CHG_score  TSS_M_CHG_density  TSS_M_CHH_freq  TSS_M_CHH_score  \\\n",
       "0            4.916667           0.000600       27.355140        10.112150   \n",
       "1            4.916667           0.000600       27.355140        10.112150   \n",
       "2            4.916667           0.000600       27.355140        10.112150   \n",
       "3            4.916667           0.000600       27.355140        10.112150   \n",
       "4            4.916667           0.000600       27.355140        10.112150   \n",
       "...               ...                ...             ...              ...   \n",
       "5141         7.742857           0.001434       26.058824         8.058824   \n",
       "5142         7.742857           0.001434       26.058824         8.058824   \n",
       "5143         5.105263           0.001825       21.777778        10.688889   \n",
       "5144        10.065574           0.000229       20.393103        10.813793   \n",
       "5145        10.065574           0.000229       20.393103        10.813793   \n",
       "\n",
       "      TSS_M_CHH_density  \n",
       "0              0.001234  \n",
       "1              0.001234  \n",
       "2              0.001234  \n",
       "3              0.001234  \n",
       "4              0.001234  \n",
       "...                 ...  \n",
       "5141           0.001493  \n",
       "5142           0.001493  \n",
       "5143           0.003822  \n",
       "5144           0.000246  \n",
       "5145           0.000246  \n",
       "\n",
       "[5146 rows x 339 columns]"
      ]
     },
     "execution_count": 557,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chr1_merged"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# OLD Section"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### RegBuild"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 407,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merged_chr1=chr1.merge(eqtl_regbuild_pivot.add_prefix('RegBuild_'),left_on='variant',right_index=True,how='left')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### DNAse-seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 408,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merged_chr1=merged_chr1.merge(dnase_eqtl.rename(columns={'name':'DNAse-seq','thickEnd':'variant'})[['variant','DNAse-seq']],how='left')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Histone Marks"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 409,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merged_chr1=merged_chr1.merge(H3K27ac_eqtl[['name','thickEnd','itemRgb']].rename(columns={'name':'variant','thickEnd':'H3K27ac_freq','itemRgb':'H3K27ac_score'}),how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 410,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merged_chr1=merged_chr1.merge(H3K4me3_eqtl[['name','thickEnd','itemRgb']].rename(columns={'name':'variant','thickEnd':'H3K4me3_freq','itemRgb':'H3K4me3_score'}),how='left')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "####Â BS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 411,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merged_chr1=merged_chr1.merge(CpG_eqtl,how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 412,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merged_chr1=merged_chr1.merge(CHG_eqtl,how='left')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 413,
   "metadata": {},
   "outputs": [],
   "source": [
    "# merged_chr1=merged_chr1.merge(CHH_eqtl,how='left')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### HiC var-TSS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 577,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/original-data/HiC/averageHiC/chr1/Unique_positions.txt') as f:\n",
    "    lines = f.read().splitlines()\n",
    "for i in range(0, len(lines)): \n",
    "    lines[i] = int(lines[i])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 715,
   "metadata": {},
   "outputs": [],
   "source": [
    "hic_positions=pd.read_csv('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/original-data/HiC/averageHiC/chr1/Unique_positions.txt',header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 716,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-716-523390f2179f>:5: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  chr1_merged['var_hic_pos'][index]=var_hic_pos\n",
      "<ipython-input-716-523390f2179f>:6: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  chr1_merged['tss_hic_pos'][index]=tss_hic_pos\n",
      "<ipython-input-716-523390f2179f>:8: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  chr1_merged['var_hic_pos'][index]=0\n",
      "<ipython-input-716-523390f2179f>:9: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  chr1_merged['tss_hic_pos'][index]=0\n"
     ]
    }
   ],
   "source": [
    "for index in chr1_merged.index:\n",
    "    var_hic_pos=min(lines, key=lambda x:abs(x-chr1_merged['position'][index]))\n",
    "    tss_hic_pos=min(lines, key=lambda x:abs(x-chr1_merged['TSS'][index]))\n",
    "    if (abs(var_hic_pos-chr1_merged['position'][index])<5000) & (abs(tss_hic_pos-chr1_merged['TSS'][index])<5000):\n",
    "        chr1_merged['var_hic_pos'][index]=var_hic_pos\n",
    "        chr1_merged['tss_hic_pos'][index]=tss_hic_pos\n",
    "    else:\n",
    "        chr1_merged['var_hic_pos'][index]=0\n",
    "        chr1_merged['tss_hic_pos'][index]=0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 717,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>eqtl_id</th>\n",
       "      <th>Alasoo_2018.macrophage_IFNg.beta</th>\n",
       "      <th>Alasoo_2018.macrophage_IFNg+Salmonella.beta</th>\n",
       "      <th>Alasoo_2018.macrophage_naive.beta</th>\n",
       "      <th>Alasoo_2018.macrophage_Salmonella.beta</th>\n",
       "      <th>BLUEPRINT_PE.T-cell.beta</th>\n",
       "      <th>BLUEPRINT_SE.monocyte.beta</th>\n",
       "      <th>BLUEPRINT_SE.neutrophil.beta</th>\n",
       "      <th>BrainSeq.brain.beta</th>\n",
       "      <th>FUSION.adipose_naive.beta</th>\n",
       "      <th>...</th>\n",
       "      <th>TSS_M_CpG_density</th>\n",
       "      <th>TSS_M_CHG_freq</th>\n",
       "      <th>TSS_M_CHG_score</th>\n",
       "      <th>TSS_M_CHG_density</th>\n",
       "      <th>TSS_M_CHH_freq</th>\n",
       "      <th>TSS_M_CHH_score</th>\n",
       "      <th>TSS_M_CHH_density</th>\n",
       "      <th>var_hic_pos</th>\n",
       "      <th>tss_hic_pos</th>\n",
       "      <th>HiC_Contact</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_169796199_G_A.ENSG00000000457</td>\n",
       "      <td>0.014878</td>\n",
       "      <td>0.019929</td>\n",
       "      <td>0.007145</td>\n",
       "      <td>0.008409</td>\n",
       "      <td>-0.009101</td>\n",
       "      <td>-0.003149</td>\n",
       "      <td>0.012835</td>\n",
       "      <td>-0.064450</td>\n",
       "      <td>-0.124903</td>\n",
       "      <td>...</td>\n",
       "      <td>0.008131</td>\n",
       "      <td>38.194444</td>\n",
       "      <td>4.916667</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>27.355140</td>\n",
       "      <td>10.112150</td>\n",
       "      <td>0.001234</td>\n",
       "      <td>169795860</td>\n",
       "      <td>169895859</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_169845687_C_CTTCCA.ENSG00000000457</td>\n",
       "      <td>0.010908</td>\n",
       "      <td>-0.003541</td>\n",
       "      <td>0.007549</td>\n",
       "      <td>0.010561</td>\n",
       "      <td>-0.016851</td>\n",
       "      <td>-0.032392</td>\n",
       "      <td>0.011638</td>\n",
       "      <td>-0.059007</td>\n",
       "      <td>-0.112610</td>\n",
       "      <td>...</td>\n",
       "      <td>0.008131</td>\n",
       "      <td>38.194444</td>\n",
       "      <td>4.916667</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>27.355140</td>\n",
       "      <td>10.112150</td>\n",
       "      <td>0.001234</td>\n",
       "      <td>169845859</td>\n",
       "      <td>169895859</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1_169821020_T_G.ENSG00000000457</td>\n",
       "      <td>0.032490</td>\n",
       "      <td>0.035531</td>\n",
       "      <td>-0.010889</td>\n",
       "      <td>0.005276</td>\n",
       "      <td>-0.024435</td>\n",
       "      <td>-0.002244</td>\n",
       "      <td>0.015308</td>\n",
       "      <td>-0.017674</td>\n",
       "      <td>-0.078033</td>\n",
       "      <td>...</td>\n",
       "      <td>0.008131</td>\n",
       "      <td>38.194444</td>\n",
       "      <td>4.916667</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>27.355140</td>\n",
       "      <td>10.112150</td>\n",
       "      <td>0.001234</td>\n",
       "      <td>169820860</td>\n",
       "      <td>169895859</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1_169982267_A_T.ENSG00000000457</td>\n",
       "      <td>-0.015936</td>\n",
       "      <td>-0.023199</td>\n",
       "      <td>-0.003959</td>\n",
       "      <td>-0.011975</td>\n",
       "      <td>0.006480</td>\n",
       "      <td>0.006495</td>\n",
       "      <td>0.001639</td>\n",
       "      <td>0.040951</td>\n",
       "      <td>0.102934</td>\n",
       "      <td>...</td>\n",
       "      <td>0.008131</td>\n",
       "      <td>38.194444</td>\n",
       "      <td>4.916667</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>27.355140</td>\n",
       "      <td>10.112150</td>\n",
       "      <td>0.001234</td>\n",
       "      <td>169980860</td>\n",
       "      <td>169895859</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_169787407_G_C.ENSG00000000457</td>\n",
       "      <td>0.017091</td>\n",
       "      <td>0.040628</td>\n",
       "      <td>0.002270</td>\n",
       "      <td>-0.012849</td>\n",
       "      <td>-0.046222</td>\n",
       "      <td>-0.001068</td>\n",
       "      <td>0.026307</td>\n",
       "      <td>-0.033465</td>\n",
       "      <td>-0.096427</td>\n",
       "      <td>...</td>\n",
       "      <td>0.008131</td>\n",
       "      <td>38.194444</td>\n",
       "      <td>4.916667</td>\n",
       "      <td>0.000600</td>\n",
       "      <td>27.355140</td>\n",
       "      <td>10.112150</td>\n",
       "      <td>0.001234</td>\n",
       "      <td>169785860</td>\n",
       "      <td>169895859</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5141</th>\n",
       "      <td>chr1_208262975_T_G.ENSG00000286198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.169476</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.016503</td>\n",
       "      <td>33.571429</td>\n",
       "      <td>7.742857</td>\n",
       "      <td>0.001434</td>\n",
       "      <td>26.058824</td>\n",
       "      <td>8.058824</td>\n",
       "      <td>0.001493</td>\n",
       "      <td>208261656</td>\n",
       "      <td>208271655</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5142</th>\n",
       "      <td>chr1_208270332_C_A.ENSG00000286198</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.481553</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.016503</td>\n",
       "      <td>33.571429</td>\n",
       "      <td>7.742857</td>\n",
       "      <td>0.001434</td>\n",
       "      <td>26.058824</td>\n",
       "      <td>8.058824</td>\n",
       "      <td>0.001493</td>\n",
       "      <td>208271655</td>\n",
       "      <td>208271655</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5143</th>\n",
       "      <td>chr1_247323434_C_T.ENSG00000286216</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.039183</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.019177</td>\n",
       "      <td>27.526316</td>\n",
       "      <td>5.105263</td>\n",
       "      <td>0.001825</td>\n",
       "      <td>21.777778</td>\n",
       "      <td>10.688889</td>\n",
       "      <td>0.003822</td>\n",
       "      <td>247321699</td>\n",
       "      <td>247346699</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5144</th>\n",
       "      <td>chr1_150320113_C_T.ENSG00000286219</td>\n",
       "      <td>-0.835799</td>\n",
       "      <td>-0.444700</td>\n",
       "      <td>-0.555345</td>\n",
       "      <td>-1.044910</td>\n",
       "      <td>-0.515010</td>\n",
       "      <td>-1.162060</td>\n",
       "      <td>-0.775167</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002071</td>\n",
       "      <td>26.704918</td>\n",
       "      <td>10.065574</td>\n",
       "      <td>0.000229</td>\n",
       "      <td>20.393103</td>\n",
       "      <td>10.813793</td>\n",
       "      <td>0.000246</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5145</th>\n",
       "      <td>chr1_150259332_C_T.ENSG00000286219</td>\n",
       "      <td>-0.162653</td>\n",
       "      <td>-0.402694</td>\n",
       "      <td>-0.188404</td>\n",
       "      <td>-1.284650</td>\n",
       "      <td>-0.188205</td>\n",
       "      <td>-0.020301</td>\n",
       "      <td>0.033504</td>\n",
       "      <td>-0.115315</td>\n",
       "      <td>-0.343010</td>\n",
       "      <td>...</td>\n",
       "      <td>0.002071</td>\n",
       "      <td>26.704918</td>\n",
       "      <td>10.065574</td>\n",
       "      <td>0.000229</td>\n",
       "      <td>20.393103</td>\n",
       "      <td>10.813793</td>\n",
       "      <td>0.000246</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5146 rows Ã— 342 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      eqtl_id  \\\n",
       "0          chr1_169796199_G_A.ENSG00000000457   \n",
       "1     chr1_169845687_C_CTTCCA.ENSG00000000457   \n",
       "2          chr1_169821020_T_G.ENSG00000000457   \n",
       "3          chr1_169982267_A_T.ENSG00000000457   \n",
       "4          chr1_169787407_G_C.ENSG00000000457   \n",
       "...                                       ...   \n",
       "5141       chr1_208262975_T_G.ENSG00000286198   \n",
       "5142       chr1_208270332_C_A.ENSG00000286198   \n",
       "5143       chr1_247323434_C_T.ENSG00000286216   \n",
       "5144       chr1_150320113_C_T.ENSG00000286219   \n",
       "5145       chr1_150259332_C_T.ENSG00000286219   \n",
       "\n",
       "      Alasoo_2018.macrophage_IFNg.beta  \\\n",
       "0                             0.014878   \n",
       "1                             0.010908   \n",
       "2                             0.032490   \n",
       "3                            -0.015936   \n",
       "4                             0.017091   \n",
       "...                                ...   \n",
       "5141                               NaN   \n",
       "5142                               NaN   \n",
       "5143                               NaN   \n",
       "5144                         -0.835799   \n",
       "5145                         -0.162653   \n",
       "\n",
       "      Alasoo_2018.macrophage_IFNg+Salmonella.beta  \\\n",
       "0                                        0.019929   \n",
       "1                                       -0.003541   \n",
       "2                                        0.035531   \n",
       "3                                       -0.023199   \n",
       "4                                        0.040628   \n",
       "...                                           ...   \n",
       "5141                                          NaN   \n",
       "5142                                          NaN   \n",
       "5143                                          NaN   \n",
       "5144                                    -0.444700   \n",
       "5145                                    -0.402694   \n",
       "\n",
       "      Alasoo_2018.macrophage_naive.beta  \\\n",
       "0                              0.007145   \n",
       "1                              0.007549   \n",
       "2                             -0.010889   \n",
       "3                             -0.003959   \n",
       "4                              0.002270   \n",
       "...                                 ...   \n",
       "5141                                NaN   \n",
       "5142                                NaN   \n",
       "5143                                NaN   \n",
       "5144                          -0.555345   \n",
       "5145                          -0.188404   \n",
       "\n",
       "      Alasoo_2018.macrophage_Salmonella.beta  BLUEPRINT_PE.T-cell.beta  \\\n",
       "0                                   0.008409                 -0.009101   \n",
       "1                                   0.010561                 -0.016851   \n",
       "2                                   0.005276                 -0.024435   \n",
       "3                                  -0.011975                  0.006480   \n",
       "4                                  -0.012849                 -0.046222   \n",
       "...                                      ...                       ...   \n",
       "5141                                     NaN                       NaN   \n",
       "5142                                     NaN                       NaN   \n",
       "5143                                     NaN                       NaN   \n",
       "5144                               -1.044910                 -0.515010   \n",
       "5145                               -1.284650                 -0.188205   \n",
       "\n",
       "      BLUEPRINT_SE.monocyte.beta  BLUEPRINT_SE.neutrophil.beta  \\\n",
       "0                      -0.003149                      0.012835   \n",
       "1                      -0.032392                      0.011638   \n",
       "2                      -0.002244                      0.015308   \n",
       "3                       0.006495                      0.001639   \n",
       "4                      -0.001068                      0.026307   \n",
       "...                          ...                           ...   \n",
       "5141                         NaN                           NaN   \n",
       "5142                         NaN                           NaN   \n",
       "5143                         NaN                           NaN   \n",
       "5144                   -1.162060                     -0.775167   \n",
       "5145                   -0.020301                      0.033504   \n",
       "\n",
       "      BrainSeq.brain.beta  FUSION.adipose_naive.beta  ...  TSS_M_CpG_density  \\\n",
       "0               -0.064450                  -0.124903  ...           0.008131   \n",
       "1               -0.059007                  -0.112610  ...           0.008131   \n",
       "2               -0.017674                  -0.078033  ...           0.008131   \n",
       "3                0.040951                   0.102934  ...           0.008131   \n",
       "4               -0.033465                  -0.096427  ...           0.008131   \n",
       "...                   ...                        ...  ...                ...   \n",
       "5141             0.169476                        NaN  ...           0.016503   \n",
       "5142            -0.481553                        NaN  ...           0.016503   \n",
       "5143            -0.039183                        NaN  ...           0.019177   \n",
       "5144                  NaN                        NaN  ...           0.002071   \n",
       "5145            -0.115315                  -0.343010  ...           0.002071   \n",
       "\n",
       "      TSS_M_CHG_freq  TSS_M_CHG_score  TSS_M_CHG_density  TSS_M_CHH_freq  \\\n",
       "0          38.194444         4.916667           0.000600       27.355140   \n",
       "1          38.194444         4.916667           0.000600       27.355140   \n",
       "2          38.194444         4.916667           0.000600       27.355140   \n",
       "3          38.194444         4.916667           0.000600       27.355140   \n",
       "4          38.194444         4.916667           0.000600       27.355140   \n",
       "...              ...              ...                ...             ...   \n",
       "5141       33.571429         7.742857           0.001434       26.058824   \n",
       "5142       33.571429         7.742857           0.001434       26.058824   \n",
       "5143       27.526316         5.105263           0.001825       21.777778   \n",
       "5144       26.704918        10.065574           0.000229       20.393103   \n",
       "5145       26.704918        10.065574           0.000229       20.393103   \n",
       "\n",
       "      TSS_M_CHH_score  TSS_M_CHH_density  var_hic_pos  tss_hic_pos  \\\n",
       "0           10.112150           0.001234    169795860    169895859   \n",
       "1           10.112150           0.001234    169845859    169895859   \n",
       "2           10.112150           0.001234    169820860    169895859   \n",
       "3           10.112150           0.001234    169980860    169895859   \n",
       "4           10.112150           0.001234    169785860    169895859   \n",
       "...               ...                ...          ...          ...   \n",
       "5141         8.058824           0.001493    208261656    208271655   \n",
       "5142         8.058824           0.001493    208271655    208271655   \n",
       "5143        10.688889           0.003822    247321699    247346699   \n",
       "5144        10.813793           0.000246            0            0   \n",
       "5145        10.813793           0.000246            0            0   \n",
       "\n",
       "      HiC_Contact  \n",
       "0             0.0  \n",
       "1             0.0  \n",
       "2             0.0  \n",
       "3             0.0  \n",
       "4             0.0  \n",
       "...           ...  \n",
       "5141          0.0  \n",
       "5142          0.0  \n",
       "5143          0.0  \n",
       "5144          0.0  \n",
       "5145          0.0  \n",
       "\n",
       "[5146 rows x 342 columns]"
      ]
     },
     "execution_count": 717,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chr1_merged"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 606,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 718,
   "metadata": {},
   "outputs": [],
   "source": [
    "chr1_hic_pairs=list(zip(chr1_merged['var_hic_pos'],chr1_merged['tss_hic_pos']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 694,
   "metadata": {},
   "outputs": [],
   "source": [
    "test3=chr1_merged\n",
    "test3['HiC_Contact']=0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 719,
   "metadata": {},
   "outputs": [],
   "source": [
    "# pd.concat([test3['var_hic_pos'],test3['tss_hic_pos']])\n",
    "with open('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/original-data/HiC/averageHiC/chr1/required_pos.txt', 'w') as f:\n",
    "    for item in sorted(set(list(set(test3['var_hic_pos'].unique()))+list(set(test3['tss_hic_pos'].unique())))):\n",
    "        f.write(\"%s\\n\" % item)\n",
    "# .to_csv('')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 736,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(10044942, 10209943)\n",
      "(100894444, 100894445)\n",
      "(101179444, 101234445)\n",
      "(101849444, 101994445)\n",
      "(10199942, 10209943)\n",
      "(102314444, 102389445)\n",
      "(10259942, 10384943)\n",
      "(10259942, 10394944)\n",
      "(10319942, 10384943)\n",
      "(10319942, 10394944)\n",
      "(107917378, 107962379)\n",
      "(109007378, 109247379)\n",
      "(109217378, 109247379)\n",
      "(109242378, 109247379)\n",
      "(109277378, 109282379)\n",
      "(109277378, 109397379)\n",
      "(109282378, 109282379)\n",
      "(109432378, 109482379)\n",
      "(109527378, 109687379)\n",
      "(10959943, 10979944)\n",
      "(109632378, 109667379)\n",
      "(109632378, 109687379)\n",
      "(109657378, 109667379)\n",
      "(109657378, 109687379)\n",
      "(109672378, 109687379)\n",
      "(109677378, 109692379)\n",
      "(109687378, 109687379)\n",
      "(109687378, 109692379)\n",
      "(1099620, 1114621)\n",
      "(110057378, 110057379)\n",
      "(110307378, 110337379)\n",
      "(110382378, 110407379)\n",
      "(110397378, 110407379)\n",
      "(110887378, 110942379)\n",
      "(110897378, 110962379)\n",
      "(111192378, 111202379)\n",
      "(111202378, 111202379)\n",
      "(111447378, 111472379)\n",
      "(111502378, 111562379)\n",
      "(111922378, 111987379)\n",
      "(112422378, 112707379)\n",
      "(112522378, 112617379)\n",
      "(112707378, 112707379)\n",
      "(113062378, 113072379)\n",
      "(113577378, 113757379)\n",
      "(113887378, 113887379)\n",
      "(113972378, 113977379)\n",
      "(114792379, 115087380)\n",
      "(11489943, 11654944)\n",
      "(115022379, 115337380)\n",
      "(115052379, 115087380)\n",
      "(115287379, 115337380)\n",
      "(115332379, 115337380)\n",
      "(115762379, 115767380)\n",
      "(115767379, 115767380)\n",
      "(115802379, 115842380)\n",
      "(11654943, 11654944)\n",
      "(11659943, 11689944)\n",
      "(11664943, 11689944)\n",
      "(116657378, 116667379)\n",
      "(116662378, 116667379)\n",
      "(11689943, 11689944)\n",
      "(117112378, 117122379)\n",
      "(11779943, 11804944)\n",
      "(11794943, 11804944)\n",
      "(11799943, 11804944)\n",
      "(11909943, 11924944)\n",
      "(11949943, 11979944)\n",
      "(119572377, 119597378)\n",
      "(119572377, 119647378)\n",
      "(12159943, 12164944)\n",
      "(1219620, 1229621)\n",
      "(1234620, 1469621)\n",
      "(1244620, 1324621)\n",
      "(1289620, 1324621)\n",
      "(1369620, 1449621)\n",
      "(1374620, 1434621)\n",
      "(1379620, 1419621)\n",
      "(1449620, 1574621)\n",
      "(145875079, 145885074)\n",
      "(145880074, 145885074)\n",
      "(147927887, 147927888)\n",
      "(148949487, 149024442)\n",
      "(150992524, 151007525)\n",
      "(15133504, 15153505)\n",
      "(15153504, 15153505)\n",
      "(151732524, 151762525)\n",
      "(151842524, 151852525)\n",
      "(151962524, 151992525)\n",
      "(152037524, 152047525)\n",
      "(152787524, 152877525)\n",
      "(153112524, 153112525)\n",
      "(153212524, 153217525)\n",
      "(153317524, 153347525)\n",
      "(153342524, 153347525)\n",
      "(153612524, 153612525)\n",
      "(153632524, 153632525)\n",
      "(153652524, 153677525)\n",
      "(15378504, 15408505)\n",
      "(153822524, 153967525)\n",
      "(153882524, 153967525)\n",
      "(15408504, 15408505)\n",
      "(154167524, 154192525)\n",
      "(154272524, 154272525)\n",
      "(154567524, 154567525)\n",
      "(155022524, 155032525)\n",
      "(155027524, 155032525)\n",
      "(15518505, 15583506)\n",
      "(155230209, 156145210)\n",
      "(15583505, 15603506)\n",
      "(1564620, 1599621)\n",
      "(156570208, 156570209)\n",
      "(156590208, 156615209)\n",
      "(156815208, 156815209)\n",
      "(156860208, 157045209)\n",
      "(156960208, 157045209)\n",
      "(156965208, 157045209)\n",
      "(157100208, 157110209)\n",
      "(15748505, 15748506)\n",
      "(157700210, 157700211)\n",
      "(157775210, 157775211)\n",
      "(158655210, 158685211)\n",
      "(158750210, 158770211)\n",
      "(158815210, 158830211)\n",
      "(159145210, 159145211)\n",
      "(1594620, 1599621)\n",
      "(1594620, 1614621)\n",
      "(159810210, 159825211)\n",
      "(15993505, 16018506)\n",
      "(160130210, 160150211)\n",
      "(16013505, 16018506)\n",
      "(160150210, 160150211)\n",
      "(160160210, 160205211)\n",
      "(160180210, 160190211)\n",
      "(16018505, 16018506)\n",
      "(160650210, 160670211)\n",
      "(16068505, 16073506)\n",
      "(160685210, 160710211)\n",
      "(160985210, 161020211)\n",
      "(161005210, 161020211)\n",
      "(161015210, 161020211)\n",
      "(161020210, 161020211)\n",
      "(161045210, 161045211)\n",
      "(161210210, 161215211)\n",
      "(161210210, 161225211)\n",
      "(161220210, 161225211)\n",
      "(161225210, 161225211)\n",
      "(161285210, 161550211)\n",
      "(161435210, 161440211)\n",
      "(161525210, 161630211)\n",
      "(161540210, 161630211)\n",
      "(16158505, 16158506)\n",
      "(161650210, 161720211)\n",
      "(161715210, 161720211)\n",
      "(162490210, 162495211)\n",
      "(162620210, 162630211)\n",
      "(163235210, 163320211)\n",
      "(163295210, 163320211)\n",
      "(16458505, 16458506)\n",
      "(165350763, 165355764)\n",
      "(165390763, 165630764)\n",
      "(165720763, 165825764)\n",
      "(165725763, 165825764)\n",
      "(166070763, 166165764)\n",
      "(1668561, 1723562)\n",
      "(167090763, 167165764)\n",
      "(167705763, 167720764)\n",
      "(167720763, 167720764)\n",
      "(167895762, 167935763)\n",
      "(168135762, 168135763)\n",
      "(168150762, 168280763)\n",
      "(168240762, 168280763)\n",
      "(168245762, 168280763)\n",
      "(168560762, 168575763)\n",
      "(169185762, 169365763)\n",
      "(169570762, 169585763)\n",
      "(169585762, 169585763)\n",
      "(169630762, 169660763)\n",
      "(170490859, 170660860)\n",
      "(17053505, 17053506)\n",
      "(170595859, 170660860)\n",
      "(170640859, 170660860)\n",
      "(170650859, 170665860)\n",
      "(171195861, 171225862)\n",
      "(171710860, 171740861)\n",
      "(171760860, 171780861)\n",
      "(171770860, 171780861)\n",
      "(171805860, 171840861)\n",
      "(172450860, 172530861)\n",
      "(175175864, 175190865)\n",
      "(1753561, 1778562)\n",
      "(175905864, 175920865)\n",
      "(176060864, 176205865)\n",
      "(177960865, 178035866)\n",
      "(178500865, 178510866)\n",
      "(178535865, 178650866)\n",
      "(179040865, 179080866)\n",
      "(179375865, 179880866)\n",
      "(179775865, 179815866)\n",
      "(179905865, 179940866)\n",
      "(180370865, 180500866)\n",
      "(181010864, 181020865)\n",
      "(182145865, 182390866)\n",
      "(182185865, 182390866)\n",
      "(182370865, 182390866)\n",
      "(182390865, 182405866)\n",
      "(183140865, 183140866)\n",
      "(183185865, 183185866)\n",
      "(18333506, 19288507)\n",
      "(183440865, 183635866)\n",
      "(183470865, 183470866)\n",
      "(183620865, 183635866)\n",
      "(184560866, 184670867)\n",
      "(18473506, 18478507)\n",
      "(184940868, 185315869)\n",
      "(185240868, 185315869)\n",
      "(1893561, 1918562)\n",
      "(19048506, 19288507)\n",
      "(19108506, 19208507)\n",
      "(1913561, 1918562)\n",
      "(1918561, 1918562)\n",
      "(192600870, 192635871)\n",
      "(193105870, 193105871)\n",
      "(19333506, 19483507)\n",
      "(1938561, 2003562)\n",
      "(19778507, 19798508)\n",
      "(199850872, 200025873)\n",
      "(201020872, 201110873)\n",
      "(201115872, 201125873)\n",
      "(201125872, 201125873)\n",
      "(201155872, 201170873)\n",
      "(201185872, 201220873)\n",
      "(201470872, 201505873)\n",
      "(201525872, 201620873)\n",
      "(201840872, 201945873)\n",
      "(202160872, 202160873)\n",
      "(202860872, 202965873)\n",
      "(202950872, 202965873)\n",
      "(202965872, 202965873)\n",
      "(203130872, 203185873)\n",
      "(203285872, 203350873)\n",
      "(203600872, 203625873)\n",
      "(204235872, 204375873)\n",
      "(205080872, 205120873)\n",
      "(205265872, 205320873)\n",
      "(205295872, 205320873)\n",
      "(205315872, 205320873)\n",
      "(205615872, 205630873)\n",
      "(20563507, 20588508)\n",
      "(2063561, 2143562)\n",
      "(2063561, 2183562)\n",
      "(206476657, 206501668)\n",
      "(206496667, 206496668)\n",
      "(206501667, 206506668)\n",
      "(206586668, 206611648)\n",
      "(207106655, 207126656)\n",
      "(209646655, 209651656)\n",
      "(209651655, 209651656)\n",
      "(209696655, 209721656)\n",
      "(210231655, 210231656)\n",
      "(2108561, 2143562)\n",
      "(212226658, 212236659)\n",
      "(212256658, 212296659)\n",
      "(212256658, 212356659)\n",
      "(212456658, 212856659)\n",
      "(212641658, 212856659)\n",
      "(212831658, 212856659)\n",
      "(212846658, 212856659)\n",
      "(212851658, 212856659)\n",
      "(212911658, 212916659)\n",
      "(21333507, 21343508)\n",
      "(21338507, 21343508)\n",
      "(21393507, 21438508)\n",
      "(21398507, 21413508)\n",
      "(21408507, 21438508)\n",
      "(214401657, 214551658)\n",
      "(214441657, 214551658)\n",
      "(214496657, 214551658)\n",
      "(214551657, 214551658)\n",
      "(21498507, 21508508)\n",
      "(21733507, 21783508)\n",
      "(21758507, 21783508)\n",
      "(21783507, 21783508)\n",
      "(219806658, 219956659)\n",
      "(22003507, 22023508)\n",
      "(220081658, 220271659)\n",
      "(22053507, 22058508)\n",
      "(220671658, 220786659)\n",
      "(220696658, 220746659)\n",
      "(221976658, 221976659)\n",
      "(222541658, 222546659)\n",
      "(222661658, 222736659)\n",
      "(222666658, 222711659)\n",
      "(222706658, 222711659)\n",
      "(223031658, 223141659)\n",
      "(225862300, 225882301)\n",
      "(225882300, 225882301)\n",
      "(225912300, 225922301)\n",
      "(225922300, 225922301)\n",
      "(226217299, 226222300)\n",
      "(226352299, 226407300)\n",
      "(226407299, 226407300)\n",
      "(226652299, 226737300)\n",
      "(226692299, 226737300)\n",
      "(227922299, 228212300)\n",
      "(228037299, 228107300)\n",
      "(228232299, 228302300)\n",
      "(228487299, 228487300)\n",
      "(228517299, 228734254)\n",
      "(22963507, 22963508)\n",
      "(230469254, 231239255)\n",
      "(230754254, 230794255)\n",
      "(230849254, 231039255)\n",
      "(230869254, 230869255)\n",
      "(231034254, 231039255)\n",
      "(231039254, 231039255)\n",
      "(231234254, 231239255)\n",
      "(231619254, 231924255)\n",
      "(232139254, 232159255)\n",
      "(232609254, 232629255)\n",
      "(232774254, 232804255)\n",
      "(23278507, 23368508)\n",
      "(233294254, 233294255)\n",
      "(23353507, 23368508)\n",
      "(233539254, 234259255)\n",
      "(23368507, 23423508)\n",
      "(233704254, 233724255)\n",
      "(234254254, 234259255)\n",
      "(23458507, 23483508)\n",
      "(234629254, 234629255)\n",
      "(234969253, 234969254)\n",
      "(235321685, 235326686)\n",
      "(2353561, 2363562)\n",
      "(235826700, 235881701)\n",
      "(236281700, 236281701)\n",
      "(236286700, 236346701)\n",
      "(236516700, 236516701)\n",
      "(23723510, 23798511)\n",
      "(241591698, 241676699)\n",
      "(241601698, 241676699)\n",
      "(241626698, 241676699)\n",
      "(241641698, 241676699)\n",
      "(2418561, 2423562)\n",
      "(24188510, 24198511)\n",
      "(24193510, 24318511)\n",
      "(242151698, 242201699)\n",
      "(244326698, 244451699)\n",
      "(246051698, 246506699)\n",
      "(246476698, 246506699)\n",
      "(246676698, 246681699)\n",
      "(246716698, 246771699)\n",
      "(246721698, 246771699)\n",
      "(246751698, 246771699)\n",
      "(246756698, 246771699)\n",
      "(246761698, 246771699)\n",
      "(246861698, 246931699)\n",
      "(24693509, 24743510)\n",
      "(246946698, 247006699)\n",
      "(247006698, 247076699)\n",
      "(247051698, 247076699)\n",
      "(247111698, 247111699)\n",
      "(247126698, 247171699)\n",
      "(24728509, 24743510)\n",
      "(247641698, 247856699)\n",
      "(247646698, 247856699)\n",
      "(247816698, 247821699)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(247966698, 248046699)\n",
      "(24808509, 25543510)\n",
      "(248810801, 248825802)\n",
      "(2493561, 2493562)\n",
      "(25353509, 25428510)\n",
      "(26168509, 26168510)\n",
      "(26183509, 26188510)\n",
      "(26203509, 26233510)\n",
      "(26208509, 26233510)\n",
      "(26323509, 26358510)\n",
      "(26328509, 26358510)\n",
      "(26353509, 26358510)\n",
      "(26358509, 26358510)\n",
      "(27573489, 27633490)\n",
      "(27898489, 27913490)\n",
      "(27903489, 27933490)\n",
      "(27968489, 28088490)\n",
      "(28158489, 28503489)\n",
      "(28308489, 28503489)\n",
      "(29088488, 29123489)\n",
      "(29118488, 29228489)\n",
      "(29198488, 29228489)\n",
      "(29203488, 29228489)\n",
      "(30517153, 31407154)\n",
      "(31217153, 31407154)\n",
      "(31392153, 31407154)\n",
      "(31609399, 31644400)\n",
      "(31759399, 32649400)\n",
      "(31849399, 31944400)\n",
      "(32634399, 32649400)\n",
      "(32674399, 32739400)\n",
      "(32709399, 32739400)\n",
      "(32724399, 32739400)\n",
      "(32739399, 32739400)\n",
      "(3298436, 3308437)\n",
      "(32994399, 33079400)\n",
      "(33314399, 33349400)\n",
      "(33339399, 33349400)\n",
      "(33339399, 33429400)\n",
      "(34849399, 34849400)\n",
      "(34849399, 34859400)\n",
      "(3618436, 3623437)\n",
      "(37449399, 37474400)\n",
      "(37499399, 37514400)\n",
      "(37514399, 37514400)\n",
      "(37584399, 37594400)\n",
      "(37729328, 37764329)\n",
      "(37739328, 37764329)\n",
      "(37809328, 37859329)\n",
      "(37834328, 37859329)\n",
      "(37914328, 38004329)\n",
      "(37934328, 37989329)\n",
      "(37984328, 37989329)\n",
      "(38664328, 39489329)\n",
      "(38859328, 38859329)\n",
      "(3893436, 3898437)\n",
      "(39654328, 39689329)\n",
      "(39674328, 39689329)\n",
      "(39674328, 39799329)\n",
      "(39739328, 39799329)\n",
      "(39749328, 39779329)\n",
      "(39754328, 39799329)\n",
      "(39784328, 39799329)\n",
      "(39949328, 39954329)\n",
      "(40959328, 40979329)\n",
      "(40964328, 40979329)\n",
      "(40969328, 40979329)\n",
      "(41749329, 42034330)\n",
      "(41854329, 42034330)\n",
      "(41904329, 42034330)\n",
      "(43119329, 43144330)\n",
      "(43119329, 43269330)\n",
      "(43174329, 43269330)\n",
      "(43324329, 43389330)\n",
      "(43704329, 43974329)\n",
      "(43834328, 43974329)\n",
      "(43934328, 43939329)\n",
      "(43944328, 43969329)\n",
      "(44669328, 44774329)\n",
      "(44674328, 44674329)\n",
      "(44954328, 44984329)\n",
      "(45009328, 45009329)\n",
      "(45059328, 45304329)\n",
      "(45279328, 45304329)\n",
      "(46199328, 46219329)\n",
      "(46894328, 46939329)\n",
      "(47579328, 48229329)\n",
      "(47999328, 48049329)\n",
      "(48049328, 48049329)\n",
      "(48164328, 48229329)\n",
      "(51214328, 51234329)\n",
      "(51719328, 51989329)\n",
      "(51844328, 52404329)\n",
      "(51929328, 51989329)\n",
      "(52354328, 52364329)\n",
      "(53789327, 53889328)\n",
      "(53889327, 53889328)\n",
      "(54114327, 54199328)\n",
      "(54209327, 54764328)\n",
      "(54254327, 54764328)\n",
      "(54609327, 54639328)\n",
      "(54754327, 54764328)\n",
      "(54784327, 54799328)\n",
      "(54799327, 54804328)\n",
      "(54969327, 54979328)\n",
      "(55719327, 56644328)\n",
      "(55919327, 55944328)\n",
      "(57724328, 58544329)\n",
      "(58574328, 58714329)\n",
      "(58579328, 58714329)\n",
      "(58584328, 58714329)\n",
      "(58644328, 58699329)\n",
      "(59189328, 59294329)\n",
      "(59294328, 59294329)\n",
      "(5929940, 5989941)\n",
      "(5934940, 5989941)\n",
      "(61724328, 61724329)\n",
      "(6199940, 6204941)\n",
      "(62114328, 62434330)\n",
      "(62179328, 62194329)\n",
      "(62274328, 62319330)\n",
      "(6259940, 6259941)\n",
      "(62879329, 62894330)\n",
      "(64744317, 64744318)\n",
      "(6509940, 6519941)\n",
      "(65399317, 65419318)\n",
      "(65419317, 65419318)\n",
      "(65424317, 65484318)\n",
      "(65444317, 65484318)\n",
      "(66909317, 66924318)\n",
      "(67364317, 67429318)\n",
      "(67574317, 67684318)\n",
      "(67684317, 67684318)\n",
      "(68029317, 68049318)\n",
      "(68459317, 68494318)\n",
      "(70344317, 70354318)\n",
      "(70529317, 71004318)\n",
      "(70539317, 71004318)\n",
      "(70894317, 71004318)\n",
      "(73304317, 73304318)\n",
      "(75424315, 75609316)\n",
      "(75429315, 75609316)\n",
      "(75694315, 75724316)\n",
      "(76019315, 76074316)\n",
      "(76064315, 76074316)\n",
      "(77914315, 78004317)\n",
      "(84084317, 84504318)\n",
      "(84574317, 84574318)\n",
      "(84644317, 84689318)\n",
      "(85264317, 85274318)\n",
      "(85274317, 85274318)\n",
      "(86359317, 86394318)\n",
      "(86369317, 86394318)\n",
      "(864620, 874621)\n",
      "(86664317, 86704318)\n",
      "(86674317, 86704318)\n",
      "(86689317, 86694318)\n",
      "(86704317, 86704318)\n",
      "(86874317, 86914318)\n",
      "(86899317, 86914318)\n",
      "(89524441, 89524442)\n",
      "(89814441, 89819442)\n",
      "(91279443, 91404444)\n",
      "(91349443, 91404444)\n",
      "(91444443, 91499444)\n",
      "(91624443, 91904444)\n",
      "(91699443, 91904444)\n",
      "(91714443, 91904444)\n",
      "(92474443, 92484444)\n",
      "(92919443, 92959444)\n",
      "(93164443, 93344444)\n",
      "(93324443, 93344444)\n",
      "(93334443, 93344444)\n",
      "(93879444, 93879445)\n",
      "(93884444, 93909445)\n",
      "(94814444, 94819445)\n",
      "(964620, 964621)\n",
      "(964620, 979621)\n",
      "(97359444, 97994445)\n",
      "(979620, 979621)\n",
      "(9864942, 9909943)\n",
      "(989620, 999621)\n",
      "(99554444, 99644445)\n",
      "(9964942, 9994943)\n",
      "(99734444, 99764445)\n"
     ]
    }
   ],
   "source": [
    "for chunk in pd.read_csv('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/original-data/HiC/averageHiC/chr1/filtered_hic',sep=' ',names=['A','chr','B','HiC_Contact'],chunksize=10000):\n",
    "    for index in chunk.index:\n",
    "        pair=chunk['A'][index],chunk['B'][index]\n",
    "        if pair in chr1_hic_pairs:\n",
    "            print(pair)\n",
    "            test3.loc[test3[(test3['var_hic_pos']==pair[0]) & (test3['tss_hic_pos']==pair[1])].index,'HiC_Contact']=chunk['HiC_Contact'][index]\n",
    "#         chr1_merged[(chr1_merged['var_hic_pos']==pair[0]) & (chr1_merged['tss_hic_pos']==pair[1])]['HiC_Contact']=test['HiC_Contact'][index]\n",
    "# for pair in list(zip(test['A'],test['B'])):\n",
    "#         print(pair)\n",
    "#         169820860, 169895859"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 741,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    585.000000\n",
       "mean       0.029027\n",
       "std        0.046604\n",
       "min        0.000044\n",
       "25%        0.003430\n",
       "50%        0.007784\n",
       "75%        0.023231\n",
       "max        0.262740\n",
       "Name: HiC_Contact, dtype: float64"
      ]
     },
     "execution_count": 741,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test3[test3['HiC_Contact']!=0]['HiC_Contact'].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 745,
   "metadata": {},
   "outputs": [],
   "source": [
    "test3['NAs']=test3.isna().sum(axis=1)#.groupby(['Mt'])['count'].transform(max) == df['count']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 755,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecular_trait_id</th>\n",
       "      <th>NAs</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000460</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000938</td>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000001460</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000001461</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1940</th>\n",
       "      <td>ENSG00000286106</td>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1941</th>\n",
       "      <td>ENSG00000286109</td>\n",
       "      <td>266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1942</th>\n",
       "      <td>ENSG00000286198</td>\n",
       "      <td>263</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1943</th>\n",
       "      <td>ENSG00000286216</td>\n",
       "      <td>101</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1944</th>\n",
       "      <td>ENSG00000286219</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1945 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     molecular_trait_id  NAs\n",
       "0       ENSG00000000457   10\n",
       "1       ENSG00000000460   10\n",
       "2       ENSG00000000938   30\n",
       "3       ENSG00000001460   41\n",
       "4       ENSG00000001461   19\n",
       "...                 ...  ...\n",
       "1940    ENSG00000286106  126\n",
       "1941    ENSG00000286109  266\n",
       "1942    ENSG00000286198  263\n",
       "1943    ENSG00000286216  101\n",
       "1944    ENSG00000286219   49\n",
       "\n",
       "[1945 rows x 2 columns]"
      ]
     },
     "execution_count": 755,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test3.groupby(['molecular_trait_id'],sort=False,as_index=False)['NAs'].min()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 758,
   "metadata": {},
   "outputs": [],
   "source": [
    "test3['HiC_distance']=abs(test3['var_hic_pos']-test3['tss_hic_pos'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 763,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>eqtl_id</th>\n",
       "      <th>Alasoo_2018.macrophage_IFNg.beta</th>\n",
       "      <th>Alasoo_2018.macrophage_IFNg+Salmonella.beta</th>\n",
       "      <th>Alasoo_2018.macrophage_naive.beta</th>\n",
       "      <th>Alasoo_2018.macrophage_Salmonella.beta</th>\n",
       "      <th>BLUEPRINT_PE.T-cell.beta</th>\n",
       "      <th>BLUEPRINT_SE.monocyte.beta</th>\n",
       "      <th>BLUEPRINT_SE.neutrophil.beta</th>\n",
       "      <th>BrainSeq.brain.beta</th>\n",
       "      <th>FUSION.adipose_naive.beta</th>\n",
       "      <th>...</th>\n",
       "      <th>TSS_M_CHG_score</th>\n",
       "      <th>TSS_M_CHG_density</th>\n",
       "      <th>TSS_M_CHH_freq</th>\n",
       "      <th>TSS_M_CHH_score</th>\n",
       "      <th>TSS_M_CHH_density</th>\n",
       "      <th>var_hic_pos</th>\n",
       "      <th>tss_hic_pos</th>\n",
       "      <th>HiC_Contact</th>\n",
       "      <th>NAs</th>\n",
       "      <th>HiC_distance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1_169629757_G_C.ENSG00000000460</td>\n",
       "      <td>0.036219</td>\n",
       "      <td>0.056579</td>\n",
       "      <td>-0.036440</td>\n",
       "      <td>0.009718</td>\n",
       "      <td>-0.033352</td>\n",
       "      <td>-0.065001</td>\n",
       "      <td>-0.029941</td>\n",
       "      <td>-0.034172</td>\n",
       "      <td>0.027336</td>\n",
       "      <td>...</td>\n",
       "      <td>3.000000</td>\n",
       "      <td>0.001431</td>\n",
       "      <td>38.492958</td>\n",
       "      <td>5.154930</td>\n",
       "      <td>0.002458</td>\n",
       "      <td>169630762</td>\n",
       "      <td>169660763</td>\n",
       "      <td>0.007751</td>\n",
       "      <td>11</td>\n",
       "      <td>30001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>chr1_27573355_T_TGCCA.ENSG00000000938</td>\n",
       "      <td>1.316990</td>\n",
       "      <td>0.359478</td>\n",
       "      <td>1.001060</td>\n",
       "      <td>0.420240</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.098242</td>\n",
       "      <td>0.353696</td>\n",
       "      <td>...</td>\n",
       "      <td>4.367965</td>\n",
       "      <td>0.000090</td>\n",
       "      <td>36.330380</td>\n",
       "      <td>7.111392</td>\n",
       "      <td>0.000147</td>\n",
       "      <td>27573489</td>\n",
       "      <td>27633490</td>\n",
       "      <td>0.002887</td>\n",
       "      <td>30</td>\n",
       "      <td>60001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>chr1_32993202_A_G.ENSG00000004455</td>\n",
       "      <td>-0.023348</td>\n",
       "      <td>-0.016181</td>\n",
       "      <td>0.054461</td>\n",
       "      <td>0.011161</td>\n",
       "      <td>-0.012826</td>\n",
       "      <td>-0.035521</td>\n",
       "      <td>-0.074424</td>\n",
       "      <td>-0.017013</td>\n",
       "      <td>-0.039564</td>\n",
       "      <td>...</td>\n",
       "      <td>2.850000</td>\n",
       "      <td>0.001706</td>\n",
       "      <td>41.354167</td>\n",
       "      <td>3.750000</td>\n",
       "      <td>0.002244</td>\n",
       "      <td>32994399</td>\n",
       "      <td>33079400</td>\n",
       "      <td>0.004198</td>\n",
       "      <td>9</td>\n",
       "      <td>85001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>chr1_54783481_T_C.ENSG00000006555</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.016904</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.054719</td>\n",
       "      <td>-0.025372</td>\n",
       "      <td>0.062545</td>\n",
       "      <td>...</td>\n",
       "      <td>3.254545</td>\n",
       "      <td>0.000880</td>\n",
       "      <td>43.848101</td>\n",
       "      <td>4.974684</td>\n",
       "      <td>0.001345</td>\n",
       "      <td>54784327</td>\n",
       "      <td>54799328</td>\n",
       "      <td>0.009796</td>\n",
       "      <td>112</td>\n",
       "      <td>15001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>chr1_1668108_C_T.ENSG00000008128</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.099361</td>\n",
       "      <td>0.117853</td>\n",
       "      <td>0.038317</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.161910</td>\n",
       "      <td>...</td>\n",
       "      <td>5.404762</td>\n",
       "      <td>0.000932</td>\n",
       "      <td>32.813333</td>\n",
       "      <td>8.053333</td>\n",
       "      <td>0.001389</td>\n",
       "      <td>1668561</td>\n",
       "      <td>1723562</td>\n",
       "      <td>0.005600</td>\n",
       "      <td>59</td>\n",
       "      <td>55001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5084</th>\n",
       "      <td>chr1_10318819_C_T.ENSG00000284735</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.550245</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>6.205479</td>\n",
       "      <td>0.000675</td>\n",
       "      <td>27.181208</td>\n",
       "      <td>6.872483</td>\n",
       "      <td>0.000747</td>\n",
       "      <td>10319942</td>\n",
       "      <td>10384943</td>\n",
       "      <td>0.002432</td>\n",
       "      <td>104</td>\n",
       "      <td>65001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5085</th>\n",
       "      <td>chr1_10258991_G_A.ENSG00000284735</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.033415</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>6.205479</td>\n",
       "      <td>0.000675</td>\n",
       "      <td>27.181208</td>\n",
       "      <td>6.872483</td>\n",
       "      <td>0.000747</td>\n",
       "      <td>10259942</td>\n",
       "      <td>10384943</td>\n",
       "      <td>0.001900</td>\n",
       "      <td>80</td>\n",
       "      <td>125001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5112</th>\n",
       "      <td>chr1_184559445_G_A.ENSG00000285847</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.084035</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>7.750000</td>\n",
       "      <td>0.004850</td>\n",
       "      <td>25.428571</td>\n",
       "      <td>13.642857</td>\n",
       "      <td>0.008537</td>\n",
       "      <td>184560866</td>\n",
       "      <td>184670867</td>\n",
       "      <td>0.002766</td>\n",
       "      <td>146</td>\n",
       "      <td>110001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5119</th>\n",
       "      <td>chr1_39748703_G_C.ENSG00000285905</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>39749328</td>\n",
       "      <td>39779329</td>\n",
       "      <td>0.009013</td>\n",
       "      <td>275</td>\n",
       "      <td>30001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5140</th>\n",
       "      <td>chr1_233703526_T_C.ENSG00000286109</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>233704254</td>\n",
       "      <td>233724255</td>\n",
       "      <td>0.008558</td>\n",
       "      <td>279</td>\n",
       "      <td>20001</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>585 rows Ã— 344 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    eqtl_id  Alasoo_2018.macrophage_IFNg.beta  \\\n",
       "6        chr1_169629757_G_C.ENSG00000000460                          0.036219   \n",
       "11    chr1_27573355_T_TGCCA.ENSG00000000938                          1.316990   \n",
       "22        chr1_32993202_A_G.ENSG00000004455                         -0.023348   \n",
       "24        chr1_54783481_T_C.ENSG00000006555                               NaN   \n",
       "40         chr1_1668108_C_T.ENSG00000008128                               NaN   \n",
       "...                                     ...                               ...   \n",
       "5084      chr1_10318819_C_T.ENSG00000284735                               NaN   \n",
       "5085      chr1_10258991_G_A.ENSG00000284735                               NaN   \n",
       "5112     chr1_184559445_G_A.ENSG00000285847                               NaN   \n",
       "5119      chr1_39748703_G_C.ENSG00000285905                               NaN   \n",
       "5140     chr1_233703526_T_C.ENSG00000286109                               NaN   \n",
       "\n",
       "      Alasoo_2018.macrophage_IFNg+Salmonella.beta  \\\n",
       "6                                        0.056579   \n",
       "11                                       0.359478   \n",
       "22                                      -0.016181   \n",
       "24                                            NaN   \n",
       "40                                            NaN   \n",
       "...                                           ...   \n",
       "5084                                          NaN   \n",
       "5085                                          NaN   \n",
       "5112                                          NaN   \n",
       "5119                                          NaN   \n",
       "5140                                          NaN   \n",
       "\n",
       "      Alasoo_2018.macrophage_naive.beta  \\\n",
       "6                             -0.036440   \n",
       "11                             1.001060   \n",
       "22                             0.054461   \n",
       "24                                  NaN   \n",
       "40                                  NaN   \n",
       "...                                 ...   \n",
       "5084                                NaN   \n",
       "5085                                NaN   \n",
       "5112                                NaN   \n",
       "5119                                NaN   \n",
       "5140                                NaN   \n",
       "\n",
       "      Alasoo_2018.macrophage_Salmonella.beta  BLUEPRINT_PE.T-cell.beta  \\\n",
       "6                                   0.009718                 -0.033352   \n",
       "11                                  0.420240                       NaN   \n",
       "22                                  0.011161                 -0.012826   \n",
       "24                                       NaN                  0.016904   \n",
       "40                                       NaN                  0.099361   \n",
       "...                                      ...                       ...   \n",
       "5084                                     NaN                       NaN   \n",
       "5085                                     NaN                       NaN   \n",
       "5112                                     NaN                       NaN   \n",
       "5119                                     NaN                       NaN   \n",
       "5140                                     NaN                       NaN   \n",
       "\n",
       "      BLUEPRINT_SE.monocyte.beta  BLUEPRINT_SE.neutrophil.beta  \\\n",
       "6                      -0.065001                     -0.029941   \n",
       "11                           NaN                           NaN   \n",
       "22                     -0.035521                     -0.074424   \n",
       "24                           NaN                      0.054719   \n",
       "40                      0.117853                      0.038317   \n",
       "...                          ...                           ...   \n",
       "5084                         NaN                           NaN   \n",
       "5085                         NaN                           NaN   \n",
       "5112                         NaN                           NaN   \n",
       "5119                         NaN                           NaN   \n",
       "5140                         NaN                           NaN   \n",
       "\n",
       "      BrainSeq.brain.beta  FUSION.adipose_naive.beta  ...  TSS_M_CHG_score  \\\n",
       "6               -0.034172                   0.027336  ...         3.000000   \n",
       "11               0.098242                   0.353696  ...         4.367965   \n",
       "22              -0.017013                  -0.039564  ...         2.850000   \n",
       "24              -0.025372                   0.062545  ...         3.254545   \n",
       "40                    NaN                  -0.161910  ...         5.404762   \n",
       "...                   ...                        ...  ...              ...   \n",
       "5084            -0.550245                        NaN  ...         6.205479   \n",
       "5085            -0.033415                        NaN  ...         6.205479   \n",
       "5112             0.084035                        NaN  ...         7.750000   \n",
       "5119                  NaN                        NaN  ...              NaN   \n",
       "5140                  NaN                        NaN  ...              NaN   \n",
       "\n",
       "      TSS_M_CHG_density  TSS_M_CHH_freq  TSS_M_CHH_score  TSS_M_CHH_density  \\\n",
       "6              0.001431       38.492958         5.154930           0.002458   \n",
       "11             0.000090       36.330380         7.111392           0.000147   \n",
       "22             0.001706       41.354167         3.750000           0.002244   \n",
       "24             0.000880       43.848101         4.974684           0.001345   \n",
       "40             0.000932       32.813333         8.053333           0.001389   \n",
       "...                 ...             ...              ...                ...   \n",
       "5084           0.000675       27.181208         6.872483           0.000747   \n",
       "5085           0.000675       27.181208         6.872483           0.000747   \n",
       "5112           0.004850       25.428571        13.642857           0.008537   \n",
       "5119                NaN             NaN              NaN                NaN   \n",
       "5140                NaN             NaN              NaN                NaN   \n",
       "\n",
       "      var_hic_pos  tss_hic_pos  HiC_Contact  NAs  HiC_distance  \n",
       "6       169630762    169660763     0.007751   11         30001  \n",
       "11       27573489     27633490     0.002887   30         60001  \n",
       "22       32994399     33079400     0.004198    9         85001  \n",
       "24       54784327     54799328     0.009796  112         15001  \n",
       "40        1668561      1723562     0.005600   59         55001  \n",
       "...           ...          ...          ...  ...           ...  \n",
       "5084     10319942     10384943     0.002432  104         65001  \n",
       "5085     10259942     10384943     0.001900   80        125001  \n",
       "5112    184560866    184670867     0.002766  146        110001  \n",
       "5119     39749328     39779329     0.009013  275         30001  \n",
       "5140    233704254    233724255     0.008558  279         20001  \n",
       "\n",
       "[585 rows x 344 columns]"
      ]
     },
     "execution_count": 763,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "test3[test3['HiC_Contact']>0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 762,
   "metadata": {},
   "outputs": [],
   "source": [
    "test3.to_csv('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/processed_data/monocyte-ML-input/merged_datasets.csv',index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Â Pending"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 417,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_chr1=merged_chr1.groupby('eqtl_id').mean()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 418,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Alasoo_2018.macrophage_IFNg.beta                  44.228527\n",
      "Alasoo_2018.macrophage_IFNg+Salmonella.beta       46.230082\n",
      "Alasoo_2018.macrophage_naive.beta                 40.788962\n",
      "Alasoo_2018.macrophage_Salmonella.beta            43.023708\n",
      "BLUEPRINT_PE.T-cell.beta                          39.914497\n",
      "BLUEPRINT_SE.monocyte.beta                        40.963855\n",
      "BLUEPRINT_SE.neutrophil.beta                      52.312476\n",
      "BrainSeq.brain.beta                               21.162068\n",
      "FUSION.adipose_naive.beta                         27.866304\n",
      "FUSION.muscle_naive.beta                          36.494365\n",
      "GENCORD.fibroblast.beta                           35.503304\n",
      "GENCORD.LCL.beta                                  34.356782\n",
      "GENCORD.T-cell.beta                               28.118927\n",
      "GEUVADIS.LCL.beta                                 31.713953\n",
      "GTEx.adipose_subcutaneous.beta                    18.577536\n",
      "GTEx.adipose_visceral.beta                        17.975126\n",
      "GTEx.adrenal_gland.beta                           18.752429\n",
      "GTEx.artery_aorta.beta                            21.861640\n",
      "GTEx.artery_coronary.beta                         18.519238\n",
      "GTEx.artery_tibial.beta                           22.289157\n",
      "GTEx.blood.beta                                   33.598912\n",
      "GTEx.brain_amygdala.beta                          16.848037\n",
      "GTEx.brain_anterior_cingulate_cortex.beta         17.333852\n",
      "GTEx.brain_caudate.beta                           15.993004\n",
      "GTEx.brain_cerebellar_hemisphere.beta             21.453556\n",
      "GTEx.brain_cerebellum.beta                        20.637388\n",
      "GTEx.brain_cortex.beta                            16.303925\n",
      "GTEx.brain_frontal_cortex.beta                    17.936261\n",
      "GTEx.brain_hippocampus.beta                       15.390595\n",
      "GTEx.brain_hypothalamus.beta                      16.226195\n",
      "GTEx.brain_nucleus_accumbens.beta                 16.362223\n",
      "GTEx.brain_putamen.beta                           17.431014\n",
      "GTEx.brain_spinal_cord.beta                       17.508745\n",
      "GTEx.brain_substantia_nigra.beta                  17.256121\n",
      "GTEx.breast.beta                                  15.759813\n",
      "GTEx.colon_sigmoid.beta                           17.567042\n",
      "GTEx.colon_transverse.beta                        17.353284\n",
      "GTEx.esophagus_gej.beta                           18.480373\n",
      "GTEx.esophagus_mucosa.beta                        20.987175\n",
      "GTEx.esophagus_muscularis.beta                    19.957248\n",
      "GTEx.fibroblast.beta                              28.896230\n",
      "GTEx.heart_atrial_appendage.beta                  20.695686\n",
      "GTEx.heart_left_ventricle.beta                    21.861640\n",
      "GTEx.LCL.beta                                     30.781189\n",
      "GTEx.liver.beta                                   24.582200\n",
      "GTEx.lung.beta                                    16.206763\n",
      "GTEx.minor_salivary_gland.beta                    18.383210\n",
      "GTEx.muscle.beta                                  26.175670\n",
      "GTEx.nerve_tibial.beta                            17.372717\n",
      "GTEx.ovary.beta                                   20.287602\n",
      "GTEx.pancreas.beta                                27.263894\n",
      "GTEx.pituitary.beta                               16.051302\n",
      "GTEx.prostate.beta                                14.963078\n",
      "GTEx.skin_not_sun_exposed.beta                    18.538671\n",
      "GTEx.skin_sun_exposed.beta                        18.480373\n",
      "GTEx.small_intestine.beta                         16.109600\n",
      "GTEx.spleen.beta                                  20.034979\n",
      "GTEx.stomach.beta                                 17.839098\n",
      "GTEx.testis.beta                                  11.659541\n",
      "GTEx.thyroid.beta                                 16.420521\n",
      "GTEx.uterus.beta                                  20.753984\n",
      "GTEx.vagina.beta                                  17.061796\n",
      "HipSci.iPSC.beta                                  33.618344\n",
      "Lepik_2017.blood.beta                             37.971240\n",
      "Nedelec_2016.macrophage_Listeria.beta             39.933929\n",
      "Nedelec_2016.macrophage_naive.beta                38.321026\n",
      "Nedelec_2016.macrophage_Salmonella.beta           42.129810\n",
      "Quach_2016.monocyte_IAV.beta                      36.669258\n",
      "Quach_2016.monocyte_LPS.beta                      41.099883\n",
      "Quach_2016.monocyte_naive.beta                    38.184998\n",
      "Quach_2016.monocyte_Pam3CSK4.beta                 40.555771\n",
      "Quach_2016.monocyte_R848.beta                     41.818888\n",
      "ROSMAP.brain_naive.beta                           27.127866\n",
      "Schmiedel_2018.B-cell_naive.beta                  36.649825\n",
      "Schmiedel_2018.CD4_T-cell_anti-CD3-CD28.beta      40.089390\n",
      "Schmiedel_2018.CD4_T-cell_naive.beta              36.358337\n",
      "Schmiedel_2018.CD8_T-cell_anti-CD3-CD28.beta      39.448115\n",
      "Schmiedel_2018.CD8_T-cell_naive.beta              35.581034\n",
      "Schmiedel_2018.monocyte_CD16_naive.beta           39.292654\n",
      "Schmiedel_2018.monocyte_naive.beta                38.534784\n",
      "Schmiedel_2018.NK-cell_naive.beta                 35.658764\n",
      "Schmiedel_2018.Tfh_memory.beta                    35.930820\n",
      "Schmiedel_2018.Th1-17_memory.beta                 37.640886\n",
      "Schmiedel_2018.Th17_memory.beta                   37.660319\n",
      "Schmiedel_2018.Th1_memory.beta                    36.416634\n",
      "Schmiedel_2018.Th2_memory.beta                    36.572095\n",
      "Schmiedel_2018.Treg_memory.beta                   37.155072\n",
      "Schmiedel_2018.Treg_naive.beta                    35.950253\n",
      "Schwartzentruber_2018.sensory_neuron.beta         30.956082\n",
      "TwinsUK.blood.beta                                40.497474\n",
      "TwinsUK.fat.beta                                  29.382044\n",
      "TwinsUK.LCL.beta                                  37.679751\n",
      "TwinsUK.skin.beta                                 24.329576\n",
      "van_de_Bunt_2015.pancreatic_islet.beta            21.511854\n",
      "Alasoo_2018.macrophage_IFNg.pvalue                44.228527\n",
      "Alasoo_2018.macrophage_IFNg+Salmonella.pvalue     46.230082\n",
      "Alasoo_2018.macrophage_naive.pvalue               40.788962\n",
      "Alasoo_2018.macrophage_Salmonella.pvalue          43.023708\n",
      "BLUEPRINT_PE.T-cell.pvalue                        39.914497\n",
      "BLUEPRINT_SE.monocyte.pvalue                      40.963855\n",
      "BLUEPRINT_SE.neutrophil.pvalue                    52.312476\n",
      "BrainSeq.brain.pvalue                             21.162068\n",
      "FUSION.adipose_naive.pvalue                       27.866304\n",
      "FUSION.muscle_naive.pvalue                        36.494365\n",
      "GENCORD.fibroblast.pvalue                         35.503304\n",
      "GENCORD.LCL.pvalue                                34.356782\n",
      "GENCORD.T-cell.pvalue                             28.118927\n",
      "GEUVADIS.LCL.pvalue                               31.713953\n",
      "GTEx.adipose_subcutaneous.pvalue                  18.577536\n",
      "GTEx.adipose_visceral.pvalue                      17.975126\n",
      "GTEx.adrenal_gland.pvalue                         18.752429\n",
      "GTEx.artery_aorta.pvalue                          21.861640\n",
      "GTEx.artery_coronary.pvalue                       18.519238\n",
      "GTEx.artery_tibial.pvalue                         22.289157\n",
      "GTEx.blood.pvalue                                 33.598912\n",
      "GTEx.brain_amygdala.pvalue                        16.848037\n",
      "GTEx.brain_anterior_cingulate_cortex.pvalue       17.333852\n",
      "GTEx.brain_caudate.pvalue                         15.993004\n",
      "GTEx.brain_cerebellar_hemisphere.pvalue           21.453556\n",
      "GTEx.brain_cerebellum.pvalue                      20.637388\n",
      "GTEx.brain_cortex.pvalue                          16.303925\n",
      "GTEx.brain_frontal_cortex.pvalue                  17.936261\n",
      "GTEx.brain_hippocampus.pvalue                     15.390595\n",
      "GTEx.brain_hypothalamus.pvalue                    16.226195\n",
      "GTEx.brain_nucleus_accumbens.pvalue               16.362223\n",
      "GTEx.brain_putamen.pvalue                         17.431014\n",
      "GTEx.brain_spinal_cord.pvalue                     17.508745\n",
      "GTEx.brain_substantia_nigra.pvalue                17.256121\n",
      "GTEx.breast.pvalue                                15.759813\n",
      "GTEx.colon_sigmoid.pvalue                         17.567042\n",
      "GTEx.colon_transverse.pvalue                      17.353284\n",
      "GTEx.esophagus_gej.pvalue                         18.480373\n",
      "GTEx.esophagus_mucosa.pvalue                      20.987175\n",
      "GTEx.esophagus_muscularis.pvalue                  19.957248\n",
      "GTEx.fibroblast.pvalue                            28.896230\n",
      "GTEx.heart_atrial_appendage.pvalue                20.695686\n",
      "GTEx.heart_left_ventricle.pvalue                  21.861640\n",
      "GTEx.LCL.pvalue                                   30.781189\n",
      "GTEx.liver.pvalue                                 24.582200\n",
      "GTEx.lung.pvalue                                  16.206763\n",
      "GTEx.minor_salivary_gland.pvalue                  18.383210\n",
      "GTEx.muscle.pvalue                                26.175670\n",
      "GTEx.nerve_tibial.pvalue                          17.372717\n",
      "GTEx.ovary.pvalue                                 20.287602\n",
      "GTEx.pancreas.pvalue                              27.263894\n",
      "GTEx.pituitary.pvalue                             16.051302\n",
      "GTEx.prostate.pvalue                              14.963078\n",
      "GTEx.skin_not_sun_exposed.pvalue                  18.538671\n",
      "GTEx.skin_sun_exposed.pvalue                      18.480373\n",
      "GTEx.small_intestine.pvalue                       16.109600\n",
      "GTEx.spleen.pvalue                                20.034979\n",
      "GTEx.stomach.pvalue                               17.839098\n",
      "GTEx.testis.pvalue                                11.659541\n",
      "GTEx.thyroid.pvalue                               16.420521\n",
      "GTEx.uterus.pvalue                                20.753984\n",
      "GTEx.vagina.pvalue                                17.061796\n",
      "HipSci.iPSC.pvalue                                33.618344\n",
      "Lepik_2017.blood.pvalue                           37.971240\n",
      "Nedelec_2016.macrophage_Listeria.pvalue           39.933929\n",
      "Nedelec_2016.macrophage_naive.pvalue              38.321026\n",
      "Nedelec_2016.macrophage_Salmonella.pvalue         42.129810\n",
      "Quach_2016.monocyte_IAV.pvalue                    36.669258\n",
      "Quach_2016.monocyte_LPS.pvalue                    41.099883\n",
      "Quach_2016.monocyte_naive.pvalue                  38.184998\n",
      "Quach_2016.monocyte_Pam3CSK4.pvalue               40.555771\n",
      "Quach_2016.monocyte_R848.pvalue                   41.818888\n",
      "ROSMAP.brain_naive.pvalue                         27.127866\n",
      "Schmiedel_2018.B-cell_naive.pvalue                36.649825\n",
      "Schmiedel_2018.CD4_T-cell_anti-CD3-CD28.pvalue    40.089390\n",
      "Schmiedel_2018.CD4_T-cell_naive.pvalue            36.358337\n",
      "Schmiedel_2018.CD8_T-cell_anti-CD3-CD28.pvalue    39.448115\n",
      "Schmiedel_2018.CD8_T-cell_naive.pvalue            35.581034\n",
      "Schmiedel_2018.monocyte_CD16_naive.pvalue         39.292654\n",
      "Schmiedel_2018.monocyte_naive.pvalue              38.534784\n",
      "Schmiedel_2018.NK-cell_naive.pvalue               35.658764\n",
      "Schmiedel_2018.Tfh_memory.pvalue                  35.930820\n",
      "Schmiedel_2018.Th1-17_memory.pvalue               37.640886\n",
      "Schmiedel_2018.Th17_memory.pvalue                 37.660319\n",
      "Schmiedel_2018.Th1_memory.pvalue                  36.416634\n",
      "Schmiedel_2018.Th2_memory.pvalue                  36.572095\n",
      "Schmiedel_2018.Treg_memory.pvalue                 37.155072\n",
      "Schmiedel_2018.Treg_naive.pvalue                  35.950253\n",
      "Schwartzentruber_2018.sensory_neuron.pvalue       30.956082\n",
      "TwinsUK.blood.pvalue                              40.497474\n",
      "TwinsUK.fat.pvalue                                29.382044\n",
      "TwinsUK.LCL.pvalue                                37.679751\n",
      "TwinsUK.skin.pvalue                               24.329576\n",
      "van_de_Bunt_2015.pancreatic_islet.pvalue          21.511854\n",
      "Alasoo_2018.macrophage_IFNg.se                    44.228527\n",
      "Alasoo_2018.macrophage_IFNg+Salmonella.se         46.230082\n",
      "Alasoo_2018.macrophage_naive.se                   40.788962\n",
      "Alasoo_2018.macrophage_Salmonella.se              43.023708\n",
      "BLUEPRINT_PE.T-cell.se                            39.914497\n",
      "BLUEPRINT_SE.monocyte.se                          40.963855\n",
      "BLUEPRINT_SE.neutrophil.se                        52.312476\n",
      "BrainSeq.brain.se                                 21.162068\n",
      "FUSION.adipose_naive.se                           27.866304\n",
      "FUSION.muscle_naive.se                            36.494365\n",
      "GENCORD.fibroblast.se                             35.503304\n",
      "GENCORD.LCL.se                                    34.356782\n",
      "GENCORD.T-cell.se                                 28.118927\n",
      "GEUVADIS.LCL.se                                   31.713953\n",
      "GTEx.adipose_subcutaneous.se                      18.577536\n",
      "GTEx.adipose_visceral.se                          17.975126\n",
      "GTEx.adrenal_gland.se                             18.752429\n",
      "GTEx.artery_aorta.se                              21.861640\n",
      "GTEx.artery_coronary.se                           18.519238\n",
      "GTEx.artery_tibial.se                             22.289157\n",
      "GTEx.blood.se                                     33.598912\n",
      "GTEx.brain_amygdala.se                            16.848037\n",
      "GTEx.brain_anterior_cingulate_cortex.se           17.333852\n",
      "GTEx.brain_caudate.se                             15.993004\n",
      "GTEx.brain_cerebellar_hemisphere.se               21.453556\n",
      "GTEx.brain_cerebellum.se                          20.637388\n",
      "GTEx.brain_cortex.se                              16.303925\n",
      "GTEx.brain_frontal_cortex.se                      17.936261\n",
      "GTEx.brain_hippocampus.se                         15.390595\n",
      "GTEx.brain_hypothalamus.se                        16.226195\n",
      "GTEx.brain_nucleus_accumbens.se                   16.362223\n",
      "GTEx.brain_putamen.se                             17.431014\n",
      "GTEx.brain_spinal_cord.se                         17.508745\n",
      "GTEx.brain_substantia_nigra.se                    17.256121\n",
      "GTEx.breast.se                                    15.759813\n",
      "GTEx.colon_sigmoid.se                             17.567042\n",
      "GTEx.colon_transverse.se                          17.353284\n",
      "GTEx.esophagus_gej.se                             18.480373\n",
      "GTEx.esophagus_mucosa.se                          20.987175\n",
      "GTEx.esophagus_muscularis.se                      19.957248\n",
      "GTEx.fibroblast.se                                28.896230\n",
      "GTEx.heart_atrial_appendage.se                    20.695686\n",
      "GTEx.heart_left_ventricle.se                      21.861640\n",
      "GTEx.LCL.se                                       30.781189\n",
      "GTEx.liver.se                                     24.582200\n",
      "GTEx.lung.se                                      16.206763\n",
      "GTEx.minor_salivary_gland.se                      18.383210\n",
      "GTEx.muscle.se                                    26.175670\n",
      "GTEx.nerve_tibial.se                              17.372717\n",
      "GTEx.ovary.se                                     20.287602\n",
      "GTEx.pancreas.se                                  27.263894\n",
      "GTEx.pituitary.se                                 16.051302\n",
      "GTEx.prostate.se                                  14.963078\n",
      "GTEx.skin_not_sun_exposed.se                      18.538671\n",
      "GTEx.skin_sun_exposed.se                          18.480373\n",
      "GTEx.small_intestine.se                           16.109600\n",
      "GTEx.spleen.se                                    20.034979\n",
      "GTEx.stomach.se                                   17.839098\n",
      "GTEx.testis.se                                    11.659541\n",
      "GTEx.thyroid.se                                   16.420521\n",
      "GTEx.uterus.se                                    20.753984\n",
      "GTEx.vagina.se                                    17.061796\n",
      "HipSci.iPSC.se                                    33.618344\n",
      "Lepik_2017.blood.se                               37.971240\n",
      "Nedelec_2016.macrophage_Listeria.se               39.933929\n",
      "Nedelec_2016.macrophage_naive.se                  38.321026\n",
      "Nedelec_2016.macrophage_Salmonella.se             42.129810\n",
      "Quach_2016.monocyte_IAV.se                        36.669258\n",
      "Quach_2016.monocyte_LPS.se                        41.099883\n",
      "Quach_2016.monocyte_naive.se                      38.184998\n",
      "Quach_2016.monocyte_Pam3CSK4.se                   40.555771\n",
      "Quach_2016.monocyte_R848.se                       41.818888\n",
      "ROSMAP.brain_naive.se                             27.127866\n",
      "Schmiedel_2018.B-cell_naive.se                    36.649825\n",
      "Schmiedel_2018.CD4_T-cell_anti-CD3-CD28.se        40.089390\n",
      "Schmiedel_2018.CD4_T-cell_naive.se                36.358337\n",
      "Schmiedel_2018.CD8_T-cell_anti-CD3-CD28.se        39.448115\n",
      "Schmiedel_2018.CD8_T-cell_naive.se                35.581034\n",
      "Schmiedel_2018.monocyte_CD16_naive.se             39.292654\n",
      "Schmiedel_2018.monocyte_naive.se                  38.534784\n",
      "Schmiedel_2018.NK-cell_naive.se                   35.658764\n",
      "Schmiedel_2018.Tfh_memory.se                      35.930820\n",
      "Schmiedel_2018.Th1-17_memory.se                   37.640886\n",
      "Schmiedel_2018.Th17_memory.se                     37.660319\n",
      "Schmiedel_2018.Th1_memory.se                      36.416634\n",
      "Schmiedel_2018.Th2_memory.se                      36.572095\n",
      "Schmiedel_2018.Treg_memory.se                     37.155072\n",
      "Schmiedel_2018.Treg_naive.se                      35.950253\n",
      "Schwartzentruber_2018.sensory_neuron.se           30.956082\n",
      "TwinsUK.blood.se                                  40.497474\n",
      "TwinsUK.fat.se                                    29.382044\n",
      "TwinsUK.LCL.se                                    37.679751\n",
      "TwinsUK.skin.se                                   24.329576\n",
      "van_de_Bunt_2015.pancreatic_islet.se              21.511854\n",
      "position                                           0.000000\n",
      "RegBuild_CTCF_binding_site                        93.528955\n",
      "RegBuild_TF_binding_site                          99.300428\n",
      "RegBuild_enhancer                                 97.842985\n",
      "RegBuild_open_chromatin_region                    98.853478\n",
      "RegBuild_promoter                                 72.114264\n",
      "RegBuild_promoter_flanking_region                 87.232802\n",
      "DNAse-seq                                         89.253789\n",
      "H3K27ac_freq                                      84.706568\n",
      "H3K27ac_score                                     84.706568\n",
      "H3K4me3_freq                                      85.930820\n",
      "H3K4me3_score                                     85.930820\n",
      "M_CpG_Freq                                        85.056354\n",
      "M_CpG_Score                                       85.056354\n",
      "M_CHG_Freq                                        98.037311\n",
      "M_CHG_Score                                       98.037311\n",
      "M_CHH_Freq                                        95.744267\n",
      "M_CHH_Score                                       95.744267\n",
      "dtype: float64\n"
     ]
    }
   ],
   "source": [
    "with pd.option_context('display.max_rows', None, 'display.max_columns', None):  # more options can be specified also\n",
    "    print((merged_chr1.isna().sum()/len(merged_chr1))*100)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 419,
   "metadata": {},
   "outputs": [],
   "source": [
    "merged_chr1.to_csv('/nfs/research1/zerbino/jhidalgo/inteql_GTEX_v8/data/processed_data/monocyte-ML-input/merged_datasets_noHiC.csv',index=False)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 421,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecular_trait_id</th>\n",
       "      <th>variant</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>chr1_169796199_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>chr1_169845687_C_CTTCCA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>chr1_169821020_T_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>chr1_169982267_A_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000000457</td>\n",
       "      <td>chr1_169787407_G_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5141</th>\n",
       "      <td>ENSG00000286198</td>\n",
       "      <td>chr1_208262975_T_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5142</th>\n",
       "      <td>ENSG00000286198</td>\n",
       "      <td>chr1_208270332_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5143</th>\n",
       "      <td>ENSG00000286216</td>\n",
       "      <td>chr1_247323434_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5144</th>\n",
       "      <td>ENSG00000286219</td>\n",
       "      <td>chr1_150320113_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5145</th>\n",
       "      <td>ENSG00000286219</td>\n",
       "      <td>chr1_150259332_C_T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5146 rows Ã— 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     molecular_trait_id                  variant\n",
       "0       ENSG00000000457       chr1_169796199_G_A\n",
       "1       ENSG00000000457  chr1_169845687_C_CTTCCA\n",
       "2       ENSG00000000457       chr1_169821020_T_G\n",
       "3       ENSG00000000457       chr1_169982267_A_T\n",
       "4       ENSG00000000457       chr1_169787407_G_C\n",
       "...                 ...                      ...\n",
       "5141    ENSG00000286198       chr1_208262975_T_G\n",
       "5142    ENSG00000286198       chr1_208270332_C_A\n",
       "5143    ENSG00000286216       chr1_247323434_C_T\n",
       "5144    ENSG00000286219       chr1_150320113_C_T\n",
       "5145    ENSG00000286219       chr1_150259332_C_T\n",
       "\n",
       "[5146 rows x 2 columns]"
      ]
     },
     "execution_count": 421,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chr1[['molecular_trait_id','variant']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 424,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 426,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene_id</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>11869</td>\n",
       "      <td>14409</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ENSG00000223972</td>\n",
       "      <td>12010</td>\n",
       "      <td>13670</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>ENSG00000227232</td>\n",
       "      <td>14404</td>\n",
       "      <td>29570</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>ENSG00000278267</td>\n",
       "      <td>17369</td>\n",
       "      <td>17436</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>ENSG00000243485</td>\n",
       "      <td>29554</td>\n",
       "      <td>31097</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234914</th>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>248906196</td>\n",
       "      <td>248919946</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234925</th>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>248906243</td>\n",
       "      <td>248918573</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234936</th>\n",
       "      <td>ENSG00000185220</td>\n",
       "      <td>248906372</td>\n",
       "      <td>248917401</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234941</th>\n",
       "      <td>ENSG00000200495</td>\n",
       "      <td>248912690</td>\n",
       "      <td>248912795</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>234944</th>\n",
       "      <td>ENSG00000233084</td>\n",
       "      <td>248936581</td>\n",
       "      <td>248937043</td>\n",
       "      <td>+</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17475 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                gene_id          3          4  6\n",
       "1       ENSG00000223972      11869      14409  +\n",
       "5       ENSG00000223972      12010      13670  +\n",
       "13      ENSG00000227232      14404      29570  -\n",
       "26      ENSG00000278267      17369      17436  -\n",
       "29      ENSG00000243485      29554      31097  +\n",
       "...                 ...        ...        ... ..\n",
       "234914  ENSG00000185220  248906196  248919946  +\n",
       "234925  ENSG00000185220  248906243  248918573  +\n",
       "234936  ENSG00000185220  248906372  248917401  +\n",
       "234941  ENSG00000200495  248912690  248912795  -\n",
       "234944  ENSG00000233084  248936581  248937043  +\n",
       "\n",
       "[17475 rows x 4 columns]"
      ]
     },
     "execution_count": 426,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gencode[['gene_id',3,4,6]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
